---
title: "Summerize for Bioinformatics with ChatGPT"
date: 2024-01-21
updated: 2024-01-21
categories:
  - å¤ä¹ ç¬”è®°
tags:
  - ç”Ÿç‰©ä¿¡æ¯å­¦
  - Bioinformatics
  - å¤ä¹ èµ„æ–™
  - å¯å‘å¼ç®—æ³•
  - PSSMæ–¹æ³•
  - GA
  - BLAST
csdn_views: 971
csdn_likes: 22
csdn_comments: 0
csdn_favorites: 13
csdn_url: https://blog.csdn.net/m0_59180666/article/details/135716837
cover: /images/posts/Summerize-for-Bioinformatics-w/517441f24ba6.png
lang_pair:
  zh-CN: "ç”Ÿç‰©ä¿¡æ¯å­¦ä¸ŽChatGPTæ€»ç»“"
---

> æœ¬æ–‡è¿ç§»è‡ªCSDNåšå®¢
> åŽŸæ–‡é“¾æŽ¥ï¼š[Summerize for Bioinformatics with ChatGPT](https://blog.csdn.net/m0_59180666/article/details/135716837)
> ðŸ“Š 971 é˜…è¯» | ðŸ‘ 22 ç‚¹èµž | ðŸ’¬ 0 è¯„è®º | â­ 13 æ”¶è—

* * *

**ç›®å½•**

Basic Introduction

Historical Events

Bioinformatics vs. Computational Biology

Levels of Bioinfo & CompBio

Molecular Biology Primer

Homework 1

A brief Introduction to DNA & RNA

A brief Introduction to Protein

Sequencing Technologies

Sequence Alignment

Pairwise sequence alignment

Basic notions

Goal of Pairwise Alignment:

Algorithms for G&L Alignments

Homework 2

Database similarity searches

BLAST: Basic Local Alignment Search Tool

FASTA: FAST-All 

Comparison between BLAST & FASTA

Homework 3

Multiple sequence alignment

1\. Progressive alignment methods

2\. Iterative alignment methods

PSSMæ–¹æ³•

Motifs and domains

Homework 4

My understanding to Expectation Maximization(EM) Algorithm:

Core Algorithm:

Code in PyCharm: (python=3.8)

The final result: 

The difficulties I met:

Machine Learning Approaches

GA for sequencing and MSA

Sequencing by Using GA

Multiple Sequence Alignments Using GA

Protein Function Prediction

Gene Function Prediction Using SVM

Protein Function Prediction Using DNN

Protein Structure Prediction

Protein Folding Based on EDA

Protein Folding Based on Deep Reinforcement Learning

AlphaFold2 for Protein Structure Prediction

References

* * *

### Basic Introduction

#### Historical Events

##### Bioinformatics vs. Computational Biology

  * Bioinformatics = the creation of tools (algorithms, databases) that solve problems. The goal is to build useful tools that work on biological data. lt is about engineering.
  * Computational biology = the study of biology using computational techniques. The goal is to learn new biology, knowledge about living systems. lt is about discovery.

##### Levels of Bioinfo & CompBio

  * Level 0: **Modeling for modeling** 's sake
  * Level 1: (entry) Use published tools to **analyze data** and **generate hypotheses** for experimentalists
  * Level 2 (Bioinfo): **Develop algorithms and databases** for data analyses on new technologies, data integration and reuse.
  * Level 3 (CompBio): Make **biological discoveries** from public dataintegration and modeling
  * Level X: Integrative studies from big consortia

* * *

#### Molecular Biology Primer

> **All life depends on 3 critical molecules**
> 
>   * DNAs 
> 
>     * Hold information on how cell works
> 
>   * RNAs 
> 
>     * Act to transfer short pieces of information to different parts  of cell 
> 
>     * Provide templates to synthesize into protein 
> 
>   * Proteins 
> 
>     * Form **enzymes(é…¶)** that send signals to other cells and regulate  gene activity 
> 
>     * Form bodyâ€™s major components (e.g. hair, skin, etc.)
> 
>     * è›‹ç™½è´¨åœ¨ç”Ÿå‘½ç³»ç»Ÿä¸­çš„é‡è¦ä½œç”¨ï¼š
> 
> â€¢ Catalysiså‚¬åŒ–ä½œç”¨
> 
> â€¢ Carrier and transportæ‰¿è¿äººå’Œè¿è¾“
> 
> â€¢ Regulatoryè°ƒèŠ‚
> 
> 

> 
> ##### **Homework 1**
> 
> Write a report describing _Cell_ , _DNA_ , _RNA_ , Proteins and discussing their relations, from information point of view. 
> 
>   * **Replication: å¤åˆ¶**
>   * **Transcription: è½¬å½•**
>   * **Translation: ç¿»è¯‘**
> 

> 
> ![](/images/posts/Summerize-for-Bioinformatics-w/517441f24ba6.png)

##### A brief Introduction to DNA & RNA

  * DNA & RNA are **polymers(èšåˆç‰©)** of**nucleotides(æ ¸è‹·é…¸)**
    * DNA is a nucleic acid, made of long chains of nucleotides, which has four kinds of bases, **A, T, C, G**. DNA is a **double strand helix(åŒé“¾èžºæ—‹)** , **hydrogen bonds(æ°¢é”®)** between **bases(ç¢±åŸº)** hold the strands together: A and T, C and G. 
      * Untwisting and replication (using DNA **polymerase(èšåˆé…¶)**): each strand is a template of a new strand
      * DNA replication begins at many specific sites

      * Each strand of the double helix is oriented in the opposite direction

      * The sequence of codons in DNA spells out the primary structure of a polypeptide

(Polypeptides form proteins that cells and organisms use)

    * RNA is also a nucleic acid: different sugar (has 1 more O on the 2nd Carbon) ;  U instead of T; Single strand, usually

##### A brief Introduction to Protein

  * Proteins are key players in our living systems. 
  * Proteins are polymers consisting of **20 kinds of amino****acids. (æ²¡æœ‰ BJOUXZè¿™å…­ä¸ªå­—æ¯)**
  * Each protein folds into a unique three-dimensional structure defined by its amino acid sequence. 
  * Protein structure has a **hierarchical nature(å±‚çº§æ€§).**
  * Protein structure is closely related to its function. 
  * Protein structure prediction is a grand challenge of computational biology

* * *

#### Sequencing Technologies

>   * 1st gen: **Sanger sequencing (æ¡‘æ ¼æµ‹åº)**
> 

>   1. **DNAç‰‡æ®µåˆ¶å¤‡ï¼š** å¤åˆ¶è¦æµ‹åºçš„DNAç‰‡æ®µã€‚
> 
>   2. **ååº”æ··åˆç‰©ï¼š** åˆ›å»ºä¸€ä¸ªååº”æ··åˆç‰©ï¼Œå…¶ä¸­åŒ…æ‹¬æ ‡å‡†æ ¸è‹·é…¸ï¼ˆAã€Tã€Cã€Gï¼‰å’Œä¸€å°éƒ¨åˆ†åŒè„±æ°§æ ¸è‹·é…¸ï¼ˆddATPã€ddTTPã€ddCTPã€ddGTPï¼‰ã€‚
> 
>   3. **DNAåˆæˆï¼š** è¿›è¡ŒDNAåˆæˆååº”ï¼Œå½“DNAé“¾ç”Ÿé•¿åˆ°å¸¦æœ‰**dideoxynucleotideï¼ˆåŒè„±æ°§æ ¸è‹·é…¸ï¼‰** çš„ä½ç½®æ—¶ï¼ŒDNAé“¾ç»ˆæ­¢ã€‚
> 
>   4. **åˆ†ç¦»DNAç‰‡æ®µï¼š** é€šè¿‡å‡èƒ¶ç”µæ³³åˆ†ç¦»ä¸åŒé•¿åº¦çš„DNAç‰‡æ®µã€‚
> 
>   5. **ç”µæ³³åˆ†æžï¼š** åˆ†ç¦»çš„DNAç‰‡æ®µå½¢æˆå¸¦çŠ¶å›¾è°±ï¼Œæ ¹æ®å…¶é•¿åº¦å¯è§†åŒ–ã€‚
> 
>   6. **ç»“æžœè§£è¯»ï¼š** é€šè¿‡åˆ†æžå›¾è°±ä¸­çš„é¢œè‰²æˆ–ä¿¡å·ï¼Œç¡®å®šDNAåºåˆ—ã€‚
> 
> 

>   * 2nd gen: **Illumina sequencing (ä¼Šå¢ç±³çº³æµ‹åº)**
> 

>   1. **æ–‡åº“åˆ¶å¤‡ï¼š** å‰ªåˆ‡DNAæˆçŸ­ç‰‡æ®µï¼Œæ·»åŠ é€‚é…å™¨å½¢æˆæ–‡åº“ã€‚
> 
>   2. **æ–‡åº“æ‰©å¢žï¼š** æ‰©å¢žæˆæ•°ç™¾ä¸‡å¤åˆ¶ã€‚
> 
>   3. **èšç±»ï¼š** å½¢æˆDNAç‰‡æ®µç°‡ã€‚
> 
>   4. **æ¡¥PCRï¼š** å°†DNAç‰‡æ®µæ¡¥æŽ¥åˆ°é€‚é…å™¨ä¸Šã€‚
> 
>   5. **æµ‹åºï¼š** é€ä¸€æ·»åŠ æ ¸è‹·é…¸ï¼Œè§‚å¯Ÿè§å…‰ä¿¡å·ï¼Œç¡®å®šDNAåºåˆ—ã€‚
> 
>   6. **æ•°æ®åˆ†æžï¼š** å¤„ç†å›¾åƒå’Œä¿¡å·ï¼Œå¾—åˆ°åŽŸå§‹æµ‹åºæ•°æ®ã€‚
> 
>   7. **åºåˆ—æ‹¼æŽ¥å’Œåˆ†æžï¼š** å°†çŸ­è¯»æ®µæ‹¼æŽ¥æˆå®Œæ•´åºåˆ—ï¼Œè¿›è¡Œè¿›ä¸€æ­¥çš„ç”Ÿç‰©ä¿¡æ¯å­¦åˆ†æžã€‚
> 
> 

>   * 3rd gen: **Third generation sequencing (ç¬¬ä¸‰ä»£æµ‹åº)**
> 

>   1. **DNAå‡†å¤‡ï¼š** æå–DNAå¹¶è¿›è¡Œè´¨æŽ§ï¼Œç¡®ä¿é€‚åˆç›´æŽ¥æµ‹åºã€‚
> 
>   2. **DNAæ¨¡æ¿åˆ¶å¤‡ï¼š** å°†DNAç‰‡æ®µå›ºå®šåœ¨SMRTç»†èƒžä¸­ï¼Œå½¢æˆå•åˆ†å­çš„DNAæ¨¡æ¿ã€‚
> 
>   3. **SMRTç»†èƒžæµ‹åºï¼š** ä½¿ç”¨èšåˆé…¶åœ¨DNAæ¨¡æ¿ä¸Šåˆæˆäº’è¡¥é“¾ï¼ŒåŒæ—¶æµ‹é‡é€ä¸ªæ ¸è‹·é…¸çš„åŠ å…¥ï¼Œå½¢æˆå®žæ—¶çš„è§å…‰ä¿¡å·ã€‚
> 
>   4. **æ•°æ®é‡‡é›†ï¼š** é€šè¿‡ç›‘æµ‹SMRTç»†èƒžä¸­çš„DNAåˆæˆè¿‡ç¨‹ï¼Œå®žæ—¶èŽ·å–é•¿è¯»é•¿çš„æµ‹åºæ•°æ®ã€‚
> 
>   5. **æ•°æ®åˆ†æžï¼š** å¯¹å¾—åˆ°çš„åŽŸå§‹æ•°æ®è¿›è¡Œå¤„ç†å’Œåˆ†æžï¼ŒåŒ…æ‹¬é”™è¯¯æ ¡æ­£å’Œåºåˆ—æ‹¼æŽ¥ã€‚
> 
> 

* * *

### Sequence Alignment

#### Pairwise sequence alignment

##### Basic notions

To express the relatedness:

  1. **identity ç›¸åŒåº¦**
  2. **similarity ç›¸ä¼¼åº¦ ï¼ˆç›¸ä¼¼ç‰©åŒ–æ€§è´¨ï¼Œå¯ä»¥è¢«æ›¿æ¢çš„ç¢±åŸºï¼‰**
     1. Calculation 1: ![\\frac{2*L_{s}}{L_{a}+L_{b}}](https://latex.csdn.net/eq?%5Cfrac%7B2*L_%7Bs%7D%7D%7BL_%7Ba%7D&plus;L_%7Bb%7D%7D) (Ls: ç›¸åŒ/ç›¸ä¼¼æ®‹åŸºå¯¹æ•°ï¼›La/Lb: ä¸¤æ¡åºåˆ—çš„é•¿åº¦)
     2. Calculation 2: ![\\frac{L_{s}}{L_{a}}](https://latex.csdn.net/eq?%5Cfrac%7BL_%7Bs%7D%7D%7BL_%7Ba%7D%7D) (Ls: ç›¸åŒ/ç›¸ä¼¼æ®‹åŸºå¯¹æ•°ï¼›La: è¾ƒçŸ­åºåˆ—çš„é•¿åº¦)
  3. **homology åŒæºæ€§ ï¼ˆåºåˆ—é—´å­˜åœ¨è¿›åŒ–å…³ç³»çš„æŽ¨æ–­ï¼‰**
     1. **Cautionï¼š åŒæºæ€§è¦ä¹ˆæœ‰è¦ä¹ˆæ²¡æœ‰ï¼Œä¸èƒ½â€œæœ‰ç‚¹â€åŒæºæ€§ï¼**
     2. **Judgementï¼š ç›¸åŒåº¦å’Œç›¸ä¼¼åº¦çš„ç™¾åˆ†æ¯”**ï¼ˆåºåˆ—è¾ƒçŸ­æ—¶ä¼šå‡ºçŽ°å¶ç„¶ï¼‰ 
        1. Solveï¼š![](/images/posts/Summerize-for-Bioinformatics-w/08d0d5ec95a7.png)
        2. **ï¼ˆâ†‘åœ¨ Safe Zoneé‡Œå°±å¯ä»¥å®‰å…¨åœ°è®¤ä¸ºå®ƒæ˜¯åŒæºçš„ï¼‰**

Amino acids varies from "R" groups.

  * Amino acids are joined by **amide linkages (é…°èƒºé”®)** to form proteins.
  * Each C-N bond is called a **peptide bond (è‚½é”®)**.

>   * A short chain of amino acids is called a **peptide (å¤šè‚½)**(usually <100). 
>   * A longer chain of amino acids is called a **protein**. 
>   * When an amino acid is part of a peptide or protein, it is called an amino acid **residue (æ®‹åŸº)**.
>   * **amino terminus** / **N-terminus (æ°¨åŸº)** , has a free amino group.
> 
>     * Residues are numbered at the beginning of N terminus.
> 
>   * **carboxyl terminus** / **C-terminus (ç¾§åŸº)** , has a free carboxyl group.
> 
> 

##### **Goal of Pairwise Alignment:**

Find best pairing of two sequences so there is maximum correspondence between residues.

**Methods:**

  * Global Alignment: æ‰¾åˆ°æœ€é€‚åˆäºŽä¸¤ä¸ªé•¿åº¦å¤§è‡´ç›¸åŒçš„é«˜åº¦ç›¸ä¼¼çš„åºåˆ—
  * Local Alignment: æ‰¾åˆ°ä¸¤ä¸ªåºåˆ—ç›¸ä¼¼æ€§æœ€é«˜çš„åŒºåŸŸï¼Œè¿™ä¸¤ä¸ªåºåˆ—çš„é•¿åº¦å¯ä»¥éžå¸¸ä¸åŒ

##### Algorithms for G&L Alignments

>   * 1\. **Dot matrix** (dot plot) **method** : ä¸€ç§ä½¿ç”¨äºŒç»´çŸ©é˜µçš„å›¾å½¢åŒ–æ–¹æ³•
> 
>     * â€‹â€‹â€‹â€‹ **æž„å»ºçŸ©é˜µï¼š** å°†ä¸¤ä¸ªåºåˆ—åˆ†åˆ«æ”¾ç½®åœ¨çŸ©é˜µçš„æ°´å¹³å’Œåž‚ç›´æ–¹å‘ä¸Šï¼Œå½¢æˆä¸€ä¸ªäºŒç»´çŸ©é˜µã€‚ 
> 
>     * **ç›¸ä¼¼æ€§æ ‡è®°ï¼š** åœ¨ç›¸åŒä½ç½®ä¸Šï¼Œå¦‚æžœä¸¤ä¸ªåºåˆ—çš„å­—ç¬¦ç›¸åŒæˆ–ç›¸ä¼¼ï¼Œå°±åœ¨è¯¥ä½ç½®ä¸Šç»˜åˆ¶ä¸€ä¸ªç‚¹ï¼ˆæˆ–å…¶ä»–ç¬¦å·ï¼‰ã€‚
> 
>     * **å¯è§†åŒ–ï¼š** æŸ¥çœ‹ç»˜åˆ¶çš„ç‚¹é˜µï¼Œè§‚å¯Ÿåºåˆ—ä¹‹é—´çš„ç›¸ä¼¼æ€§æ¨¡å¼ã€‚
> 
>   * 2.**â­** **Dynamic programming** (DP) **method** : ç”Ÿæˆä¸€ä¸ªåŒ…å«æ‰€æœ‰å¯èƒ½å¯¹é½çš„åˆ†æ•°çš„äºŒç»´å¯¹é½çŸ©é˜µï¼Œä»¥ä¾¿å®šé‡åœ°æ‰¾åˆ°æœ€ä½³å¯¹é½
> 
>     * éœ€è¦ä¸€ä¸ªè¯„åˆ†æ–¹æ³•ï¼ˆScoring Systemï¼‰ï¼š
> 
>       * ![score\(A\) = \\alpha w- \\beta x - \\gamma y](https://latex.csdn.net/eq?score%28A%29%20%3D%20%5Calpha%20w-%20%5Cbeta%20x%20-%20%5Cgamma%20y)
> 
>       * w: é…å¯¹æ•°ï¼›x: æœªé…å¯¹æ•°ï¼›y: é—´éš™ï¼ˆgapï¼‰æ•° 
> 
>         * Reward for matches:  Î± (positive score) 
> 
> Mismatch penalty: -Î² (negative score) 
> 
> Gap penalty: -Î³ (negative score) 
> 
>     * **Global alignment: Needleman-Wunsch (NW) algorithm**
>       * **å®šä¹‰çŠ¶æ€ï¼š** å°†æ¯”å¯¹é—®é¢˜åˆ’åˆ†ä¸ºä¸€ä¸ªäºŒç»´çŸ©é˜µï¼ŒçŸ©é˜µçš„æ¯ä¸ªå…ƒç´ å¯¹åº”äºŽä¸¤ä¸ªåºåˆ—çš„æŸä¸€å¯¹å­—ç¬¦ã€‚çŸ©é˜µä¸­çš„æ¯ä¸ªå…ƒç´ è¡¨ç¤ºåœ¨è¯¥ä½ç½®ä¸Šçš„æ¯”å¯¹åˆ†æ•°ï¼Œå³å¯¹åº”å­—ç¬¦çš„ç›¸ä¼¼æ€§å¾—åˆ†ã€‚
>       * **åˆå§‹åŒ–ï¼š** åœ¨çŸ©é˜µçš„ç¬¬ä¸€è¡Œå’Œç¬¬ä¸€åˆ—ä¸­è®¾ç½®åˆå§‹å€¼ï¼Œè¡¨ç¤ºç¬¬ä¸€ä¸ªåºåˆ—æˆ–ç¬¬äºŒä¸ªåºåˆ—ä¸Žç©ºåºåˆ—ä¹‹é—´çš„æ¯”å¯¹åˆ†æ•°ã€‚è¿™äº›åˆå§‹å€¼æ˜¯æ ¹æ®æ‰€é€‰æ‹©çš„æ¯”å¯¹å¾—åˆ†è§„åˆ™ç¡®å®šçš„ã€‚**C(i,0)=-i*Î³ C(0,j)=-j*Î³**
> 
>       * **çŠ¶æ€è½¬ç§»æ–¹ç¨‹ï¼š** ç¡®å®šçŠ¶æ€ä¹‹é—´çš„è½¬ç§»å…³ç³»ï¼Œå³å¦‚ä½•ä»Žå…ˆå‰çš„çŠ¶æ€å¾—åˆ°å½“å‰çŠ¶æ€ã€‚åœ¨NWç®—æ³•ä¸­ï¼ŒçŠ¶æ€è½¬ç§»æ–¹ç¨‹é€šå¸¸åŒ…æ‹¬åŒ¹é…ã€æ’å…¥å’Œåˆ é™¤æ“ä½œï¼Œå¯¹åº”äºŽåœ¨ä¸¤ä¸ªåºåˆ—ä¸­ç§»åŠ¨ã€åŒ¹é…æˆ–æ’å…¥ç¼ºå¤±å­—ç¬¦ã€‚
> 
>         * **w(a,b)=+Î±(if a=b)/-Î²(if aâ‰ b)**
> 
>         * For 1â‰¤iâ‰¤n,1â‰¤jâ‰¤m: 
>           * **C(i,j)=max{ [****C(i-1, j-1)+w(Si,Tj)], [C(i-1, j)-Î³], [C(i, j-1)-Î³]}**
>           * Si/Tj: Prefix of length i of S/T sequence
>         * ![](/images/posts/Summerize-for-Bioinformatics-w/2c8a42e8a541.png)
>       * **è®¡ç®—çŸ©é˜µï¼š** æ ¹æ®çŠ¶æ€è½¬ç§»æ–¹ç¨‹ï¼Œè®¡ç®—æ•´ä¸ªçŸ©é˜µçš„å€¼ã€‚è¿™å¯ä»¥é€šè¿‡è‡ªåº•å‘ä¸Šçš„æ–¹å¼ï¼Œä»Žå·¦ä¸Šè§’é€æ­¥è®¡ç®—åˆ°å³ä¸‹è§’ï¼Œä¹Ÿå°±æ˜¯ä»Žåºåˆ—çš„å¼€å¤´åˆ°æœ«å°¾ã€‚
> 
>       * ![](/images/posts/Summerize-for-Bioinformatics-w/d8c97735c058.png)
> 
>       * **å›žæº¯ï¼š** æ ¹æ®è®¡ç®—å¾—åˆ°çš„çŸ©é˜µï¼Œä»Žå³ä¸‹è§’å¼€å§‹å›žæº¯ï¼Œæ‰¾åˆ°æœ€ä¼˜æ¯”å¯¹è·¯å¾„ã€‚è¿™ä¸ªè·¯å¾„å°±æ˜¯ä½¿ä¸¤ä¸ªåºåˆ—å¾—åˆ†æœ€é«˜çš„æ¯”å¯¹æ–¹å¼ã€‚æ—¶é—´å¤æ‚åº¦O(mn)
> 
>     * **Local alignment: Smith-Waterman (SW) algorithm**
>       * **â€‹â€‹â€‹â€‹â€‹â€‹â€‹****å®šä¹‰çŠ¶æ€ï¼š** å°†æ¯”å¯¹é—®é¢˜åˆ’åˆ†ä¸ºä¸€ä¸ªäºŒç»´çŸ©é˜µï¼ŒçŸ©é˜µçš„æ¯ä¸ªå…ƒç´ å¯¹åº”äºŽä¸¤ä¸ªåºåˆ—çš„æŸä¸€å¯¹å­—ç¬¦ã€‚çŸ©é˜µä¸­çš„æ¯ä¸ªå…ƒç´ è¡¨ç¤ºåœ¨è¯¥ä½ç½®ä¸Šçš„æ¯”å¯¹åˆ†æ•°ï¼Œå³å¯¹åº”å­—ç¬¦çš„ç›¸ä¼¼æ€§å¾—åˆ†ã€‚(è›‹ç™½è´¨å±€éƒ¨ç›¸ä¼¼å¯èƒ½å¯¹åº”ç€â€œåŠŸèƒ½äºšåŸºâ€ç›¸ä¼¼ï¼Œè¿™æ˜¯ä¸»è¦çš„)
>       * **åˆå§‹åŒ–ï¼š** **åœ¨çŸ©é˜µçš„ç¬¬ä¸€è¡Œå’Œç¬¬ä¸€åˆ—ä¸­è®¾ç½®åˆå§‹å€¼ä¸ºé›¶** ï¼Œè¡¨ç¤ºèµ·å§‹ä½ç½®ã€‚ä¸ŽNeedleman-Wunschç®—æ³•ä¸åŒï¼ŒSmith-Watermanç®—æ³•çš„åˆå§‹åŒ–å€¼ä¸ºé›¶ï¼Œä¸è€ƒè™‘ä¸Žç©ºåºåˆ—çš„æ¯”å¯¹ã€‚
> 
>       * **çŠ¶æ€è½¬ç§»æ–¹ç¨‹ï¼š** ç¡®å®šçŠ¶æ€ä¹‹é—´çš„è½¬ç§»å…³ç³»ï¼Œå³å¦‚ä½•ä»Žå…ˆå‰çš„çŠ¶æ€å¾—åˆ°å½“å‰çŠ¶æ€ã€‚åœ¨SWç®—æ³•ä¸­ï¼ŒçŠ¶æ€è½¬ç§»æ–¹ç¨‹åŒ…æ‹¬åŒ¹é…ã€æ’å…¥ã€åˆ é™¤æ“ä½œï¼Œå¯¹åº”äºŽåœ¨ä¸¤ä¸ªåºåˆ—ä¸­ç§»åŠ¨ã€åŒ¹é…æˆ–æ’å…¥ç¼ºå¤±å­—ç¬¦ã€‚
> 
>         * **C(i,j)=max{ [****C(i-1, j-1)+score(Si,Tj)], [C(i-1, j)-Î³], [C(i, j-1)-Î³], 0 }**
> 
>       * **è®¡ç®—çŸ©é˜µï¼š** æ ¹æ®çŠ¶æ€è½¬ç§»æ–¹ç¨‹ï¼Œè®¡ç®—æ•´ä¸ªçŸ©é˜µçš„å€¼ã€‚è¿™å¯ä»¥é€šè¿‡è‡ªåº•å‘ä¸Šçš„æ–¹å¼ï¼Œä»Žå·¦ä¸Šè§’é€æ­¥è®¡ç®—åˆ°å³ä¸‹è§’ï¼Œä¹Ÿå°±æ˜¯ä»Žåºåˆ—çš„å¼€å¤´åˆ°æœ«å°¾ã€‚
> 
>       * **å›žæº¯ï¼š** æ ¹æ®è®¡ç®—å¾—åˆ°çš„çŸ©é˜µï¼Œæ‰¾åˆ°çŸ©é˜µä¸­çš„æœ€é«˜å¾—åˆ†åŠå…¶ä½ç½®ã€‚ä»Žè¯¥ä½ç½®å¼€å§‹å›žæº¯ï¼Œæ²¿ç€å¾—åˆ†æœ€é«˜çš„è·¯å¾„ï¼Œæ‰¾åˆ°æœ€ä¼˜çš„å±€éƒ¨æ¯”å¯¹ã€‚æ—¶é—´å¤æ‚åº¦O(mn)
> 
>     * **Gap Panalty Function (é—´éš™æƒ©ç½šå‡½æ•°)**
> 
>       * **â€‹â€‹â€‹â€‹â€‹â€‹â€‹ Gap penalty = _h + gk_**
> 
> where  _k =_ length of gap;  _h =_ gap opening penalty;  _g =_ gap continuation penalty ï¼ˆgï¼œh, å› ä¸ºéšç€ç¼ºå£å¼€æ”¾ä¼šæœ‰å‡ ä¸ªæ®‹åŸºä¼´éšç€æ’å…¥æˆ–åˆ é™¤ï¼Œæ‰€ä»¥åº”è¯¥å‡å°‘é¢å¤–æ®‹åŸºçš„æƒ©ç½šï¼‰
> 
>       * **æƒ©ç½šåˆ†æ•°è¦åˆç†** ï¼šå¤ªé«˜ä¼šå¯¼è‡´æ²¡æœ‰ç©ºéš™ä»Žè€Œæ²¡æœ‰åˆç†çš„å¯¹é½ï¼›å¤ªä½Žä¼šå¯¼è‡´é—´éš™å¤ªå¤šè®©ä¸ç›¸å…³çš„åºåˆ—ä»¥é«˜åˆ†å¯¹é½ã€‚ 
> 
>     * **Substitution Matrix or Scoring Matrix ï¼ˆæ›¿ä»£çŸ©é˜µ/è¯„åˆ†çŸ©é˜µï¼‰**
> 
>       * **â€‹â€‹â€‹â€‹â€‹â€‹â€‹****PAMçŸ©é˜µï¼ˆPoint Accepted Mutationï¼‰ï¼š** ç”¨äºŽæè¿°è›‹ç™½è´¨çš„æ°¨åŸºé…¸æ›¿ä»£æ¦‚çŽ‡ï¼ˆå…¨å±€æ¯”å¯¹ï¼‰ã€‚
> 
>       * **BLOSUMçŸ©é˜µï¼ˆBLOcks SUbstitution Matrixï¼‰ï¼š** ç”¨äºŽæ¯”å¯¹è›‹ç™½è´¨åºåˆ—ã€‚BLOSUMçŸ©é˜µçš„ç‰ˆæœ¬å¦‚BLOSUM62å¸¸ç”¨äºŽè›‹ç™½è´¨åºåˆ—æ¯”å¯¹ï¼ˆå±€éƒ¨æ¯”å¯¹ï¼‰ã€‚
> 
>       * **DNAçŸ©é˜µï¼š** ç”¨äºŽæ¯”å¯¹DNAåºåˆ—ï¼Œé€šå¸¸åŒ…æ‹¬ä¸åŒç¢±åŸºä¹‹é—´çš„æ›¿ä»£å¾—åˆ†ã€‚
> 
>     * **Extreme Value Distribution (æžå€¼åˆ†å¸ƒ)**
> 
>       * **ç”Ÿæˆåˆ†å¸ƒï¼š** **(ä¾‹)** é€šè¿‡å¯¹1000æ¬¡ä¸Ž1000ä¸ªéšæœºæ‰“ä¹±çš„åºåˆ—è¿›è¡Œæ¯”å¯¹ï¼Œå¾—åˆ°äº†ä¸€ä¸ªåŒ…å«1000ä¸ªæ¯”å¯¹å¾—åˆ†çš„åˆ†å¸ƒã€‚è¿™ä¸ªåˆ†å¸ƒè¢«ç§°ä¸ºExtreme Value Distributionã€‚
> 
>       * **ç»˜åˆ¶å›¾è¡¨ï¼š** åˆ©ç”¨è¿™1000ä¸ªæ¯”å¯¹å¾—åˆ†ï¼Œåˆ¶ä½œäº†ä¸€ä¸ªå›¾è¡¨ã€‚é€šè¿‡è§‚å¯Ÿå›¾è¡¨ï¼Œå¯ä»¥çœ‹å‡ºæ¯”å¯¹å¾—åˆ†çš„æ•´ä½“åˆ†å¸ƒæ¨¡å¼ï¼Œå³æžå€¼åˆ†å¸ƒçš„ç‰¹å®šæ¨¡å¼ã€‚
> 
>       * **æ¯”å¯¹åŽŸå§‹å¾—åˆ†ï¼š** å°†åŽŸå§‹çš„æ¯”å¯¹å¾—åˆ†ï¼ˆSï¼‰ä¸Žæžå€¼åˆ†å¸ƒè¿›è¡Œæ¯”å¯¹ã€‚å¦‚æžœåŽŸå§‹å¾—åˆ†ï¼ˆSï¼‰ä½äºŽåˆ†å¸ƒçš„æžå³è¾¹ç¼˜ï¼Œé‚£ä¹ˆè¯´æ˜Žè¯¥æ¯”å¯¹ç»“æžœå¾ˆå¯èƒ½ä¸æ˜¯ç”±äºŽéšæœºæƒ…å†µè€Œäº§ç”Ÿçš„ï¼Œè€Œæ˜¯ç”±äºŽåºåˆ—çš„åŒæºæ€§ã€‚
> 
>       * **è®¡ç®—P-valueï¼š** é€šè¿‡æžå€¼åˆ†å¸ƒçš„å…¬å¼![P\(X\\geq S\)=1-e^{-Kmne^{\\lambda S}}](https://latex.csdn.net/eq?P%28X%5Cgeq%20S%29%3D1-e%5E%7B-Kmne%5E%7B%5Clambda%20S%7D%7D)è®¡ç®—P-valueã€‚ï¼ˆmå’Œnæ˜¯åºåˆ—é•¿åº¦ï¼ŒKå’ŒÎ»æ˜¯åŸºäºŽè¯„åˆ†çŸ©é˜µçš„å¸¸æ•°ï¼‰P-valueè¡¨ç¤ºåŽŸå§‹æ¯”å¯¹å¾—åˆ†å‡ºçŽ°åœ¨æˆ–æ›´æžç«¯çš„ä½ç½®çš„æ¦‚çŽ‡ã€‚P-valueè¶Šå°ï¼Œè¯´æ˜Žæ¯”å¯¹ä¸å¤ªå¯èƒ½æ˜¯ç”±äºŽéšæœºæƒ…å†µè€Œäº§ç”Ÿçš„ï¼Œä»Žè€Œå¢žåŠ äº†å¯¹åŒæºæ€§çš„æŽ¨æ–­çš„å¯ä¿¡åº¦ã€‚
> 
>   * 3\. **Word method**
> 
>     * Introduced in **"** **Database similarity searches"**
> 
>     * "word method" å¯èƒ½æŒ‡ä»£å¯¹ç”Ÿç‰©åºåˆ—ä¸­çš„k-mer (ä¸€ä¸ª k-mer æ˜¯æŒ‡é•¿åº¦ä¸º k çš„å­ä¸²) è¿›è¡Œå¤„ç†å’Œåˆ†æžçš„æ–¹æ³•ã€‚ 
> 
> 

#### Homework 2

1\. There are two amino acid sequences, **seq1: COELACANTH** and **seq2: PELICAN.****** Obtain the global alignment by using DP (the Needleman-Wunsch algorithm) . 

![](/images/posts/Summerize-for-Bioinformatics-w/7233de09aa50.png)

> ![](/images/posts/Summerize-for-Bioinformatics-w/4ef17d5d907b.png)
> 
> ![](/images/posts/Summerize-for-Bioinformatics-w/c9b2a8e7da26.png)

2\. Two different dot matrix analysis servers to analyze the sequence of the human low density lipoprotein receptor(np_000518)

> 1\.  https://www.ncbi.nlm.nih.gov/protein/NP_000518.1  get the origin 
> 
> 2\.  https://www.ebi.ac.uk/Tools/seqstats/emboss_dottup/  generate the result 
> 
> 3\.  https://www.bioinformatics.nl/cgi-bin/emboss/dotmatcher  generate the result 
> 
> ![](/images/posts/Summerize-for-Bioinformatics-w/a2b6cd1c9394.png)
> 
> The long diagonal indicates that there exists a perfect match from the beginning to the end between the two sequences. (Because the two sequences are the same.)
> 
> The shorter diagonals represent the local approximate matches between the two sequences. The reason that those shorter diagonals mostly occur in the lower left corner is that there are many similar patterns before the position of 400.

#### Database similarity searches

> _" An application of Pairwise Sequence Alignment"_

> **The basic procedure for doing this:**
> 
>   * Submit the sequence of interest (the**â€œ****query****â€ sequence**) 
>   * It is aligned in a pairwise manner to EVERY sequence in the**database**
>   * Based on these pairwise comparisons, all sequences that have similarity to the query are found 
>   * Pairwise alignments between the query and each of these similar sequences are returned as output
> 

> 
> ![](/images/posts/Summerize-for-Bioinformatics-w/0b2bd240e658.png)

> **Requirements:** ï¼ˆå°è¯•è®©è¿™ä¸‰è€…è¾¾åˆ°å¹³è¡¡ï¼Œå› ä¸ºæ˜¯ç›¸äº’æŽ£è‚˜çš„ï¼‰
> 
> 1\. **sensitivity** (æ¼åˆ¤è¶Šå°‘æ•æ„Ÿæ€§è¶Šå¥½)
> 
> 2\. **specificity(selectivity)** (è¯¯åˆ¤è¶Šå°‘ç‰¹å¼‚æ€§è¶Šå¥½)
> 
> 3\. **speed** (ä»Žæ•°æ®åº“æœç´¢èŽ·å–è¾“å‡ºæ‰€éœ€çš„æ—¶é—´è¶Šå°‘é€Ÿåº¦è¶Šå¿«)
> 
> Idea: åŠ¨æ€è§„åˆ’æ˜¯ç©·ä¸¾ç®—æ³•ï¼Œæ—¶é—´æ•ˆçŽ‡ä½Žï¼Œåœ¨ä¸å½±å“å¤ªå¤šæ•æ„Ÿæ€§å’Œç‰¹å¼‚æ€§çš„æ¡ä»¶ä¸‹ï¼Œæˆ‘ä»¬å¯ä»¥é‡‡ç”¨**å¯å‘å¼ç®—æ³•(Heuristic Algorithms),** but ä¸èƒ½ä¿è¯æ‰¾åˆ°æœ€ä¼˜/æœ€å‡†ç¡®çš„ç»“æžœ
> 
> **Shortcut:** â€œTestâ€ each sequence to see whether it is worth aligning with **query_seq**
> 
> å¸¸è§çš„å¯å‘å¼ç®—æ³•ï¼š**BLAST** å’Œ **FASTA** å®ƒä»¬å°±æ˜¯æ‰€è°“çš„**WORD METHOD**
> 
>   * **Meritsï¼š** 50-100 times faster than DP
>   * **Shortsï¼š** not guaranteed to find the best/most accurate result
>   * Based on finding short stretches of identical or nearly identical letters (which are called â€œwordsâ€, â€œ _k-tuples_ â€, or â€œ _ktups_ â€) in two sequences.
> 
>   * åŸºæœ¬è¿‡ç¨‹ï¼š
> 
>     * â€‹â€‹â€‹â€‹â€‹â€‹â€‹â€‹â€‹â€‹â€‹â€‹â€‹â€‹ **å¯»æ‰¾ç›¸åŒçš„word(s)ï¼š** æ‰¾åˆ°ä¸¤ä¸ªåºåˆ—ä¸­å…±æœ‰çš„wordæˆ–è¯ç»„ï¼Œå¹¶å¯¹å®ƒä»¬è¿›è¡Œæ¯”å¯¹ã€‚ 
> 
>     * **é€šè¿‡å¯¹é½wordæ‰©å±•æ¯”å¯¹ï¼š** é€šè¿‡å¯¹é½æ¯ä¸ªwordä¸¤ä¾§çš„ç›¸ä¼¼åŒºåŸŸï¼Œæ‰©å±•æ¯”å¯¹çš„èŒƒå›´ã€‚
> 
>     * **è®¡ç®—å¯¹é½åŒºåŸŸçš„å¾—åˆ†ï¼š** å¯¹æ‰©å±•åŽçš„å¯¹é½åŒºåŸŸè¿›è¡Œå¾—åˆ†è®¡ç®—ï¼Œé€šå¸¸ä½¿ç”¨æŸç§è¯„åˆ†çŸ©é˜µæˆ–å¾—åˆ†è§„åˆ™æ¥è¡¡é‡ç›¸ä¼¼æ€§ã€‚
> 
>     * **è¿žæŽ¥é«˜åˆ†åŒºåŸŸï¼š** å°†ç›¸é‚»çš„é«˜åˆ†å¯¹é½åŒºåŸŸè¿žæŽ¥åœ¨ä¸€èµ·ï¼Œå½¢æˆæœ€ç»ˆçš„å±€éƒ¨æ¯”å¯¹ç»“æžœã€‚
> 
> 

##### **BLAST: B** asic **L** ocal **A** lignment **S** earch **T** ool

>   1. **å±€éƒ¨æ¯”å¯¹ï¼š** BLASTé‡‡ç”¨å±€éƒ¨æ¯”å¯¹çš„ç­–ç•¥ï¼Œå³å…ˆå¯»æ‰¾ä¸¤ä¸ªåºåˆ—ä¹‹é—´çš„å±€éƒ¨ç›¸ä¼¼æ€§ï¼Œç„¶åŽæ ¹æ®è¿™äº›å±€éƒ¨ç›¸ä¼¼æ€§è¿›è¡Œè¿›ä¸€æ­¥çš„åˆ†æžã€‚
> 
>   2. **æŸ¥è¯¢åºåˆ—ï¼š** ç”¨æˆ·æä¾›ä¸€ä¸ªæŸ¥è¯¢åºåˆ—ï¼Œå¯ä»¥æ˜¯DNAã€RNAæˆ–è›‹ç™½è´¨åºåˆ—ã€‚
> 
>   3. **æ•°æ®åº“æœç´¢ï¼š** BLASTå°†æŸ¥è¯¢åºåˆ—ä¸Žç›®æ ‡æ•°æ®åº“ä¸­çš„åºåˆ—è¿›è¡Œæ¯”å¯¹ã€‚ä»¥è›‹ç™½è´¨ä¸ºä¾‹ï¼Œä¸€ä¸ªé•¿åº¦ä¸ºä¸‰çš„è›‹ç™½è´¨å•è¯è¦ä¸Ž**20*20*20** =8000ä¸ªè¯æ¯”å¯¹ã€‚
> 
>   4. **æ¯”å¯¹ç®—æ³•ï¼š** BLASTä½¿ç”¨å¯å‘å¼ç®—æ³•ï¼Œé€šè¿‡æž„å»ºçŸ­è¯ï¼ˆwordsï¼‰çš„ç´¢å¼•æ¥åŠ é€Ÿæœç´¢ã€‚å®ƒé¦–å…ˆæ‰¾åˆ°ä¸ŽæŸ¥è¯¢åºåˆ—ä¸­çš„çŸ­è¯ç›¸åŒ¹é…çš„ç›®æ ‡åºåˆ—ï¼Œç„¶åŽé€šè¿‡å±€éƒ¨æ¯”å¯¹çš„æ–¹å¼é€æ­¥æ‰©å±•åŒ¹é…åŒºåŸŸã€‚å•è¯é•¿åº¦è›‹ç™½è´¨é€šå¸¸æ˜¯3ä¸ªï¼ŒDNAé€šå¸¸æ˜¯11ä¸ª
> 
>   5. **è®¡ç®—åˆ†æ•°ï¼š** BLASTæ ¹æ®ç›¸ä¼¼æ€§å¾—åˆ†ï¼ˆæ ¹æ®é€‰æ‹©çš„æ›¿ä»£çŸ©é˜µï¼Œä¾‹å¦‚BLOSUMæˆ–PAMçŸ©é˜µï¼‰å¯¹åŒ¹é…åŒºåŸŸè¿›è¡Œæ‰“åˆ†ï¼Œé»˜è®¤æ˜¯BLOSUM62ã€‚ä¸€èˆ¬æ¯”å¯¹**é˜ˆå€¼(threshold)** æ˜¯T=11. å¦‚æžœè¶…è¿‡äº†é˜ˆå€¼ï¼Œé‚£ä¹ˆå°±è®¤ä¸ºå¯ä»¥è¢«æ›¿æ¢ï¼Œä½œä¸º"seed", ä¾‹å¦‚ä¸‹å›¾ï¼š
> 
>      1. ![](/images/posts/Summerize-for-Bioinformatics-w/31b8ab8f5b38.png)
> 
>      2. å¦‚æžœåœ¨è¿™ä¸ªé…å¯¹é™„è¿‘åˆå‡ºçŽ°äº†é…å¯¹ï¼Œå¯ä»¥å°†å®ƒä»¬**è¿žæŽ¥** èµ·æ¥å½¢æˆä¸€æ®µæ›´é•¿çš„æ— é—´éš”çš„é…å¯¹ï¼ˆæ—©æœŸBLASTä¸ä¼šè¿™æ ·ï¼‰
> 
>      3. ![](/images/posts/Summerize-for-Bioinformatics-w/ef7954f03d1e.png)
> 
>      4. åŸºäºŽè¿™ä¸ªé…å¯¹åºåˆ—ï¼Œç»§ç»­å‘ä¸¤è¾¹æ‰©å±•ï¼Œç›´åˆ°åˆ†æ•°å¼€å§‹ä¸‹é™ï¼Œæœ€ç»ˆå¾—åˆ°çš„å°±æ˜¯**HSP(é«˜åˆ†æ®µé…å¯¹)**
> 
>      5. HSPåˆ†æ•°å¤§äºŽæŸ**æˆªæ–­å€¼(cutoff value)** Sæ—¶ä¿ç•™ï¼Œå¦åˆ™ä¸¢å¼ƒï¼›
> 
>      6. å•ä¸ªæ•°æ®åº“æ‰¾åˆ°å¤šä¸ªHSPå°±è¿žæŽ¥èµ·æ¥å½¢æˆ**æœ‰ç¼ºå£çš„** é…å¯¹ï¼ˆæ—©æœŸBLASTä¸ä¼šè¿™æ ·ï¼‰
> 
>      7. æœ€åŽç»™æŸ¥è¯¢åºåˆ—å’Œé…å¯¹åºåˆ—è¿›è¡ŒSWé…å¯¹ï¼Œé…å¯¹çš„æ•°æ®åº“åºåˆ—ç§°ä¸ºâ€œ**å‘½ä¸­(hits)** â€æŸ¥è¯¢åºåˆ—
> 
>   6. **è¾“å‡ºç»“æžœï¼š** BLASTè¾“å‡ºåŒ¹é…çš„åºåˆ—åŠå…¶ç›¸ä¼¼æ€§å¾—åˆ†ï¼ŒåŒæ—¶æä¾›äº†å¯è§†åŒ–çš„æ¯”å¯¹å›¾å½¢å’Œå…¶ä»–æœ‰å…³åŒ¹é…çš„ä¿¡æ¯ã€‚
> 
>   7. **P-valueï¼š** BLASTçš„è¾“å‡ºä¸­é€šå¸¸åŒ…æ‹¬P-valueï¼Œè¡¨ç¤ºåœ¨éšæœºæƒ…å†µä¸‹å¾—åˆ°ç›¸åŒæˆ–æ›´å¥½åŒ¹é…çš„æœŸæœ›æ¬¡æ•°ã€‚P-valueè¶Šå°ï¼Œè¡¨ç¤ºåŒ¹é…è¶Šæ˜¾è‘—ã€‚(We have mentioned this in the SW algorithm part)
> 
>      1. **æ³¨æ„** ï¼šå®žé™…ä¸Šmã€nçš„å®šä¹‰å˜åŒ–äº†ï¼Œåœ¨è¿™é‡Œå®ƒä»¬æŒ‡çš„éƒ½æ˜¯æœ‰æ•ˆå€¼ï¼Œéƒ½æ˜¯ç”¨**å®žé™…é•¿åº¦å‡åŽ»ä¸¤ä¸ªå¯¹é½åºåˆ—çš„å¯¹é½é•¿åº¦çš„å¹³å‡å€¼**
> 
>   8. **E-valueï¼šExpected Value** æŒ‡çš„æ˜¯the number of high-scoring matches we expect to observe between the query and unrelated sequences. è®¡ç®—å…¬å¼å¦‚ä¸‹
> 
>      1. â€‹â€‹â€‹â€‹â€‹â€‹â€‹![](/images/posts/Summerize-for-Bioinformatics-w/ea06a6081db8.png)
> 
>      2. **D** : **æ•°æ®åº“é‡Œåºåˆ—æ•°ç›®** ï¼ˆæ•°æ®åº“çš„å¤§å°sizeï¼‰
> 
>      3. **p** : æ ¹æ®æžå€¼åˆ†å¸ƒï¼ŒèŽ·å¾—å¯¹é½**å¾—åˆ†â‰¥Sçš„æ¦‚çŽ‡**
> 
>      4. **mâ€™: æ•°æ®åº“çš„æœ‰æ•ˆé•¿åº¦ï¼ˆæ•°æ®åº“å¤§å°çš„å¦ä¸€ç§åº¦é‡ï¼‰**
> 
>      5. The  **E-value** is  **the number of HSPâ€™s** with an alignment score â‰¥ S that are expected to occur  by chance when searching a database of _D_ sequences.
> 
>      6. "_The smaller the E-value, the less likely the alignment is due to chance_ "
> 
>      7. æœç´¢æŸä¸€ä¸ªæ•°æ®åº“èŽ·å¾—çš„Eå€¼ä¸èƒ½ç”¨äºŽå…¶ä»–æ•°æ®åº“ï¼Œå› ä¸ºæ•°æ®åº“å¤§å°è¶Šå¤§ï¼ŒEå€¼æ›´å€¾å‘äºŽè¶Šå¤§ã€‚ 
> 
>   9. _"A p-value is a way of representing the significance of an alignment. An E-value is simply another way of representing the significance of that alignment."_
> 
>   10. **The normalization of scores** : å¯¹é½çš„åŽŸå§‹åˆ†æ•°å–å†³äºŽè¯„åˆ†çŸ©é˜µå’Œé—´éš™æƒ©ç½šï¼Œè€Œæ ‡å‡†åŒ–åˆ†æ•°å¯ä»¥ç‹¬ç«‹äºŽè¯„åˆ†çŸ©é˜µæ¥è¯„ä¼°ä¸åŒBLASTæŸ¥è¯¢å¾—å‡ºçš„ç»“æžœæ¥åˆ¤æ–­å“ªä¸ªå¯¹é½æ›´åˆç†ã€‚
> 
>      1. â€‹â€‹â€‹â€‹â€‹â€‹â€‹è®¡ç®—å…¬å¼ ![](/images/posts/Summerize-for-Bioinformatics-w/d2ea6887b295.png)
> 
>      2. _S_ = raw alignment score åŽŸå§‹å¯¹é½åˆ†æ•°ï¼› Î» = constant that depends on scoring matrix è¯„åˆ†çŸ©é˜µç»™å‡ºçš„å¸¸æ•°ï¼› _K_ = constant that depends on scoring matrix è¯„åˆ†çŸ©é˜µç»™å‡ºçš„å¸¸æ•°
> 
>      3. ![](/images/posts/Summerize-for-Bioinformatics-w/d3d9ece75110.png)
> 
>   11. **LCR: Low Complexity Regions** ï¼š ä½Žå¤æ‚åº¦åŒºåŸŸï¼Œä¿¡æ¯é‡å¤ä¸”å•ä¸€ï¼Œéƒ½æ˜¯çŸ­è€Œé‡å¤çš„ç‰‡æ®µï¼Œä¼šå¹²æ‰°æœ€ç»ˆè¾“å‡ºçš„é…å¯¹ç‰‡æ®µï¼Œåœ¨è›‹ç™½è´¨åºåˆ—ä¸­å çº¦15%ï¼Œå¯èƒ½ä¼šå¯¼è‡´ è™šå‡å‘½ä¸­å’Œè™šé«˜é…å¯¹åˆ†æ•°ï¼ˆspurious hits and artificially high alignment scoresï¼‰ã€‚ 
> 
>      1. Avoid method 1: **Filtering**. åœ¨æŸ¥è¯¢åŒºåŸŸæŠŠ LCRç‰‡æ®µç”¨å°å†™å­—æ¯æˆ–æ­§ä¹‰å­—ç¬¦æ›¿æ¢ï¼ˆ **è›‹ç™½è´¨Xï¼ŒDNAæ˜¯N** ï¼‰ï¼Œå½“ç„¶è¿™ä¹ˆåš å¯èƒ½ä¼šé—æ¼ä¸€äº›çœŸå®žåŒ¹é… 
> 
>      2. Avoid Method 2: **Masking**. LCRä¸­çš„å­—ç¬¦åœ¨æž„ å»ºæŸ¥è¯¢å•è¯è¡¨ä¸­è¢«å¿½ç•¥ï¼Œä½†æ˜¯å¯ä»¥ åœ¨æ‰©å±•å’Œå¯¹é½ä¼˜åŒ–ä¸­ä½¿ç”¨
> 
> 

##### **FASTA: FAST-All**

>   1. **å¼€å‘è€…ï¼š** ç”± David Lipman å’Œ William Pearson äºŽ 1988 å¹´å¼€å‘ã€‚
> 
>   2. **é¦–ä¸ªå¹¿æ³›ä½¿ç”¨çš„å¿«é€Ÿæ•°æ®åº“æœç´¢ç¨‹åºï¼š** FASTA æ˜¯ç¬¬ä¸€ä¸ªè¢«å¹¿æ³›ä½¿ç”¨çš„ç”¨äºŽè¿›è¡Œå¿«é€Ÿæ•°æ®åº“æœç´¢çš„ç¨‹åºã€‚å®ƒçš„ä¸»è¦ç›®çš„æ˜¯åœ¨å¤§åž‹ç”Ÿç‰©æ•°æ®åº“ä¸­å¿«é€Ÿæ‰¾åˆ°ä¸ŽæŸ¥è¯¢åºåˆ—ç›¸ä¼¼çš„åºåˆ—ã€‚
> 
>   3. **å¯å‘å¼è¯æ³•æ–¹æ³•ï¼š** FASTA ä½¿ç”¨ä¸€ç§å¯å‘å¼çš„ word æ–¹æ³•æ¥åˆ›å»ºæŸ¥è¯¢åºåˆ—ä¸Žæ•°æ®åº“åºåˆ—ä¹‹é—´çš„å±€éƒ¨æ¯”å¯¹ã€‚è¯¥æ–¹æ³•é€šè¿‡å¯»æ‰¾ä¸¤ä¸ªåºåˆ—ä¸­çš„è¯ï¼ˆwordsï¼‰çš„ç›¸ä¼¼æ€§æ¥è¿›è¡Œæ¯”å¯¹ã€‚
> 
>   4. **ä»Žç²¾ç¡®åŒ¹é…çš„è¯å¼€å§‹ï¼š** FASTA åœ¨æ¯”å¯¹è¿‡ç¨‹ä¸­ä»Žå¯»æ‰¾ä¸¤ä¸ªåºåˆ—ä¸­çš„è¯çš„ç²¾ç¡®åŒ¹é…å¼€å§‹ã€‚åœ¨è›‹ç™½è´¨åºåˆ—ä¸­ï¼Œé€šå¸¸ä½¿ç”¨ ktup = 1 æˆ– 2ï¼Œè€Œåœ¨DNAåºåˆ—ä¸­ï¼Œé€šå¸¸ä½¿ç”¨ 4-6ã€‚
> 
> 

##### Comparison between BLAST & FASTA

> â€¢ BLAST is faster 
> 
> â€¢ FASTA generally produces a better final alignment 
> 
> â€¢ FASTA is more likely to find distantly related sequences 
> 
> â€¢ Performance of the two methods is similar for highly similar sequences 
> 
> â€¢ Both methods are appropriate for rapid initial searches

#### Homework 3

**1.** Answer the following questions about scoring an alignment.

**A.** Calculate the score for the DNA sequence alignment shown below, using the scoring matrix below. Use an affine gap penalty to score the gaps, with -11 for opening the gap and -1 for each additional position in the gap. (â€œAffine gap penaltyâ€ refers to a situation when the gap opening and gap extension penalties are not the same. The gap opening penalty should be greater than the gap extension penalty.) 

![](/images/posts/Summerize-for-Bioinformatics-w/5ccd10240cec.png)

> **2x27â€“5x4â€“11â€“2=21 **
> 
> Match scores: 27 matching bases: 27 * 2 = 54 points
> 
> Mismatch score: 4 mismatched base: 4 * -5 = -20 points
> 
> Gap scores: 1 gap in the first sequence: 1 * -11 = -11 points  
>  Each extension of the gap: 2 * -1 = -2 points 
> 
> The total gap score: -11 - 2 = -13 points
> 
> Overall alignment score = Match scores + Mismatch scores + Total Gap score = 54 + (-20) + (-13) = ****21****

**B.** How would the score change if you were to use a nonaffine gap penalty? To answer the question, try a nonaffine penalty of -2, and then -6.

> 2x27 â€“ 5x4 â€“ 3x2 = 28(for nonaffine penalty -2) 
> 
> 2x27 â€“ 5x4 â€“ 3x6 = 16(for nonaffine penalty -6)
> 
> With non-affine gap penalty of -2 per gap:
> 
> There would now be 1 gap that receives a linear penalty of -2 * 3 bases = -6 
> 
> Overall score is: Matches (54) + Mismatch (-20) \+ Gaps (-6) = 54-20-6 = ****28****
> 
> With non-affine penalty of -6 per gap: 
> 
> The 1 gap gets a score of -6 * 1 gap = -6 
> 
> Overall score is: Matches (54) + Mismatch (-20) + Gaps (-18) = 54-20-18 = ****16****

**2.** Below is part of the BLOSUM62 matrix. Answer the following questions using this matrix.

![](/images/posts/Summerize-for-Bioinformatics-w/763d47eb658b.png)

![](/images/posts/Summerize-for-Bioinformatics-w/d5030e19f033.png)

**A.** Using this matrix, two aligned cysteines (C) would receive a score of 9 while two aligned threonines (T) would only receive a score of 5. What can you conclude about cysteine relative to threonine?

> Cysteines (C) is rarer than threonines (T) . The higher the match score the rarer the match.

**B.** A serine (S) aligned with a cysteine (C) would receive a negative score (-1) while a serine aligned with a threonine would receive a positive score (1). Offer a possible explanation for this in terms of physicochemical properties of the amino acid side chains.

> The Mercapto is difficult to generate hydrogen bond than Hydroxyl 

****C.******** The bestng an alignment score. Use the BLOSUM62 matrix alignment of the two amino acid sequences â€œLDSâ€ and â€œLNSâ€ is obvious (itâ€™s shown below). Given a scoring system, you could easily calculate an alignment score. Set up a matrix and use the dynamic programming algorithm to â€œproveâ€ that this is the best alignment by calculating x (see Powerpoint notes) to score aligned residues, and use a gap penalty of -1. (You may hand write the matrix in your homework rather than typing it if you like.

![](/images/posts/Summerize-for-Bioinformatics-w/0dd1f343c9a7.png)

> ![](/images/posts/Summerize-for-Bioinformatics-w/cea6d3abe084.png)![](/images/posts/Summerize-for-Bioinformatics-w/252617ffc1b5.png)
> 
> It receives an alignment score of 4 \+ 1 \+ 4 = 9
> 
> The matrix verifies this is the maximum possible alignment score using these sequences and rules. So it proves this alignment is optimal.

**3****.** Find the optimal global alignment of the two sequences **_Seq_ 1: THISLINE and _Seq_ 2: ISALIGNED **based on the BLOSUM62 matrix with linear gap penalty of -4.

> ![](/images/posts/Summerize-for-Bioinformatics-w/526bbd13b42a.png)
> 
> ![](/images/posts/Summerize-for-Bioinformatics-w/067e65bdb364.png)

#### Multiple sequence alignment

![](/images/posts/Summerize-for-Bioinformatics-w/e6808c3bf895.png)

*å…¨åŠ¨æ€ç¼–ç¨‹å¯¹äºŽå¤šåºåˆ—æ¯”å¯¹æ˜¯ä¸å®žç”¨çš„ã€‚åºåˆ—é•¿æ®‹åŸºå¤šæ—¶å€™æ¯”è¾ƒæŒ‡æ•°å¢žé•¿

* Categories of Heuristic Algorithms for Multiple Sequence Alignment:

##### 1\. Progressive alignment methods

ã€first calculate a pairwise alignment for the two most closely related sequences and 

progressively add more sequences to the alignment.ã€‘P8/p16

![](/images/posts/Summerize-for-Bioinformatics-w/8a232a542315.png)

##### 2\. Iterative alignment methods

ã€Iterative alignment methods attempt to correct a problem of progressive methods: 

Errors in the initial alignments of the most closely related sequences are propagated to the final multiple sequence alignment.ã€‘

![](/images/posts/Summerize-for-Bioinformatics-w/ae7fb1ebf981.png)

##### ****PSSMæ–¹æ³•****

> ã€PSSMä¸­çš„æ¯ä¸ªå€¼æ˜¯åœ¨å¤šåºåˆ—æ¯”å¯¹ä¸­ç»™å®šä½ç½®å¤„è§‚å¯Ÿåˆ°çš„è¯¥æ°¨åŸºé…¸çš„è®¡æ•°é™¤ä»¥åœ¨æ¯”å¯¹ä¸­è¯¥ä½ç½®å¤„è¯¥æ°¨åŸºé…¸çš„é¢„æœŸè®¡æ•°çš„æ¯”çŽ‡ã€‚ç„¶åŽå°†è¯¥å€¼è½¬æ¢ä¸ºå¯¹æ•°åˆ»åº¦ï¼ˆé€šå¸¸ä»¥2ä¸ºåŸºæ•°ï¼‰ã€‚
> 
> PSSMä¸­çš„å€¼åæ˜ äº†ä»»ä½•ç»™å®šæ°¨åŸºé…¸å‡ºçŽ°åœ¨æ¯”å¯¹ä¸­çš„æ¯ä¸ªä½ç½®çš„æ¦‚çŽ‡ï¼Œä»¥åŠåœ¨æ¯”å¯¹ä¸­æ¯ä¸ªä½ç½®çš„ä¿å®ˆæˆ–éžä¿å®ˆå–ä»£çš„å½±å“ã€‚ã€‘

#### Motifs and domains

> *Motifâ€“ a short conserved sequence pattern; can be just a few amino acid residues, up to ~20.
> 
> *Domainâ€“ a longer conserved sequence pattern which adopts a particular three-dimensional structure and is an independent functional and structural unit; typically 40-700 residues.
> 
> *å‰å¸ƒæ–¯é‡‡æ ·ç®—æ³•å‰å¸ƒæ–¯é‡‡æ ·æ˜¯ä¸€ä¸ªåŽŸç†ä¸Šç±»ä¼¼äºŽEMçš„è¿­ä»£è¿‡ç¨‹ï¼Œä½†ç®—æ³•ä¸åŒã€‚åœ¨æ¯æ¬¡è¿­
> 
> ä»£ä¸­ï¼Œç§»é™¤ä¸€ä¸ªåºåˆ—ï¼Œå¹¶ä»Žå‰©ä½™åºåˆ—ä¸­æž„å»ºè¯•éªŒæ¯”å¯¹ã€‚ä¸ŽEMä¸€æ ·ï¼ŒGibbsé‡‡æ ·åœ¨ä¸€ç»„åº
> 
> åˆ—ä¸­æœç´¢ç»Ÿè®¡ä¸Šæœ€å¯èƒ½çš„åŸºåºï¼Œå¹¶å¯ä»¥åœ¨æ¯ä¸ªåºåˆ—ä¸­æ‰¾åˆ°è¿™äº›åŸºåºçš„æœ€ä½³å®½åº¦å’Œæ•°é‡ã€‚
> 
> *é…ç½®æ–‡ä»¶HMMï¼šè½®å»“HMMå¯ä»¥ä»Žä¸€ç»„æœªå¯¹é½çš„åºåˆ—ä¸­æž„å»ºã€‚å¯¹æ¨¡åž‹çš„é•¿åº¦ï¼ˆåŒ¹é…çŠ¶æ€çš„æ•°é‡ï¼‰å’Œæ¦‚çŽ‡å‚æ•°è¿›è¡Œäº†åˆæ­¥çŒœæµ‹ã€‚ç„¶åŽé€šè¿‡è¿­ä»£æ›´æ–°æ¦‚çŽ‡å‚æ•°æ¥è®­ç»ƒæ¨¡åž‹ï¼ˆè¯¥è¿‡ç¨‹å¯ä»¥ä½¿ç”¨å„ç§ç®—æ³•ï¼‰ã€‚
> 
> * EM-algorithmï¼šStep 1. Maximization Step 2. Expectation Step 3. Repeat Step 1 and 2.

>   1. æœŸæœ›æœ€å¤§åŒ–(Expectation Maximization, EM)ç®—æ³• è¯¥ç®—æ³•é€šè¿‡è¿­ä»£çš„æ–¹æ³•ä»Žä¸€ç»„åºåˆ—ä¸­å‘çŽ°ç»Ÿè®¡ä¸Šæœ€å¯èƒ½çš„åŸºåºæ¨¡å¼(motifæˆ–domain)ã€‚å®ƒä¸ä¾èµ–äºŽäº‹å…ˆäº§ç”Ÿçš„å¤šé‡åºåˆ—æ¯”å¯¹ã€‚
>   2. å‰å¸ƒæ–¯é‡‡æ ·(Gibbs Sampler)ç®—æ³•  
>  ä¸ŽEMç®—æ³•ç±»ä¼¼çš„è¿­ä»£è¿‡ç¨‹,ä¹Ÿèƒ½åœ¨ä¸€ç»„åºåˆ—ä¸­å‘çŽ°motifæˆ–domain,å¹¶èƒ½æ‰¾å‡ºæ¯æ¡åºåˆ—ä¸­è¿™äº›æ¨¡å¼çš„æœ€ä½³é•¿åº¦å’Œæ•°é‡ã€‚
>   3. é…ç½®éšé©¬å°”å¯å¤«æ¨¡åž‹(Profile Hidden Markov Model) å¯ä»¥ä»Žä¸€ç»„æœªæ¯”å¯¹çš„åºåˆ—ä¸­æž„å»ºã€‚é€šè¿‡è¿­ä»£æ›´æ–°æ¦‚çŽ‡å‚æ•°å¯¹æ¨¡åž‹è¿›è¡Œè®­ç»ƒã€‚
> 

> 
> è¿™äº›éƒ½æ˜¯ç”¨ç»Ÿè®¡å­¦æ–¹æ³•ä»Žåºåˆ—æ•°æ®é›†ä¸­å­¦ä¹ motifå’Œdomainçš„æ— ç›‘ç£å­¦ä¹ ç®—æ³•ã€‚ä¸éœ€è¦äº‹å…ˆçš„åºåˆ—æ¯”å¯¹å°±èƒ½å‘çŽ°åºåˆ—æ¨¡å¼ã€‚

#### Homework 4

  1. **Identifying protein motifs using Expectation Maximization (EM) algorithm.**

Suppose we have 200 protein sequences of length 100, in the dataset. Each character represents an amino acid. The length of the motif is 10. The figure below shows the first 10 samples in the dataset.

![](/images/posts/Summerize-for-Bioinformatics-w/bd76fccc0100.png)

![](/images/posts/Summerize-for-Bioinformatics-w/d1330c486812.png)Write a program to identify the starting position of the motif in each protein sequence. Write a report which includes: a) your understanding on EM algorithm, b) the core algorithm, c) the final result (starting positions and PSSM of the motif), d) the difficulties you met.

> ![](/images/posts/Summerize-for-Bioinformatics-w/a9c5e768b0b0.png)
> 
> ![](/images/posts/Summerize-for-Bioinformatics-w/272b78939fdd.png)
> 
> ![](/images/posts/Summerize-for-Bioinformatics-w/b39e68e2bfa6.png)
> 
>   * ##### **My understanding to Expectation Maximization(EM) Algorithm:**
> 
> 

>   1. The EM algorithm is an iterative method that estimates model parameters where there are missing, hidden, or unobserved values.
>   2. It proceeds in two steps that are repeated until convergence: 
>      1. Expectation (E) Step: Creating a function for the expectation of the log-likelihood evaluated using the current estimate for parameters. This involves estimating values for latent variables based on the current parameter values.
>      2. Maximization (M) Step: Finding parameters that maximize the expected log-likelihood found on the E step. These become the new parameter estimates.
>   3. By iteratively doing E and M steps, EM finds the parameters that locally maximize the likelihood of the data comprising both observed and unobserved values.
>   4. For motif finding, the "unobserved" data are the true starting positions of each motif instance. The expectation step estimates motif locations, while the maximization step finds an updated motif model that fits best based on those expected motif positions.
> 

>   * ##### **Core Algorithm:**
> 
> 

> 
> Fake code:
> 
>   1. Initialize motif starting positions randomly in each sequence
>   2. E-step: 
>      1. Extract motif instances from each sequence based on current start positions
>      2. Calculate probability of each position being the true start given current motif model
>   3. M-step: 
>      1. Update start position for each sequence to position with highest probability from E-step
>      2. Update PSSM for motif based on new aligned motif instances
>   4. Repeat E-step and M-step until convergence
>   5. Output final motif start positions and PSSM
> 

> 
> ##### Code in PyCharm: (python=3.8)
> 
> ```python # Created at WASEDA Univ. # Author: Vector Kun # Time: 2023/12/17 12:44 import csv from random import randint as randint from math import log as log import matplotlib.pyplot as plt from numpy import mean from matplotlib.pyplot import MultipleLocator # Define basic parameters proteinsDataset = "proteins.csv" resultFile = "result.txt" trainNum = 200 trainTime = 10 motif_length = 10 def csv_reader(path): """Read data from csv""" origin_protein_list = [] with open(path) as f: reader = csv.reader(f) for item in reader: origin_protein_list.append(item) # print(origin_protein_list) return origin_protein_list def data_preprocess(path): """Remove order in the data""" protein_order_list = csv_reader(path)[1:] protein_list = [] for protein in protein_order_list: protein_list.append(protein[1:]) # print(protein_list) return protein_list def matrix_transposition(matrix): """Transpose the matrix""" new_matrix = [] for i in range(len(matrix[0])): matrix1 = [] for j in range(len(matrix)): matrix1.append(matrix[j][i]) new_matrix.append(matrix1) # print(new_matrix) return new_matrix def PSSM_calculation(seqs): """Calculate pssm matrix""" # The species of the aminos amino_acids_list = ['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y'] # PSSM matrix weight_matrix = {} # Frequency of amino acids fre_matrix = {} # Number of occurrences of amino acids at each position count_matrix = {} # Initialization for amino in amino_acids_list: count_matrix[amino] = [0] * motif_length fre_matrix[amino] = 0 weight_matrix[amino] = [0] * motif_length # Calculate the number of amino acids in each position col_num = 0 new_seqs = matrix_transposition(seqs) for col in new_seqs: for amino in col: i = col_num count_matrix[amino][i] += 1 col_num += 1 # Calculate the frequency of amino acids for amino in count_matrix: amino_sum = 0 for count in count_matrix[amino]: amino_sum += count frequency = amino_sum / (len(seqs) * motif_length) fre_matrix[amino] = frequency # Calculate weight matrix for i in range(0, motif_length): for amino in count_matrix: # Calculate WM by dividing it by the total nums of swqs count_divided_nums = count_matrix[amino][i] / len(seqs) if fre_matrix[amino] == 0: weight = 0 # Calculate WM by dividing it by the frequency of the aminos else: weight = count_divided_nums / fre_matrix[amino] # Logarithmic operation on matrix if weight == 0: weight = -10 else: weight = log(weight, 10) weight_matrix[amino][i] += float(format(weight, '.4f')) # print(count_matrix, '\n', fre_matrix, '\n', weight_matrix, '\n') return weight_matrix def get_random_seqs(protein_list): """Randomly initialize the starting position""" train_seq = [] for i in range(0, trainNum): head = randint(0, 90) tail = head + motif_length seq = protein_list[i][head:tail] train_seq.append(seq) # print(train_seq) return train_seq def cal_score(seqs, weight_matrix): """Calculate the score based on sequences and the WM""" score = 0 for i in range(0, motif_length): score += weight_matrix[seqs[i]][i] return score def get_train_seqs(protein_list, weight_matrix): """Calculate scores os all the motifs with different starting position, select the optimal position """ max_score = [] train_seqs = [] max_index = [] for protein in protein_list: score_list = [] for i in range(0, 90): test_seq = protein[i:i + motif_length] # Calculate scores of different position in one protein score_list.append(cal_score(test_seq, weight_matrix)) # Get the max one max_position = score_list.index(max(score_list)) # Get the optimal sequences of one protein new_train_seq = protein[max_position:max_position + motif_length] # Get the best score to cal ave max_score.append(max(score_list)) train_seqs.append(new_train_seq) max_index.append(max_position) return train_seqs, max_index, max_score def data_training(protein_list): """Train the data""" # Number of iterations iteration_num = 100 Best_ave = 0 Best_iteration = 0 for i in range(iteration_num): # Average score of every iteration max_score_list = [] # Initialize the position randomly first train_seq = get_random_seqs(protein_list) # Calculate PSSM matrix weight_matrix = PSSM_calculation(train_seq) for j in range(0, iteration_num): # Estimate and choose the best position new_train_seqs, max_index, max_score = get_train_seqs(protein_list, weight_matrix) # Calculate new matrix weight_matrix = PSSM_calculation(new_train_seqs) max_score_list.append(mean(max_score)) # Plot convergence generation = [] for k in range(1, iteration_num + 1): generation.append(k) max_score_list = [] plt.plot(generation, max_score_list) y_major_locator = MultipleLocator(0.3) ax = plt.gca() ax.yaxis.set_major_locator(y_major_locator) plt.ylabel('Mean scores', fontsize=10) plt.show() # Write the result to txt f = open("result.txt", "a") f.write("\n" + f"Iteration {i}" + '\n' + "Starting positions:" + str(max_index)) f.write('\n' + "PSSM:" + str(weight_matrix)) f.write('\n' + "Scores:" + str(max_score_list)) f.write('\n' + "Average best score:" + str(mean(max_score))) # Get the max score if Best_ave <= mean(max_score): Best_ave = mean(max_score) Best_iteration = i print(f"Iteration {i}: Best average score = {Best_ave}") print('Best iteration: ', Best_iteration) if __name__ == '__main__': fl = open("result.txt", "wt") fl.close() # Get the protein list proteinList = data_preprocess(proteinsDataset) data_training(proteinList) ``` 
> 
>   * ##### **The final result:**
> 
> 
```python D:\Anaconda3\envs\spider\python.exe Iteration 0: Best average score = 4.3729255 Best iteration: 0 Iteration 1: Best average score = 5.199039 Best iteration: 1 Iteration 2: Best average score = 5.199039 Best iteration: 1 Iteration 3: Best average score = 5.199039 Best iteration: 1 Iteration 4: Best average score = 6.680054999999999 Best iteration: 4 Iteration 5: Best average score = 6.680054999999999 Best iteration: 4 Iteration 6: Best average score = 6.680054999999999 Best iteration: 4 Iteration 7: Best average score = 6.680054999999999 Best iteration: 4 Iteration 8: Best average score = 6.680054999999999 Best iteration: 4 Iteration 9: Best average score = 6.680054999999999 Best iteration: 4 Iteration 10: Best average score = 6.680054999999999 Best iteration: 4 Iteration 11: Best average score = 6.680054999999999 Best iteration: 4 Iteration 12: Best average score = 6.680054999999999 Best iteration: 4 Iteration 13: Best average score = 6.680054999999999 Best iteration: 4 Iteration 14: Best average score = 6.680054999999999 Best iteration: 4 Iteration 15: Best average score = 6.680054999999999 Best iteration: 4 Iteration 16: Best average score = 6.680054999999999 Best iteration: 4 Iteration 17: Best average score = 6.680054999999999 Best iteration: 4 Iteration 18: Best average score = 6.680054999999999 Best iteration: 4 Iteration 19: Best average score = 6.680054999999999 Best iteration: 4 Iteration 20: Best average score = 6.680054999999999 Best iteration: 4 Iteration 21: Best average score = 6.680054999999999 Best iteration: 4 Iteration 22: Best average score = 6.680054999999999 Best iteration: 4 Iteration 23: Best average score = 6.680054999999999 Best iteration: 4 Iteration 24: Best average score = 6.680054999999999 Best iteration: 4 Iteration 25: Best average score = 6.680054999999999 Best iteration: 4 Iteration 26: Best average score = 6.680054999999999 Best iteration: 4 Iteration 27: Best average score = 6.680054999999999 Best iteration: 4 Iteration 28: Best average score = 6.680054999999999 Best iteration: 4 Iteration 29: Best average score = 6.680054999999999 Best iteration: 4 Iteration 30: Best average score = 6.680054999999999 Best iteration: 4 Iteration 31: Best average score = 6.680054999999999 Best iteration: 4 Iteration 32: Best average score = 6.832890500000001 Best iteration: 32 Iteration 33: Best average score = 6.832890500000001 Best iteration: 32 Iteration 34: Best average score = 6.832890500000001 Best iteration: 32 Iteration 35: Best average score = 6.832890500000001 Best iteration: 32 Iteration 36: Best average score = 6.832890500000001 Best iteration: 32 Iteration 37: Best average score = 6.855327999999999 Best iteration: 37 Iteration 38: Best average score = 6.870583999999999 Best iteration: 38 Iteration 39: Best average score = 6.870583999999999 Best iteration: 38 Iteration 40: Best average score = 6.9964825 Best iteration: 40 Iteration 41: Best average score = 6.9964825 Best iteration: 40 Iteration 42: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 43: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 44: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 45: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 46: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 47: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 48: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 49: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 50: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 51: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 52: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 53: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 54: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 55: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 56: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 57: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 58: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 59: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 60: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 61: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 62: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 63: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 64: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 65: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 66: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 67: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 68: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 69: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 70: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 71: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 72: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 73: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 74: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 75: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 76: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 77: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 78: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 79: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 80: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 81: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 82: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 83: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 84: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 85: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 86: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 87: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 88: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 89: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 90: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 91: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 92: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 93: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 94: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 95: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 96: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 97: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 98: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 99: Best average score = 7.2110034999999995 Best iteration: 42 Process finished with exit code 0 ``` 
> 
> From the output, we can get:
> 
> Iteration 99: Best average score = 7.2110034999999995  
>  Best iteration: 42
> 
> So the best average score of 100 iterations is roughly 7.211, which iteration number is 42.
> 
> The **Convergence Plot** is shown below:
> 
> ![](/images/posts/Summerize-for-Bioinformatics-w/57f6ffdb711a.png)
> 
> The result to iteration 42 in result.txt is:
> 
> ```python Iteration 42 Starting positions:[40, 23, 0, 57, 8, 73, 49, 85, 29, 46, 82, 4, 49, 14, 32, 23, 78, 41, 0, 43, 69, 82, 76, 3, 25, 5, 36, 82, 34, 74, 30, 89, 15, 54, 42, 49, 84, 66, 71, 22, 23, 11, 51, 31, 77, 45, 30, 83, 48, 7, 80, 45, 60, 33, 84, 82, 86, 13, 79, 22, 21, 12, 68, 69, 81, 38, 68, 64, 89, 65, 47, 58, 53, 12, 56, 30, 12, 72, 16, 85, 57, 42, 86, 81, 30, 8, 6, 65, 32, 82, 24, 53, 11, 26, 15, 75, 44, 41, 56, 7, 27, 76, 34, 63, 46, 9, 32, 35, 13, 27, 88, 78, 86, 76, 26, 26, 74, 44, 51, 74, 29, 87, 46, 8, 64, 34, 13, 5, 57, 19, 80, 10, 68, 1, 28, 23, 33, 34, 56, 88, 63, 41, 57, 31, 30, 29, 8, 51, 26, 29, 31, 58, 69, 84, 47, 1, 55, 88, 34, 55, 65, 66, 7, 7, 89, 76, 60, 58, 37, 73, 84, 87, 24, 6, 18, 14, 83, 1, 60, 16, 68, 42, 31, 10, 47, 8, 57, 32, 52, 51, 32, 59, 64, 2, 72, 26, 66, 58, 7, 5] PSSM:{'A': [0.9723, -1.2504, -0.9494, -1.2504, -10.0, -1.2504, -0.6484, -0.9494, -10.0, -10.0], 'C': [-1.5514, -1.5514, 0.6891, -1.2504, -1.5514, -10.0, -1.2504, 0.6841, -1.2504, -1.5514], 'D': [-0.0414, -0.3424, -0.3424, -10.0, 0.1347, 0.1347, 0.1347, -0.0414, 0.3565, -0.0414], 'E': [-0.0414, 0.2596, -0.0414, -10.0, -0.0414, -0.0414, -10.0, -10.0, 0.2596, 0.4357], 'F': [0.1549, 0.331, 0.5528, -10.0, -10.0, -0.1461, -0.1461, -10.0, 0.1549, -10.0], 'G': [-0.9661, -0.6651, -0.7901, -0.9661, -10.0, -1.2672, -10.0, -0.9661, 0.9658, -10.0], 'H': [-10.0, 0.0706, 0.2467, 0.2467, -10.0, -10.0, -0.2304, 0.3716, 0.0706, 0.0706], 'I': [-10.0, -10.0, 0.301, -10.0, 0.426, -10.0, -0.1761, 0.301, 0.1249, 0.1249], 'K': [-0.1761, 0.426, -10.0, -0.1761, 0.426, -0.1761, -0.1761, -10.0, 0.1249, -0.1761], 'L': [-1.2648, -1.2648, -1.2648, -1.2648, -10.0, 0.9807, -0.7877, -1.2648, -10.0, -10.0], 'M': [-10.0, -0.0792, -10.0, 0.6198, 0.2218, -10.0, 0.2218, -10.0, -0.0792, -0.0792], 'N': [0.5229, -10.0, -10.0, -0.0792, -10.0, -0.0792, -10.0, -0.0792, -0.0792, 0.5229], 'P': [-10.0, -10.0, -1.2492, 0.6977, -10.0, -1.5502, -1.5502, -10.0, -1.5502, 0.6878], 'Q': [0.301, -10.0, -0.1761, -10.0, -0.1761, 0.301, -0.1761, 0.1249, 0.1249, 0.1249], 'R': [-0.0212, 0.1549, -10.0, 0.2798, -0.3222, -0.0212, 0.4559, -0.0212, -10.0, -0.3222], 'S': [-0.7924, 0.9685, -10.0, -1.2695, -10.0, -0.9685, -0.7924, -0.6675, -10.0, -10.0], 'T': [-1.2553, -1.2553, -1.2553, -10.0, -0.9542, -10.0, 0.9726, -10.0, -0.9542, -0.6532], 'V': [-0.9845, -0.9845, -10.0, -1.2856, 0.9721, -0.9845, -0.9845, -1.2856, -0.9845, -10.0], 'W': [0.426, -0.1761, 0.301, -10.0, -10.0, 0.1249, -10.0, -0.1761, 0.301, -0.1761], 'Y': [0.4559, -10.0, -10.0, -0.1461, -10.0, 0.331, -10.0, 0.331, -10.0, 0.331]} Scores:[1.6438155, 7.475081, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999] Average best score:7.2110034999999995 ``` 
> 
>   * ##### **The difficulties I met:**
> 
> 

>   1. Determining suitable stopping criteria and determining when the algorithm has converged to a stable solution. In practice, heuristics like a maximum number of iterations are used.
>   2. Avoid getting stuck in poor local optima rather than the global optimum. Using multiple random restarts can mitigate this issue.
>   3. Runtime complexity for large datasets. Matrix computations may be intensive for large sequence sets and long motifs. Parallelization and approximation methods can improve scalability.
>   4. Interpreting the resulting motif model. Statistical significance testing should be done to evaluate if the identified motif represents a true common signal vs random shared patterns.
> 

* * *

### Machine Learning Approaches

#### GA for sequencing and MSA

##### Sequencing by Using GA

> GA â€“ Genetic Algorithmï¼ˆåŸºå› äº’æ¢ã€å˜å¼‚ï¼‰
> 
> EDA â€“ Estimation of Distribution Algorithm
> 
> ![](/images/posts/Summerize-for-Bioinformatics-w/0fc8221dff2d.png)
> 
> *Sequencing by Using GA
> 
> ![](/images/posts/Summerize-for-Bioinformatics-w/843d7478e4ff.png)

##### Multiple Sequence Alignments Using GA

*Using Genetic Algorithms for Pairwise and Multiple Sequence Alignments

#### Protein Function Prediction

##### Gene Function Prediction Using SVM

> æœ¬èŠ‚è®¨è®ºäº†ä½¿ç”¨SVMè¿›è¡ŒåŸºå› åŠŸèƒ½é¢„æµ‹çš„å¤šæ ‡ç­¾åˆ†ç±»å’Œå±‚æ¬¡å¤šæ ‡ç­¾åˆ†ç±»æ–¹æ³•ã€‚
> 
> ä¸»è¦å†…å®¹æœ‰:
> 
> 1\. åŸºäºŽæ ‡ç­¾æŽ’åºå’Œç²¾ç»†å†³ç­–è¾¹ç•ŒSVMçš„å¤šæ ‡ç­¾åˆ†ç±»æ–¹æ³•ã€‚è¯¥æ–¹æ³•è€ƒè™‘äº†æ ·æœ¬é‡å çš„ä¿¡æ¯,é€šè¿‡Biasæ¨¡åž‹æ”¹è¿›é‡å æ ·æœ¬çš„åˆ†ç±»å‡†ç¡®çŽ‡;å¹¶æå‡ºäº†ä¸€ä¸ªåŸºäºŽè¾“å…¥å®žä¾‹å’Œæ ‡ç­¾æŽ’åºçš„å®žä¾‹ç›¸å…³é˜ˆå€¼é€‰å–ç­–ç•¥ã€‚
> 
> 2\. åŸºäºŽè¿‡é‡‡æ ·å’Œå±‚æ¬¡çº¦æŸçš„å±‚æ¬¡å¤šæ ‡ç­¾åˆ†ç±»æ–¹æ³•ã€‚è¯¥æ–¹æ³•é€šè¿‡æå‡ºåˆ†å±‚SMOTEæ–¹æ³•è§£å†³æ•°æ®é›†ä¸å¹³è¡¡é—®é¢˜;å¹¶é€šè¿‡åœ¨TPRè§„åˆ™é›†æˆä¸­å¼•å…¥åˆ†ç±»å™¨æ€§èƒ½æƒé‡æ¥æ”¹è¿›é›†æˆ,é™åˆ¶å·®åˆ†ç±»å™¨çš„é”™è¯¯ä¼ æ’­æ•ˆåº”ã€‚
> 
> 3\. åœ¨Yeastå’ŒGenbaseæ•°æ®é›†ä¸Šçš„è¯•éªŒè¡¨æ˜Ž,ä¸Šè¿°ä¸¤ç§æ–¹æ³•åœ¨åŸºå› åŠŸèƒ½é¢„æµ‹ä»»åŠ¡ä¸Šéƒ½èƒ½å–å¾—ä¸é”™çš„æ€§èƒ½ã€‚
> 
> 4\. è¿™äº›æ–¹æ³•ä¸ºè§£å†³åŸºå› æµ‹åºæ•°æ®çš„åŠŸèƒ½æ ‡æ³¨ä¸Žåˆ†ç±»é—®é¢˜æä¾›äº†æœ‰æ•ˆçš„æœºå™¨å­¦ä¹ æ¡†æž¶ã€‚

##### Protein Function Prediction Using DNN

> æœ¬èŠ‚ä¸»è¦å†…å®¹æ˜¯å…³äºŽä½¿ç”¨æ·±åº¦è¿ç§»å­¦ä¹ æ–¹æ³•è¿›è¡Œè›‹ç™½è´¨åŠŸèƒ½é¢„æµ‹çš„ç ”ç©¶ã€‚
> 
> ä¸»è¦æœ‰ä»¥ä¸‹å‡ ä¸ªæ–¹é¢:
> 
> 1\. ä½¿ç”¨å¤šå¤´å¤šå°¾æ·±åº¦å·ç§¯ç¥žç»ç½‘ç»œæ¨¡åž‹è¿›è¡ŒåŸºå› æœ¬ä½“è®º(GO)æ³¨é‡Šé¢„æµ‹ã€‚è¯¥æ¨¡åž‹å¯ä»¥å®žçŽ°çŸ¥è¯†è¿ç§»,æé«˜æ¨¡åž‹æ€§èƒ½ã€‚
> 
> 2\. é€šè¿‡GOæ³¨é‡Šé¢„æµ‹çš„è¿ç§»å­¦ä¹ æ¥å¢žå¼ºäºšç»†èƒžå®šä½é¢„æµ‹ã€‚å…±äº«ç‰¹å¾æå–å™¨,å®žçŽ°é¢„è®­ç»ƒå’Œå¾®è°ƒã€‚
> 
> 3\. åŸºäºŽè¿ç§»å­¦ä¹ çš„è›‹ç™½è´¨ç›¸äº’ä½œç”¨é¢„æµ‹,ç”¨äºŽå¤åˆç‰©æ£€æµ‹ã€‚è¿ç§»æ¨¡åž‹ä»ŽGOå’Œå®šä½æ³¨é‡Šä»»åŠ¡ä¸­å­¦ä¹ çŸ¥è¯†ã€‚
> 
> 4\. åœ¨GOæ³¨é‡Šã€å®šä½å’Œç›¸äº’ä½œç”¨é¢„æµ‹ä»»åŠ¡ä¹‹é—´å®žçŽ°äº†è¿ç§»å­¦ä¹ ã€‚æå‡ºçš„æ¨¡åž‹å¯ä»¥æ‰©å±•åˆ°å…¶ä»–ç”Ÿç‰©ä¿¡æ¯å­¦ä»»åŠ¡ã€‚
> 
> 5\. è¯•éªŒç»“æžœè¡¨æ˜Ž,è¿ç§»å­¦ä¹ å¯ä»¥æœ‰æ•ˆæ”¹è¿›è›‹ç™½è´¨åŠŸèƒ½é¢„æµ‹çš„æ€§èƒ½ã€‚
> 
> æ€»ä½“è€Œè¨€,æœ¬èŠ‚ç ”ç©¶äº†å¦‚ä½•åˆ©ç”¨æ·±åº¦å­¦ä¹ å’Œè¿ç§»å­¦ä¹ çš„æ–¹æ³•,æ”¹è¿›è›‹ç™½è´¨åŠŸèƒ½é¢„æµ‹ä¸Žæ ‡æ³¨çš„æ€§èƒ½ã€‚

#### Protein Structure Prediction

##### Protein Folding Based on EDA

> æœ¬èŠ‚ä¸»è¦å†…å®¹æ˜¯å…³äºŽä½¿ç”¨åŸºäºŽEDAçš„æ··åˆæ™ºèƒ½ç®—æ³•è¿›è¡Œè›‹ç™½è´¨ç»“æž„é¢„æµ‹çš„ç ”ç©¶ã€‚
> 
> ä¸»è¦åŒ…å«ä»¥ä¸‹å‡ ä¸ªæ–¹é¢:
> 
> 1\. ä½¿ç”¨æ··åˆEDAç®—æ³•æ±‚è§£ç®€åŒ–çš„HPæ¨¡åž‹è›‹ç™½è´¨æŠ˜å é—®é¢˜ã€‚æå‡ºå¤åˆé€‚åº”åº¦å‡½æ•°ã€å¼•å¯¼ç®—å­å±€éƒ¨æœç´¢å’Œæ”¹è¿›çš„å›žæº¯ä¿®å¤æ–¹æ³•ã€‚
> 
> 2\. æå‡ºè‡ªé€‚åº”åˆ†ç¾¤EDAç®—æ³•,ä½¿ç”¨äº²å’Œä¼ æ’­èšç±»å®žçŽ°è‡ªé€‚åº”åˆ†ç¾¤,å¹¶ä½¿ç”¨çŽ»å°”å…¹æ›¼æœºåˆ¶å®žçŽ°å¹³è¡¡æœç´¢ã€‚åº”ç”¨äºŽè¿žç»­ä¼˜åŒ–å’ŒHPæ¨¡åž‹è›‹ç™½è´¨ç»“æž„é¢„æµ‹é—®é¢˜ã€‚
> 
> 3\. è¯•éªŒç»“æžœè¡¨æ˜Ž,æ‰€æå‡ºçš„æ··åˆEDAå’Œè‡ªé€‚åº”åˆ†ç¾¤EDAç®—æ³•éƒ½å¯ä»¥æœ‰æ•ˆæé«˜è›‹ç™½è´¨ç»“æž„é¢„æµ‹çš„æ€§èƒ½ã€‚
> 
> 4\. EDAç®—æ³•ä¸Žç‰¹å®šé—®é¢˜ä¿¡æ¯çš„ç»“åˆä¸ºè§£å†³è›‹ç™½è´¨ç»“æž„é¢„æµ‹ç­‰å¤æ‚ä¼˜åŒ–é—®é¢˜æä¾›äº†ä¸€ç§æœ‰æ•ˆé€”å¾„ã€‚
> 
> æ€»çš„æ¥è¯´, å®ƒæŽ¢è®¨äº†EDAç®—æ³•åœ¨è›‹ç™½è´¨ç»“æž„é¢„æµ‹é—®é¢˜ä¸Šçš„åº”ç”¨,ä»¥åŠå¦‚ä½•è®¾è®¡å‡ºæ€§èƒ½æ›´ä¼˜çš„æ··åˆEDAå’Œè‡ªé€‚åº”EDAç®—æ³•ã€‚

##### Protein Folding Based on Deep Reinforcement Learning

> æœ¬èŠ‚æ˜¯å…³äºŽè›‹ç™½è´¨æŠ˜å é—®é¢˜çš„ç ”ç©¶æŠ¥å‘Šã€‚ä¸»è¦å†…å®¹åŒ…æ‹¬:
> 
> 1\. ç®€ä»‹è›‹ç™½è´¨æŠ˜å é—®é¢˜å’ŒHPæ¨¡åž‹  
>  2\. æ·±åº¦å¼ºåŒ–å­¦ä¹ æ–¹æ³•ç”¨äºŽè§£å†³HPæ¨¡åž‹ä¸­çš„è›‹ç™½è´¨æŠ˜å é—®é¢˜  
>  3\. AlphaGoç³»åˆ—ä¸­è’™ç‰¹å¡æ´›æ ‘æœç´¢(MCTS)çš„åº”ç”¨  
>  4\. å°†MCTSåº”ç”¨äºŽHPæ¨¡åž‹ä»¥è§£å†³è›‹ç™½è´¨æŠ˜å é—®é¢˜  
>  5\. è›‹ç™½è´¨ç»“æž„é¢„æµ‹çš„æœ€æ–°è¿›å±•,å¦‚AlphaFold
> 
> æŠ¥å‘Šä»‹ç»äº†å¦‚ä½•é€šè¿‡æ·±åº¦å¼ºåŒ–å­¦ä¹ å’Œè’™ç‰¹å¡æ´›æ ‘æœç´¢ç­‰æ–¹æ³•æ¥é¢„æµ‹è›‹ç™½è´¨åºåˆ—çš„ä¸‰ç»´ç»“æž„,è¿™æ˜¯è›‹ç™½è´¨æŠ˜å é—®é¢˜çš„æ ¸å¿ƒã€‚æ€»çš„æ¥è¯´,è¿™ç¯‡æŠ¥å‘Šèšç„¦åœ¨åˆ©ç”¨AIæŠ€æœ¯æ¥è§£å†³è›‹ç™½è´¨ç»“æž„é¢„æµ‹è¿™ä¸ªåŸºç¡€ç§‘å­¦éš¾é¢˜ä¸Šã€‚

##### AlphaFold2 for Protein Structure Prediction

> æœ¬èŠ‚æ˜¯å…³äºŽAlphaFold2è›‹ç™½è´¨ç»“æž„é¢„æµ‹æ–¹æ³•çš„ä»‹ç»å’Œåˆ†æžæŠ¥å‘Šã€‚ä¸»è¦å†…å®¹åŒ…æ‹¬:
> 
> 1\. è›‹ç™½è´¨æŠ˜å é—®é¢˜çš„èƒŒæ™¯ç®€ä»‹  
>  2\. AlphaFold2çš„æ•´ä½“æµç¨‹æ¦‚è§ˆ,åŒ…æ‹¬æ•°æ®å‡†å¤‡ã€åµŒå…¥ã€Evoformeræ¨¡å—ã€ç»“æž„æ¨¡å—ç­‰  
>  3\. AlphaFold2çš„å…·ä½“æŠ€æœ¯ç»†èŠ‚,å¦‚å¤šåºåˆ—æ¯”å¯¹ã€è¡¨å¾æå–ã€æ³¨æ„åŠ›æœºåˆ¶ã€æŸå¤±å‡½æ•°ç­‰  
>  4\. AlphaFold2é¢„æµ‹è›‹ç™½è´¨ç»“æž„çš„åº”ç”¨,å¦‚è›‹ç™½è´¨ç»“æž„æ•°æ®åº“
> 
> ç®€è€Œè¨€ä¹‹,è¿™æ˜¯ä¸€ç¯‡ä»‹ç»å’Œåˆ†æžAlphaFold2åœ¨è›‹ç™½è´¨ç»“æž„é¢„æµ‹é¢†åŸŸé‡Œç¨‹ç¢‘å¼çš„è¿›å±•çš„æŠ¥å‘Šã€‚å®ƒæ¦‚è¿°äº†è¿™ä¸€æœ€å…ˆè¿›çš„AIç³»ç»Ÿæ˜¯å¦‚ä½•è§£å†³è¿™ä¸€å›°æ‰°ç”Ÿç‰©å­¦å®¶50å¹´ä¹‹ä¹…çš„éš¾é¢˜çš„ã€‚æ€»çš„æ¥è¯´,è¿™ç¯‡æŠ¥å‘Šèšç„¦åœ¨åˆ©ç”¨æ·±åº¦å­¦ä¹ æŠ€æœ¯æ¥æŽ¨åŠ¨åŸºç¡€ç§‘å­¦å‘å±•æ–¹é¢çš„å†…å®¹ã€‚

* * *

### References

  * EMBL-EBI, [A Quick Introduction to Elements of Biology](https://www2.hflab.ips.waseda.ac.jp/~jinglu/Courses/11300ABio/EBI%20Research%20-%20Functional%20Genomics%20-%20Introduction%20To%20Biology.mht "A Quick Introduction to Elements of Biology") \- Cells, Molecules, Genes, Functional Genomics, Microarrays
  * DOE Human Genome Program:  _[Primer on Molecular Genetics](https://www2.hflab.ips.waseda.ac.jp/~jinglu/Courses/11300ABio/Primer%20on%20Molecular%20Genetics%20%28DOE%20Human%20Genome%20Program,%201992%29.pdf "Primer on Molecular Genetics")_ , 1992
  * Jin Xiong, [Essential Bioinformatics](https://www2.hflab.ips.waseda.ac.jp/~jinglu/Courses/11300ABio/Essential%20Bioinformation%20%28Jin%20Xiong,%20Cambridge%20University%20Press,%202006%29.pdf "Essential Bioinformatics"), Cambridge University Press, 2006
  * P. Larranaga et.al., "[Machine Learning in Bioinformatics](https://www2.hflab.ips.waseda.ac.jp/~jinglu/Courses/11300ABio/Machine%20Learning%20in%20Bioinformatics,%20Briefings%20in%20Bioinformatics,%207%281%29,%2086-112.pdf "Machine Learning in Bioinformatics")",  _Briefings in Bioinformatics_ , 7(1), 86-112, 2006
  * J.M. Claverie and C. Notredame,  _[Bioinformatics for Dummies, 2nd Ed.](https://www2.hflab.ips.waseda.ac.jp/~jinglu/Courses/11300ABio/Bioinformatics.For.Dummies.2nd.Ed%20%28for%20Dummies,%202007%29.pdf "Bioinformatics for Dummies, 2nd Ed.")_ , Wiley Publishing Inc, 2007
  * B. Alberts et.al.,  _[Molecular Biology of THE CELL, 5th Ed.](https://www2.hflab.ips.waseda.ac.jp/~jinglu/Courses/11300ABio/Molecular%20Biology%20of%20The%20Cell%205th%20Ed%20%28Garland%20Science,%202008%29.pdf "Molecular Biology of THE CELL, 5th Ed.")_ , Garland Science, 2008
  * N.C. Jones and P.A. Pevzner,  _[An Introduction to Bioinformatics Algorithms](https://www2.hflab.ips.waseda.ac.jp/~jinglu/Courses/11300ABio/An%20Introduction%20to%20Bioinformatics%20Algorithms%20%28Neil%20C.%20Jones%20and%20Pavel%20A.%20Pevzner,%20The%20MIT%20Press,%202004%29.pdf "An Introduction to Bioinformatics Algorithms")_ , The MIT Press, 2004
  * DNAFTB: [DNA from the Beginning](http://dnaftb.org/ "DNA from the Beginning") (An animated primer of 75 experiments that made modern genetics)

  
---
