<!DOCTYPE html><html lang="en" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0,viewport-fit=cover"><title>Summerize for Bioinformatics with ChatGPT | Likun Wang | ç‹ç«‹å¤</title><meta name="author" content="Likun Wang (ç‹ç«‹å¤)"><meta name="copyright" content="Likun Wang (ç‹ç«‹å¤)"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta name="description" content="æœ¬æ–‡è¿ç§»è‡ªCSDNåšå®¢åŸæ–‡é“¾æ¥ï¼šSummerize for Bioinformatics with ChatGPTğŸ“Š 971 é˜…è¯» | ğŸ‘ 22 ç‚¹èµ | ğŸ’¬ 0 è¯„è®º | â­ 13 æ”¶è—   ç›®å½• Basic Introduction Historical Events Bioinformatics vs. Computational Biology Levels of Bioinfo">
<meta property="og:type" content="article">
<meta property="og:title" content="Summerize for Bioinformatics with ChatGPT">
<meta property="og:url" content="https://your-domain.com/2024/01/20/en/Summerize-for-Bioinformatics-with-ChatGPT/index.html">
<meta property="og:site_name" content="Likun Wang | ç‹ç«‹å¤">
<meta property="og:description" content="æœ¬æ–‡è¿ç§»è‡ªCSDNåšå®¢åŸæ–‡é“¾æ¥ï¼šSummerize for Bioinformatics with ChatGPTğŸ“Š 971 é˜…è¯» | ğŸ‘ 22 ç‚¹èµ | ğŸ’¬ 0 è¯„è®º | â­ 13 æ”¶è—   ç›®å½• Basic Introduction Historical Events Bioinformatics vs. Computational Biology Levels of Bioinfo">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://your-domain.com/images/posts/Summerize-for-Bioinformatics-w/517441f24ba6.png">
<meta property="article:published_time" content="2024-01-20T15:00:00.000Z">
<meta property="article:modified_time" content="2024-01-20T15:00:00.000Z">
<meta property="article:author" content="Likun Wang (ç‹ç«‹å¤)">
<meta property="article:tag" content="BLAST">
<meta property="article:tag" content="Bioinformatics">
<meta property="article:tag" content="GA">
<meta property="article:tag" content="PSSMæ–¹æ³•">
<meta property="article:tag" content="å¯å‘å¼ç®—æ³•">
<meta property="article:tag" content="å¤ä¹ èµ„æ–™">
<meta property="article:tag" content="ç”Ÿç‰©ä¿¡æ¯å­¦">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://your-domain.com/images/posts/Summerize-for-Bioinformatics-w/517441f24ba6.png"><script type="application/ld+json">{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "headline": "Summerize for Bioinformatics with ChatGPT",
  "url": "https://your-domain.com/2024/01/20/en/Summerize-for-Bioinformatics-with-ChatGPT/",
  "image": "https://your-domain.com/images/posts/Summerize-for-Bioinformatics-w/517441f24ba6.png",
  "datePublished": "2024-01-20T15:00:00.000Z",
  "dateModified": "2024-01-20T15:00:00.000Z",
  "author": [
    {
      "@type": "Person",
      "name": "Likun Wang (ç‹ç«‹å¤)",
      "url": "https://github.com/veckun"
    }
  ]
}</script><link rel="shortcut icon" href="/img/favicon.png"><link rel="canonical" href="https://your-domain.com/2024/01/20/en/Summerize-for-Bioinformatics-with-ChatGPT/index.html"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><link rel="stylesheet" href="/css/index.css?v=5.5.3"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@7.1.0/css/all.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/node-snackbar@0.1.16/dist/snackbar.min.css" media="print" onload="this.media='all'"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/ui@6.1.7/dist/fancybox/fancybox.min.css" media="print" onload="this.media='all'"><script>
    (() => {
      
    const saveToLocal = {
      set: (key, value, ttl) => {
        if (!ttl) return
        const expiry = Date.now() + ttl * 86400000
        localStorage.setItem(key, JSON.stringify({ value, expiry }))
      },
      get: key => {
        const itemStr = localStorage.getItem(key)
        if (!itemStr) return undefined
        const { value, expiry } = JSON.parse(itemStr)
        if (Date.now() > expiry) {
          localStorage.removeItem(key)
          return undefined
        }
        return value
      }
    }

    window.btf = {
      saveToLocal,
      getScript: (url, attr = {}) => new Promise((resolve, reject) => {
        const script = document.createElement('script')
        script.src = url
        script.async = true
        Object.entries(attr).forEach(([key, val]) => script.setAttribute(key, val))
        script.onload = script.onreadystatechange = () => {
          if (!script.readyState || /loaded|complete/.test(script.readyState)) resolve()
        }
        script.onerror = reject
        document.head.appendChild(script)
      }),
      getCSS: (url, id) => new Promise((resolve, reject) => {
        const link = document.createElement('link')
        link.rel = 'stylesheet'
        link.href = url
        if (id) link.id = id
        link.onload = link.onreadystatechange = () => {
          if (!link.readyState || /loaded|complete/.test(link.readyState)) resolve()
        }
        link.onerror = reject
        document.head.appendChild(link)
      }),
      addGlobalFn: (key, fn, name = false, parent = window) => {
        if (!true && key.startsWith('pjax')) return
        const globalFn = parent.globalFn || {}
        globalFn[key] = globalFn[key] || {}
        globalFn[key][name || Object.keys(globalFn[key]).length] = fn
        parent.globalFn = globalFn
      }
    }
  
      
      const activateDarkMode = () => {
        document.documentElement.setAttribute('data-theme', 'dark')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', 'undefined')
        }
      }
      const activateLightMode = () => {
        document.documentElement.setAttribute('data-theme', 'light')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', 'undefined')
        }
      }

      btf.activateDarkMode = activateDarkMode
      btf.activateLightMode = activateLightMode

      const theme = saveToLocal.get('theme')
    
          const mediaQueryDark = window.matchMedia('(prefers-color-scheme: dark)')
          const mediaQueryLight = window.matchMedia('(prefers-color-scheme: light)')

          if (theme === undefined) {
            if (mediaQueryLight.matches) activateLightMode()
            else if (mediaQueryDark.matches) activateDarkMode()
            else {
              const hour = new Date().getHours()
              const isNight = hour <= 6 || hour >= 18
              isNight ? activateDarkMode() : activateLightMode()
            }
            mediaQueryDark.addEventListener('change', () => {
              if (saveToLocal.get('theme') === undefined) {
                e.matches ? activateDarkMode() : activateLightMode()
              }
            })
          } else {
            theme === 'light' ? activateLightMode() : activateDarkMode()
          }
        
      
      const asideStatus = saveToLocal.get('aside-status')
      if (asideStatus !== undefined) {
        document.documentElement.classList.toggle('hide-aside', asideStatus === 'hide')
      }
    
      
    const detectApple = () => {
      if (/iPad|iPhone|iPod|Macintosh/.test(navigator.userAgent)) {
        document.documentElement.classList.add('apple')
      }
    }
    detectApple()
  
    })()
  </script><script>const GLOBAL_CONFIG = {
  root: '/',
  algolia: undefined,
  localSearch: {"path":"/search.xml","preload":true,"top_n_per_article":1,"unescape":false,"pagination":{"enable":false,"hitsPerPage":8},"languages":{"hits_empty":"No results found for: ${query}","hits_stats":"${hits} articles found"}},
  translate: undefined,
  highlight: {"plugin":"highlight.js","highlightCopy":true,"highlightLang":true,"highlightHeightLimit":400,"highlightFullpage":true,"highlightMacStyle":true},
  copy: {
    success: 'Copy Successful',
    error: 'Copy Failed',
    noSupport: 'Browser Not Supported'
  },
  relativeDate: {
    homepage: false,
    post: false
  },
  runtime: '',
  dateSuffix: {
    just: 'Just now',
    min: 'minutes ago',
    hour: 'hours ago',
    day: 'days ago',
    month: 'months ago'
  },
  copyright: {"limitCount":50,"languages":{"author":"Author: Likun Wang (ç‹ç«‹å¤)","link":"Link: ","source":"Source: Likun Wang | ç‹ç«‹å¤","info":"Copyright belongs to the author. For commercial use, please contact the author for authorization. For non-commercial use, please indicate the source."}},
  lightbox: 'fancybox',
  Snackbar: {"chs_to_cht":"You have switched to Traditional Chinese","cht_to_chs":"You have switched to Simplified Chinese","day_to_night":"You have switched to Dark Mode","night_to_day":"You have switched to Light Mode","bgLight":"#49b1f5","bgDark":"#1f1f1f","position":"bottom-left"},
  infinitegrid: {
    js: 'https://cdn.jsdelivr.net/npm/@egjs/infinitegrid@4.12.0/dist/infinitegrid.min.js',
    buttonText: 'Load More'
  },
  isPhotoFigcaption: false,
  islazyloadPlugin: true,
  isAnchor: true,
  percent: {
    toc: true,
    rightside: true,
  },
  autoDarkmode: true
}</script><script id="config-diff">var GLOBAL_CONFIG_SITE = {
  title: 'Summerize for Bioinformatics with ChatGPT',
  isHighlightShrink: false,
  isToc: true,
  pageType: 'post'
}</script><link rel="stylesheet" href="/css/csdn-stats.css"><link rel="stylesheet" href="/css/lang-switch.css"><meta name="generator" content="Hexo 8.1.1"></head><body><div id="loading-box"><div class="loading-left-bg"></div><div class="loading-right-bg"></div><div class="spinner-box"><div class="configure-border-1"><div class="configure-core"></div></div><div class="configure-border-2"><div class="configure-core"></div></div><div class="loading-word">Loading...</div></div></div><script>(()=>{
  const $loadingBox = document.getElementById('loading-box')
  const $body = document.body
  const preloader = {
    endLoading: () => {
      if ($loadingBox.classList.contains('loaded')) return
      $body.style.overflow = ''
      $loadingBox.classList.add('loaded')
    },
    initLoading: () => {
      $body.style.overflow = 'hidden'
      $loadingBox.classList.remove('loaded')
    }
  }

  preloader.initLoading()

  if (document.readyState === 'complete') {
    preloader.endLoading()
  } else {
    window.addEventListener('load', preloader.endLoading)
    document.addEventListener('DOMContentLoaded', preloader.endLoading)
    // Add timeout protection: force end after 7 seconds
    setTimeout(preloader.endLoading, 7000)
  }

  if (true) {
    btf.addGlobalFn('pjaxSend', preloader.initLoading, 'preloader_init')
    btf.addGlobalFn('pjaxComplete', preloader.endLoading, 'preloader_end')
  }
})()</script><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="avatar-img text-center"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/img/avatar.png" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/></div><div class="site-data text-center"><a href="/archives/"><div class="headline">Articles</div><div class="length-num">302</div></a><a href="/tags/"><div class="headline">Tags</div><div class="length-num">311</div></a><a href="/categories/"><div class="headline">Categories</div><div class="length-num">21</div></a></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> Home</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fas fa-archive"></i><span> Archives</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fas fa-tags"></i><span> Tags</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> Categories</span></a></div><div class="menus_item"><a class="site-page" href="/cv/"><i class="fa-fw fas fa-graduation-cap"></i><span> CV</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-user"></i><span> About</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg fixed" id="page-header" style="background-image: url(/images/posts/Summerize-for-Bioinformatics-w/517441f24ba6.png);"><nav id="nav"><span id="blog-info"><a class="nav-site-title" href="/"><span class="site-name">Likun Wang | ç‹ç«‹å¤</span></a><a class="nav-page-title" href="/"><span class="site-name">Summerize for Bioinformatics with ChatGPT</span><span class="site-name"><i class="fa-solid fa-circle-arrow-left"></i><span>  Back to Home</span></span></a></span><div id="menus"><div id="search-button"><span class="site-page social-icon search"><i class="fas fa-search fa-fw"></i><span> Search</span></span></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> Home</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fas fa-archive"></i><span> Archives</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fas fa-tags"></i><span> Tags</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> Categories</span></a></div><div class="menus_item"><a class="site-page" href="/cv/"><i class="fa-fw fas fa-graduation-cap"></i><span> CV</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-user"></i><span> About</span></a></div></div><div id="toggle-menu"><span class="site-page"><i class="fas fa-bars fa-fw"></i></span></div></div></nav><div id="post-info"><h1 class="post-title">Summerize for Bioinformatics with ChatGPT</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">Created</span><time class="post-meta-date-created" datetime="2024-01-20T15:00:00.000Z" title="Created 2024-01-21 00:00:00">2024-01-21</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">Updated</span><time class="post-meta-date-updated" datetime="2024-01-20T15:00:00.000Z" title="Updated 2024-01-21 00:00:00">2024-01-21</time></span><span class="post-meta-categories"><span class="post-meta-separator">|</span><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/%E5%A4%8D%E4%B9%A0%E7%AC%94%E8%AE%B0/">å¤ä¹ ç¬”è®°</a></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span class="post-meta-wordcount"><i class="far fa-file-word fa-fw post-meta-icon"></i><span class="post-meta-label">Word Count:</span><span class="word-count">10.7k</span><span class="post-meta-separator">|</span><i class="far fa-clock fa-fw post-meta-icon"></i><span class="post-meta-label">Reading Time:</span><span>49mins</span></span><span class="post-meta-separator">|</span><span class="post-meta-pv-cv" id="" data-flag-title=""><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">Post Views:</span><span id="busuanzi_value_page_pv"><i class="fa-solid fa-spinner fa-spin"></i></span></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="container post-content" id="article-container"><div id="post-outdate-notice" data="{&quot;limitDay&quot;:180,&quot;messagePrev&quot;:&quot;This article was last updated&quot;,&quot;messageNext&quot;:&quot;days ago. The content may be outdated.&quot;,&quot;postUpdate&quot;:&quot;2024-01-21 00:00:00&quot;}" hidden></div><blockquote>
<p>æœ¬æ–‡è¿ç§»è‡ªCSDNåšå®¢<br>åŸæ–‡é“¾æ¥ï¼š<a target="_blank" rel="noopener" href="https://blog.csdn.net/m0_59180666/article/details/135716837">Summerize for Bioinformatics with ChatGPT</a><br>ğŸ“Š 971 é˜…è¯» | ğŸ‘ 22 ç‚¹èµ | ğŸ’¬ 0 è¯„è®º | â­ 13 æ”¶è—</p>
</blockquote>
<hr>
<p><strong>ç›®å½•</strong></p>
<p>Basic Introduction</p>
<p>Historical Events</p>
<p>Bioinformatics vs. Computational Biology</p>
<p>Levels of Bioinfo &amp; CompBio</p>
<p>Molecular Biology Primer</p>
<p>Homework 1</p>
<p>A brief Introduction to DNA &amp; RNA</p>
<p>A brief Introduction to Protein</p>
<p>Sequencing Technologies</p>
<p>Sequence Alignment</p>
<p>Pairwise sequence alignment</p>
<p>Basic notions</p>
<p>Goal of Pairwise Alignment:</p>
<p>Algorithms for G&amp;L Alignments</p>
<p>Homework 2</p>
<p>Database similarity searches</p>
<p>BLAST: Basic Local Alignment Search Tool</p>
<p>FASTA: FAST-All </p>
<p>Comparison between BLAST &amp; FASTA</p>
<p>Homework 3</p>
<p>Multiple sequence alignment</p>
<p>1. Progressive alignment methods</p>
<p>2. Iterative alignment methods</p>
<p>PSSMæ–¹æ³•</p>
<p>Motifs and domains</p>
<p>Homework 4</p>
<p>My understanding to Expectation Maximization(EM) Algorithm:</p>
<p>Core Algorithm:</p>
<p>Code in PyCharm: (python&#x3D;3.8)</p>
<p>The final result: </p>
<p>The difficulties I met:</p>
<p>Machine Learning Approaches</p>
<p>GA for sequencing and MSA</p>
<p>Sequencing by Using GA</p>
<p>Multiple Sequence Alignments Using GA</p>
<p>Protein Function Prediction</p>
<p>Gene Function Prediction Using SVM</p>
<p>Protein Function Prediction Using DNN</p>
<p>Protein Structure Prediction</p>
<p>Protein Folding Based on EDA</p>
<p>Protein Folding Based on Deep Reinforcement Learning</p>
<p>AlphaFold2 for Protein Structure Prediction</p>
<p>References</p>
<hr>
<h3 id="Basic-Introduction"><a href="#Basic-Introduction" class="headerlink" title="Basic Introduction"></a>Basic Introduction</h3><h4 id="Historical-Events"><a href="#Historical-Events" class="headerlink" title="Historical Events"></a>Historical Events</h4><h5 id="Bioinformatics-vs-Computational-Biology"><a href="#Bioinformatics-vs-Computational-Biology" class="headerlink" title="Bioinformatics vs. Computational Biology"></a>Bioinformatics vs. Computational Biology</h5><ul>
<li>Bioinformatics &#x3D; the creation of tools (algorithms, databases) that solve problems. The goal is to build useful tools that work on biological data. lt is about engineering.</li>
<li>Computational biology &#x3D; the study of biology using computational techniques. The goal is to learn new biology, knowledge about living systems. lt is about discovery.</li>
</ul>
<h5 id="Levels-of-Bioinfo-CompBio"><a href="#Levels-of-Bioinfo-CompBio" class="headerlink" title="Levels of Bioinfo &amp; CompBio"></a>Levels of Bioinfo &amp; CompBio</h5><ul>
<li>Level 0: <strong>Modeling for modeling</strong> â€˜s sake</li>
<li>Level 1: (entry) Use published tools to <strong>analyze data</strong> and <strong>generate hypotheses</strong> for experimentalists</li>
<li>Level 2 (Bioinfo): <strong>Develop algorithms and databases</strong> for data analyses on new technologies, data integration and reuse.</li>
<li>Level 3 (CompBio): Make <strong>biological discoveries</strong> from public dataintegration and modeling</li>
<li>Level X: Integrative studies from big consortia</li>
</ul>
<hr>
<h4 id="Molecular-Biology-Primer"><a href="#Molecular-Biology-Primer" class="headerlink" title="Molecular Biology Primer"></a>Molecular Biology Primer</h4><blockquote>
<p><strong>All life depends on 3 critical molecules</strong></p>
<ul>
<li><p>DNAs </p>
<ul>
<li>Hold information on how cell works</li>
</ul>
</li>
<li><p>RNAs </p>
<ul>
<li><p>Act to transfer short pieces of information to different parts  of cell </p>
</li>
<li><p>Provide templates to synthesize into protein</p>
</li>
</ul>
</li>
<li><p>Proteins </p>
<ul>
<li><p>Form <strong>enzymes(é…¶)</strong> that send signals to other cells and regulate  gene activity </p>
</li>
<li><p>Form bodyâ€™s major components (e.g. hair, skin, etc.)</p>
</li>
<li><p>è›‹ç™½è´¨åœ¨ç”Ÿå‘½ç³»ç»Ÿä¸­çš„é‡è¦ä½œç”¨ï¼š</p>
</li>
</ul>
</li>
</ul>
<p>â€¢ Catalysiså‚¬åŒ–ä½œç”¨</p>
<p>â€¢ Carrier and transportæ‰¿è¿äººå’Œè¿è¾“</p>
<p>â€¢ Regulatoryè°ƒèŠ‚</p>
</blockquote>
<blockquote>
<h5 id="Homework-1"><a href="#Homework-1" class="headerlink" title="Homework 1"></a><strong>Homework 1</strong></h5><p>Write a report describing <em>Cell</em> , <em>DNA</em> , <em>RNA</em> , Proteins and discussing their relations, from information point of view. </p>
<ul>
<li><strong>Replication: å¤åˆ¶</strong></li>
<li><strong>Transcription: è½¬å½•</strong></li>
<li><strong>Translation: ç¿»è¯‘</strong></li>
</ul>
</blockquote>
<blockquote>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/517441f24ba6.png"></p>
</blockquote>
<h5 id="A-brief-Introduction-to-DNA-RNA"><a href="#A-brief-Introduction-to-DNA-RNA" class="headerlink" title="A brief Introduction to DNA &amp; RNA"></a>A brief Introduction to DNA &amp; RNA</h5><ul>
<li>DNA &amp; RNA are <strong>polymers(èšåˆç‰©)</strong> of<strong>nucleotides(æ ¸è‹·é…¸)</strong><ul>
<li>DNA is a nucleic acid, made of long chains of nucleotides, which has four kinds of bases, <strong>A, T, C, G</strong>. DNA is a <strong>double strand helix(åŒé“¾èºæ—‹)</strong> , <strong>hydrogen bonds(æ°¢é”®)</strong> between <strong>bases(ç¢±åŸº)</strong> hold the strands together: A and T, C and G. <ul>
<li><p>Untwisting and replication (using DNA <strong>polymerase(èšåˆé…¶)</strong>): each strand is a template of a new strand</p>
</li>
<li><p>DNA replication begins at many specific sites</p>
</li>
<li><p>Each strand of the double helix is oriented in the opposite direction</p>
</li>
<li><p>The sequence of codons in DNA spells out the primary structure of a polypeptide</p>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<p>(Polypeptides form proteins that cells and organisms use)</p>
<pre><code>* RNA is also a nucleic acid: different sugar (has 1 more O on the 2nd Carbon) ;  U instead of T; Single strand, usually
</code></pre>
<h5 id="A-brief-Introduction-to-Protein"><a href="#A-brief-Introduction-to-Protein" class="headerlink" title="A brief Introduction to Protein"></a>A brief Introduction to Protein</h5><ul>
<li>Proteins are key players in our living systems. </li>
<li>Proteins are polymers consisting of <strong>20 kinds of amino****acids. (æ²¡æœ‰ BJOUXZè¿™å…­ä¸ªå­—æ¯)</strong></li>
<li>Each protein folds into a unique three-dimensional structure defined by its amino acid sequence. </li>
<li>Protein structure has a <strong>hierarchical nature(å±‚çº§æ€§).</strong></li>
<li>Protein structure is closely related to its function. </li>
<li>Protein structure prediction is a grand challenge of computational biology</li>
</ul>
<hr>
<h4 id="Sequencing-Technologies"><a href="#Sequencing-Technologies" class="headerlink" title="Sequencing Technologies"></a>Sequencing Technologies</h4><blockquote>
<ul>
<li>1st gen: <strong>Sanger sequencing (æ¡‘æ ¼æµ‹åº)</strong></li>
</ul>
</blockquote>
<blockquote>
<ol>
<li><p><strong>DNAç‰‡æ®µåˆ¶å¤‡ï¼š</strong> å¤åˆ¶è¦æµ‹åºçš„DNAç‰‡æ®µã€‚</p>
</li>
<li><p><strong>ååº”æ··åˆç‰©ï¼š</strong> åˆ›å»ºä¸€ä¸ªååº”æ··åˆç‰©ï¼Œå…¶ä¸­åŒ…æ‹¬æ ‡å‡†æ ¸è‹·é…¸ï¼ˆAã€Tã€Cã€Gï¼‰å’Œä¸€å°éƒ¨åˆ†åŒè„±æ°§æ ¸è‹·é…¸ï¼ˆddATPã€ddTTPã€ddCTPã€ddGTPï¼‰ã€‚</p>
</li>
<li><p><strong>DNAåˆæˆï¼š</strong> è¿›è¡ŒDNAåˆæˆååº”ï¼Œå½“DNAé“¾ç”Ÿé•¿åˆ°å¸¦æœ‰<strong>dideoxynucleotideï¼ˆåŒè„±æ°§æ ¸è‹·é…¸ï¼‰</strong> çš„ä½ç½®æ—¶ï¼ŒDNAé“¾ç»ˆæ­¢ã€‚</p>
</li>
<li><p><strong>åˆ†ç¦»DNAç‰‡æ®µï¼š</strong> é€šè¿‡å‡èƒ¶ç”µæ³³åˆ†ç¦»ä¸åŒé•¿åº¦çš„DNAç‰‡æ®µã€‚</p>
</li>
<li><p><strong>ç”µæ³³åˆ†æï¼š</strong> åˆ†ç¦»çš„DNAç‰‡æ®µå½¢æˆå¸¦çŠ¶å›¾è°±ï¼Œæ ¹æ®å…¶é•¿åº¦å¯è§†åŒ–ã€‚</p>
</li>
<li><p><strong>ç»“æœè§£è¯»ï¼š</strong> é€šè¿‡åˆ†æå›¾è°±ä¸­çš„é¢œè‰²æˆ–ä¿¡å·ï¼Œç¡®å®šDNAåºåˆ—ã€‚</p>
</li>
</ol>
</blockquote>
<blockquote>
<ul>
<li>2nd gen: <strong>Illumina sequencing (ä¼Šå¢ç±³çº³æµ‹åº)</strong></li>
</ul>
</blockquote>
<blockquote>
<ol>
<li><p><strong>æ–‡åº“åˆ¶å¤‡ï¼š</strong> å‰ªåˆ‡DNAæˆçŸ­ç‰‡æ®µï¼Œæ·»åŠ é€‚é…å™¨å½¢æˆæ–‡åº“ã€‚</p>
</li>
<li><p><strong>æ–‡åº“æ‰©å¢ï¼š</strong> æ‰©å¢æˆæ•°ç™¾ä¸‡å¤åˆ¶ã€‚</p>
</li>
<li><p><strong>èšç±»ï¼š</strong> å½¢æˆDNAç‰‡æ®µç°‡ã€‚</p>
</li>
<li><p><strong>æ¡¥PCRï¼š</strong> å°†DNAç‰‡æ®µæ¡¥æ¥åˆ°é€‚é…å™¨ä¸Šã€‚</p>
</li>
<li><p><strong>æµ‹åºï¼š</strong> é€ä¸€æ·»åŠ æ ¸è‹·é…¸ï¼Œè§‚å¯Ÿè§å…‰ä¿¡å·ï¼Œç¡®å®šDNAåºåˆ—ã€‚</p>
</li>
<li><p><strong>æ•°æ®åˆ†æï¼š</strong> å¤„ç†å›¾åƒå’Œä¿¡å·ï¼Œå¾—åˆ°åŸå§‹æµ‹åºæ•°æ®ã€‚</p>
</li>
<li><p><strong>åºåˆ—æ‹¼æ¥å’Œåˆ†æï¼š</strong> å°†çŸ­è¯»æ®µæ‹¼æ¥æˆå®Œæ•´åºåˆ—ï¼Œè¿›è¡Œè¿›ä¸€æ­¥çš„ç”Ÿç‰©ä¿¡æ¯å­¦åˆ†æã€‚</p>
</li>
</ol>
</blockquote>
<blockquote>
<ul>
<li>3rd gen: <strong>Third generation sequencing (ç¬¬ä¸‰ä»£æµ‹åº)</strong></li>
</ul>
</blockquote>
<blockquote>
<ol>
<li><p><strong>DNAå‡†å¤‡ï¼š</strong> æå–DNAå¹¶è¿›è¡Œè´¨æ§ï¼Œç¡®ä¿é€‚åˆç›´æ¥æµ‹åºã€‚</p>
</li>
<li><p><strong>DNAæ¨¡æ¿åˆ¶å¤‡ï¼š</strong> å°†DNAç‰‡æ®µå›ºå®šåœ¨SMRTç»†èƒä¸­ï¼Œå½¢æˆå•åˆ†å­çš„DNAæ¨¡æ¿ã€‚</p>
</li>
<li><p><strong>SMRTç»†èƒæµ‹åºï¼š</strong> ä½¿ç”¨èšåˆé…¶åœ¨DNAæ¨¡æ¿ä¸Šåˆæˆäº’è¡¥é“¾ï¼ŒåŒæ—¶æµ‹é‡é€ä¸ªæ ¸è‹·é…¸çš„åŠ å…¥ï¼Œå½¢æˆå®æ—¶çš„è§å…‰ä¿¡å·ã€‚</p>
</li>
<li><p><strong>æ•°æ®é‡‡é›†ï¼š</strong> é€šè¿‡ç›‘æµ‹SMRTç»†èƒä¸­çš„DNAåˆæˆè¿‡ç¨‹ï¼Œå®æ—¶è·å–é•¿è¯»é•¿çš„æµ‹åºæ•°æ®ã€‚</p>
</li>
<li><p><strong>æ•°æ®åˆ†æï¼š</strong> å¯¹å¾—åˆ°çš„åŸå§‹æ•°æ®è¿›è¡Œå¤„ç†å’Œåˆ†æï¼ŒåŒ…æ‹¬é”™è¯¯æ ¡æ­£å’Œåºåˆ—æ‹¼æ¥ã€‚</p>
</li>
</ol>
</blockquote>
<hr>
<h3 id="Sequence-Alignment"><a href="#Sequence-Alignment" class="headerlink" title="Sequence Alignment"></a>Sequence Alignment</h3><h4 id="Pairwise-sequence-alignment"><a href="#Pairwise-sequence-alignment" class="headerlink" title="Pairwise sequence alignment"></a>Pairwise sequence alignment</h4><h5 id="Basic-notions"><a href="#Basic-notions" class="headerlink" title="Basic notions"></a>Basic notions</h5><p>To express the relatedness:</p>
<ol>
<li><strong>identity ç›¸åŒåº¦</strong></li>
<li><strong>similarity ç›¸ä¼¼åº¦ ï¼ˆç›¸ä¼¼ç‰©åŒ–æ€§è´¨ï¼Œå¯ä»¥è¢«æ›¿æ¢çš„ç¢±åŸºï¼‰</strong><ol>
<li>Calculation 1: <img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="https://latex.csdn.net/eq?%5Cfrac%7B2*L_%7Bs%7D%7D%7BL_%7Ba%7D&plus;L_%7Bb%7D%7D" alt="\\frac{2*L_{s}}{L_{a}+L_{b}}"> (Ls: ç›¸åŒ&#x2F;ç›¸ä¼¼æ®‹åŸºå¯¹æ•°ï¼›La&#x2F;Lb: ä¸¤æ¡åºåˆ—çš„é•¿åº¦)</li>
<li>Calculation 2: <img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="https://latex.csdn.net/eq?%5Cfrac%7BL_%7Bs%7D%7D%7BL_%7Ba%7D%7D" alt="\\frac{L_{s}}{L_{a}}"> (Ls: ç›¸åŒ&#x2F;ç›¸ä¼¼æ®‹åŸºå¯¹æ•°ï¼›La: è¾ƒçŸ­åºåˆ—çš„é•¿åº¦)</li>
</ol>
</li>
<li><strong>homology åŒæºæ€§ ï¼ˆåºåˆ—é—´å­˜åœ¨è¿›åŒ–å…³ç³»çš„æ¨æ–­ï¼‰</strong><ol>
<li><strong>Cautionï¼š åŒæºæ€§è¦ä¹ˆæœ‰è¦ä¹ˆæ²¡æœ‰ï¼Œä¸èƒ½â€œæœ‰ç‚¹â€åŒæºæ€§ï¼</strong></li>
<li><strong>Judgementï¼š ç›¸åŒåº¦å’Œç›¸ä¼¼åº¦çš„ç™¾åˆ†æ¯”</strong>ï¼ˆåºåˆ—è¾ƒçŸ­æ—¶ä¼šå‡ºç°å¶ç„¶ï¼‰ <ol>
<li>Solveï¼š<img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/08d0d5ec95a7.png"></li>
<li><strong>ï¼ˆâ†‘åœ¨ Safe Zoneé‡Œå°±å¯ä»¥å®‰å…¨åœ°è®¤ä¸ºå®ƒæ˜¯åŒæºçš„ï¼‰</strong></li>
</ol>
</li>
</ol>
</li>
</ol>
<p>Amino acids varies from â€œRâ€ groups.</p>
<ul>
<li>Amino acids are joined by <strong>amide linkages (é…°èƒºé”®)</strong> to form proteins.</li>
<li>Each C-N bond is called a <strong>peptide bond (è‚½é”®)</strong>.</li>
</ul>
<blockquote>
<ul>
<li><p>A short chain of amino acids is called a <strong>peptide (å¤šè‚½)</strong>(usually &lt;100). </p>
</li>
<li><p>A longer chain of amino acids is called a <strong>protein</strong>. </p>
</li>
<li><p>When an amino acid is part of a peptide or protein, it is called an amino acid <strong>residue (æ®‹åŸº)</strong>.</p>
</li>
<li><p><strong>amino terminus</strong> &#x2F; <strong>N-terminus (æ°¨åŸº)</strong> , has a free amino group.</p>
<ul>
<li>Residues are numbered at the beginning of N terminus.</li>
</ul>
</li>
<li><p><strong>carboxyl terminus</strong> &#x2F; <strong>C-terminus (ç¾§åŸº)</strong> , has a free carboxyl group.</p>
</li>
</ul>
</blockquote>
<h5 id="Goal-of-Pairwise-Alignment"><a href="#Goal-of-Pairwise-Alignment" class="headerlink" title="Goal of Pairwise Alignment:"></a><strong>Goal of Pairwise Alignment:</strong></h5><p>Find best pairing of two sequences so there is maximum correspondence between residues.</p>
<p><strong>Methods:</strong></p>
<ul>
<li>Global Alignment: æ‰¾åˆ°æœ€é€‚åˆäºä¸¤ä¸ªé•¿åº¦å¤§è‡´ç›¸åŒçš„é«˜åº¦ç›¸ä¼¼çš„åºåˆ—</li>
<li>Local Alignment: æ‰¾åˆ°ä¸¤ä¸ªåºåˆ—ç›¸ä¼¼æ€§æœ€é«˜çš„åŒºåŸŸï¼Œè¿™ä¸¤ä¸ªåºåˆ—çš„é•¿åº¦å¯ä»¥éå¸¸ä¸åŒ</li>
</ul>
<h5 id="Algorithms-for-G-L-Alignments"><a href="#Algorithms-for-G-L-Alignments" class="headerlink" title="Algorithms for G&amp;L Alignments"></a>Algorithms for G&amp;L Alignments</h5><blockquote>
<ul>
<li><p>1. <strong>Dot matrix</strong> (dot plot) <strong>method</strong> : ä¸€ç§ä½¿ç”¨äºŒç»´çŸ©é˜µçš„å›¾å½¢åŒ–æ–¹æ³•</p>
<ul>
<li><p>â€‹â€‹â€‹â€‹ <strong>æ„å»ºçŸ©é˜µï¼š</strong> å°†ä¸¤ä¸ªåºåˆ—åˆ†åˆ«æ”¾ç½®åœ¨çŸ©é˜µçš„æ°´å¹³å’Œå‚ç›´æ–¹å‘ä¸Šï¼Œå½¢æˆä¸€ä¸ªäºŒç»´çŸ©é˜µã€‚ </p>
</li>
<li><p><strong>ç›¸ä¼¼æ€§æ ‡è®°ï¼š</strong> åœ¨ç›¸åŒä½ç½®ä¸Šï¼Œå¦‚æœä¸¤ä¸ªåºåˆ—çš„å­—ç¬¦ç›¸åŒæˆ–ç›¸ä¼¼ï¼Œå°±åœ¨è¯¥ä½ç½®ä¸Šç»˜åˆ¶ä¸€ä¸ªç‚¹ï¼ˆæˆ–å…¶ä»–ç¬¦å·ï¼‰ã€‚</p>
</li>
<li><p><strong>å¯è§†åŒ–ï¼š</strong> æŸ¥çœ‹ç»˜åˆ¶çš„ç‚¹é˜µï¼Œè§‚å¯Ÿåºåˆ—ä¹‹é—´çš„ç›¸ä¼¼æ€§æ¨¡å¼ã€‚</p>
</li>
</ul>
</li>
<li><p>2.<strong>â­</strong> <strong>Dynamic programming</strong> (DP) <strong>method</strong> : ç”Ÿæˆä¸€ä¸ªåŒ…å«æ‰€æœ‰å¯èƒ½å¯¹é½çš„åˆ†æ•°çš„äºŒç»´å¯¹é½çŸ©é˜µï¼Œä»¥ä¾¿å®šé‡åœ°æ‰¾åˆ°æœ€ä½³å¯¹é½</p>
<ul>
<li><p>éœ€è¦ä¸€ä¸ªè¯„åˆ†æ–¹æ³•ï¼ˆScoring Systemï¼‰ï¼š</p>
<ul>
<li><p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="https://latex.csdn.net/eq?score(A)%20=%20%5Calpha%20w-%20%5Cbeta%20x%20-%20%5Cgamma%20y" alt="score\(A\) &#x3D; \\alpha w- \\beta x - \\gamma y"></p>
</li>
<li><p>w: é…å¯¹æ•°ï¼›x: æœªé…å¯¹æ•°ï¼›y: é—´éš™ï¼ˆgapï¼‰æ•° </p>
<ul>
<li>Reward for matches:  Î± (positive score)</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<p>Mismatch penalty: -Î² (negative score) </p>
<p>Gap penalty: -Î³ (negative score) </p>
<pre><code>* **Global alignment: Needleman-Wunsch (NW) algorithm**
  * **å®šä¹‰çŠ¶æ€ï¼š** å°†æ¯”å¯¹é—®é¢˜åˆ’åˆ†ä¸ºä¸€ä¸ªäºŒç»´çŸ©é˜µï¼ŒçŸ©é˜µçš„æ¯ä¸ªå…ƒç´ å¯¹åº”äºä¸¤ä¸ªåºåˆ—çš„æŸä¸€å¯¹å­—ç¬¦ã€‚çŸ©é˜µä¸­çš„æ¯ä¸ªå…ƒç´ è¡¨ç¤ºåœ¨è¯¥ä½ç½®ä¸Šçš„æ¯”å¯¹åˆ†æ•°ï¼Œå³å¯¹åº”å­—ç¬¦çš„ç›¸ä¼¼æ€§å¾—åˆ†ã€‚
  * **åˆå§‹åŒ–ï¼š** åœ¨çŸ©é˜µçš„ç¬¬ä¸€è¡Œå’Œç¬¬ä¸€åˆ—ä¸­è®¾ç½®åˆå§‹å€¼ï¼Œè¡¨ç¤ºç¬¬ä¸€ä¸ªåºåˆ—æˆ–ç¬¬äºŒä¸ªåºåˆ—ä¸ç©ºåºåˆ—ä¹‹é—´çš„æ¯”å¯¹åˆ†æ•°ã€‚è¿™äº›åˆå§‹å€¼æ˜¯æ ¹æ®æ‰€é€‰æ‹©çš„æ¯”å¯¹å¾—åˆ†è§„åˆ™ç¡®å®šçš„ã€‚**C(i,0)=-i*Î³ C(0,j)=-j*Î³**

  * **çŠ¶æ€è½¬ç§»æ–¹ç¨‹ï¼š** ç¡®å®šçŠ¶æ€ä¹‹é—´çš„è½¬ç§»å…³ç³»ï¼Œå³å¦‚ä½•ä»å…ˆå‰çš„çŠ¶æ€å¾—åˆ°å½“å‰çŠ¶æ€ã€‚åœ¨NWç®—æ³•ä¸­ï¼ŒçŠ¶æ€è½¬ç§»æ–¹ç¨‹é€šå¸¸åŒ…æ‹¬åŒ¹é…ã€æ’å…¥å’Œåˆ é™¤æ“ä½œï¼Œå¯¹åº”äºåœ¨ä¸¤ä¸ªåºåˆ—ä¸­ç§»åŠ¨ã€åŒ¹é…æˆ–æ’å…¥ç¼ºå¤±å­—ç¬¦ã€‚

    * **w(a,b)=+Î±(if a=b)/-Î²(if aâ‰ b)**

    * For 1â‰¤iâ‰¤n,1â‰¤jâ‰¤m: 
      * **C(i,j)=max{ [****C(i-1, j-1)+w(Si,Tj)], [C(i-1, j)-Î³], [C(i, j-1)-Î³]}**
      * Si/Tj: Prefix of length i of S/T sequence
    * ![](/images/posts/Summerize-for-Bioinformatics-w/2c8a42e8a541.png)
  * **è®¡ç®—çŸ©é˜µï¼š** æ ¹æ®çŠ¶æ€è½¬ç§»æ–¹ç¨‹ï¼Œè®¡ç®—æ•´ä¸ªçŸ©é˜µçš„å€¼ã€‚è¿™å¯ä»¥é€šè¿‡è‡ªåº•å‘ä¸Šçš„æ–¹å¼ï¼Œä»å·¦ä¸Šè§’é€æ­¥è®¡ç®—åˆ°å³ä¸‹è§’ï¼Œä¹Ÿå°±æ˜¯ä»åºåˆ—çš„å¼€å¤´åˆ°æœ«å°¾ã€‚

  * ![](/images/posts/Summerize-for-Bioinformatics-w/d8c97735c058.png)

  * **å›æº¯ï¼š** æ ¹æ®è®¡ç®—å¾—åˆ°çš„çŸ©é˜µï¼Œä»å³ä¸‹è§’å¼€å§‹å›æº¯ï¼Œæ‰¾åˆ°æœ€ä¼˜æ¯”å¯¹è·¯å¾„ã€‚è¿™ä¸ªè·¯å¾„å°±æ˜¯ä½¿ä¸¤ä¸ªåºåˆ—å¾—åˆ†æœ€é«˜çš„æ¯”å¯¹æ–¹å¼ã€‚æ—¶é—´å¤æ‚åº¦O(mn)

* **Local alignment: Smith-Waterman (SW) algorithm**
  * **â€‹â€‹â€‹â€‹â€‹â€‹â€‹****å®šä¹‰çŠ¶æ€ï¼š** å°†æ¯”å¯¹é—®é¢˜åˆ’åˆ†ä¸ºä¸€ä¸ªäºŒç»´çŸ©é˜µï¼ŒçŸ©é˜µçš„æ¯ä¸ªå…ƒç´ å¯¹åº”äºä¸¤ä¸ªåºåˆ—çš„æŸä¸€å¯¹å­—ç¬¦ã€‚çŸ©é˜µä¸­çš„æ¯ä¸ªå…ƒç´ è¡¨ç¤ºåœ¨è¯¥ä½ç½®ä¸Šçš„æ¯”å¯¹åˆ†æ•°ï¼Œå³å¯¹åº”å­—ç¬¦çš„ç›¸ä¼¼æ€§å¾—åˆ†ã€‚(è›‹ç™½è´¨å±€éƒ¨ç›¸ä¼¼å¯èƒ½å¯¹åº”ç€â€œåŠŸèƒ½äºšåŸºâ€ç›¸ä¼¼ï¼Œè¿™æ˜¯ä¸»è¦çš„)
  * **åˆå§‹åŒ–ï¼š** **åœ¨çŸ©é˜µçš„ç¬¬ä¸€è¡Œå’Œç¬¬ä¸€åˆ—ä¸­è®¾ç½®åˆå§‹å€¼ä¸ºé›¶** ï¼Œè¡¨ç¤ºèµ·å§‹ä½ç½®ã€‚ä¸Needleman-Wunschç®—æ³•ä¸åŒï¼ŒSmith-Watermanç®—æ³•çš„åˆå§‹åŒ–å€¼ä¸ºé›¶ï¼Œä¸è€ƒè™‘ä¸ç©ºåºåˆ—çš„æ¯”å¯¹ã€‚

  * **çŠ¶æ€è½¬ç§»æ–¹ç¨‹ï¼š** ç¡®å®šçŠ¶æ€ä¹‹é—´çš„è½¬ç§»å…³ç³»ï¼Œå³å¦‚ä½•ä»å…ˆå‰çš„çŠ¶æ€å¾—åˆ°å½“å‰çŠ¶æ€ã€‚åœ¨SWç®—æ³•ä¸­ï¼ŒçŠ¶æ€è½¬ç§»æ–¹ç¨‹åŒ…æ‹¬åŒ¹é…ã€æ’å…¥ã€åˆ é™¤æ“ä½œï¼Œå¯¹åº”äºåœ¨ä¸¤ä¸ªåºåˆ—ä¸­ç§»åŠ¨ã€åŒ¹é…æˆ–æ’å…¥ç¼ºå¤±å­—ç¬¦ã€‚

    * **C(i,j)=max{ [****C(i-1, j-1)+score(Si,Tj)], [C(i-1, j)-Î³], [C(i, j-1)-Î³], 0 }**

  * **è®¡ç®—çŸ©é˜µï¼š** æ ¹æ®çŠ¶æ€è½¬ç§»æ–¹ç¨‹ï¼Œè®¡ç®—æ•´ä¸ªçŸ©é˜µçš„å€¼ã€‚è¿™å¯ä»¥é€šè¿‡è‡ªåº•å‘ä¸Šçš„æ–¹å¼ï¼Œä»å·¦ä¸Šè§’é€æ­¥è®¡ç®—åˆ°å³ä¸‹è§’ï¼Œä¹Ÿå°±æ˜¯ä»åºåˆ—çš„å¼€å¤´åˆ°æœ«å°¾ã€‚

  * **å›æº¯ï¼š** æ ¹æ®è®¡ç®—å¾—åˆ°çš„çŸ©é˜µï¼Œæ‰¾åˆ°çŸ©é˜µä¸­çš„æœ€é«˜å¾—åˆ†åŠå…¶ä½ç½®ã€‚ä»è¯¥ä½ç½®å¼€å§‹å›æº¯ï¼Œæ²¿ç€å¾—åˆ†æœ€é«˜çš„è·¯å¾„ï¼Œæ‰¾åˆ°æœ€ä¼˜çš„å±€éƒ¨æ¯”å¯¹ã€‚æ—¶é—´å¤æ‚åº¦O(mn)

* **Gap Panalty Function (é—´éš™æƒ©ç½šå‡½æ•°)**

  * **â€‹â€‹â€‹â€‹â€‹â€‹â€‹ Gap penalty = _h + gk_**
</code></pre>
<p>where  <em>k &#x3D;</em> length of gap;  <em>h &#x3D;</em> gap opening penalty;  <em>g &#x3D;</em> gap continuation penalty ï¼ˆgï¼œh, å› ä¸ºéšç€ç¼ºå£å¼€æ”¾ä¼šæœ‰å‡ ä¸ªæ®‹åŸºä¼´éšç€æ’å…¥æˆ–åˆ é™¤ï¼Œæ‰€ä»¥åº”è¯¥å‡å°‘é¢å¤–æ®‹åŸºçš„æƒ©ç½šï¼‰</p>
<pre><code>  * **æƒ©ç½šåˆ†æ•°è¦åˆç†** ï¼šå¤ªé«˜ä¼šå¯¼è‡´æ²¡æœ‰ç©ºéš™ä»è€Œæ²¡æœ‰åˆç†çš„å¯¹é½ï¼›å¤ªä½ä¼šå¯¼è‡´é—´éš™å¤ªå¤šè®©ä¸ç›¸å…³çš„åºåˆ—ä»¥é«˜åˆ†å¯¹é½ã€‚ 

* **Substitution Matrix or Scoring Matrix ï¼ˆæ›¿ä»£çŸ©é˜µ/è¯„åˆ†çŸ©é˜µï¼‰**

  * **â€‹â€‹â€‹â€‹â€‹â€‹â€‹****PAMçŸ©é˜µï¼ˆPoint Accepted Mutationï¼‰ï¼š** ç”¨äºæè¿°è›‹ç™½è´¨çš„æ°¨åŸºé…¸æ›¿ä»£æ¦‚ç‡ï¼ˆå…¨å±€æ¯”å¯¹ï¼‰ã€‚

  * **BLOSUMçŸ©é˜µï¼ˆBLOcks SUbstitution Matrixï¼‰ï¼š** ç”¨äºæ¯”å¯¹è›‹ç™½è´¨åºåˆ—ã€‚BLOSUMçŸ©é˜µçš„ç‰ˆæœ¬å¦‚BLOSUM62å¸¸ç”¨äºè›‹ç™½è´¨åºåˆ—æ¯”å¯¹ï¼ˆå±€éƒ¨æ¯”å¯¹ï¼‰ã€‚

  * **DNAçŸ©é˜µï¼š** ç”¨äºæ¯”å¯¹DNAåºåˆ—ï¼Œé€šå¸¸åŒ…æ‹¬ä¸åŒç¢±åŸºä¹‹é—´çš„æ›¿ä»£å¾—åˆ†ã€‚

* **Extreme Value Distribution (æå€¼åˆ†å¸ƒ)**

  * **ç”Ÿæˆåˆ†å¸ƒï¼š** **(ä¾‹)** é€šè¿‡å¯¹1000æ¬¡ä¸1000ä¸ªéšæœºæ‰“ä¹±çš„åºåˆ—è¿›è¡Œæ¯”å¯¹ï¼Œå¾—åˆ°äº†ä¸€ä¸ªåŒ…å«1000ä¸ªæ¯”å¯¹å¾—åˆ†çš„åˆ†å¸ƒã€‚è¿™ä¸ªåˆ†å¸ƒè¢«ç§°ä¸ºExtreme Value Distributionã€‚

  * **ç»˜åˆ¶å›¾è¡¨ï¼š** åˆ©ç”¨è¿™1000ä¸ªæ¯”å¯¹å¾—åˆ†ï¼Œåˆ¶ä½œäº†ä¸€ä¸ªå›¾è¡¨ã€‚é€šè¿‡è§‚å¯Ÿå›¾è¡¨ï¼Œå¯ä»¥çœ‹å‡ºæ¯”å¯¹å¾—åˆ†çš„æ•´ä½“åˆ†å¸ƒæ¨¡å¼ï¼Œå³æå€¼åˆ†å¸ƒçš„ç‰¹å®šæ¨¡å¼ã€‚

  * **æ¯”å¯¹åŸå§‹å¾—åˆ†ï¼š** å°†åŸå§‹çš„æ¯”å¯¹å¾—åˆ†ï¼ˆSï¼‰ä¸æå€¼åˆ†å¸ƒè¿›è¡Œæ¯”å¯¹ã€‚å¦‚æœåŸå§‹å¾—åˆ†ï¼ˆSï¼‰ä½äºåˆ†å¸ƒçš„æå³è¾¹ç¼˜ï¼Œé‚£ä¹ˆè¯´æ˜è¯¥æ¯”å¯¹ç»“æœå¾ˆå¯èƒ½ä¸æ˜¯ç”±äºéšæœºæƒ…å†µè€Œäº§ç”Ÿçš„ï¼Œè€Œæ˜¯ç”±äºåºåˆ—çš„åŒæºæ€§ã€‚

  * **è®¡ç®—P-valueï¼š** é€šè¿‡æå€¼åˆ†å¸ƒçš„å…¬å¼![P\(X\\geq S\)=1-e^{-Kmne^{\\lambda S}}](https://latex.csdn.net/eq?P%28X%5Cgeq%20S%29%3D1-e%5E%7B-Kmne%5E%7B%5Clambda%20S%7D%7D)è®¡ç®—P-valueã€‚ï¼ˆmå’Œnæ˜¯åºåˆ—é•¿åº¦ï¼ŒKå’ŒÎ»æ˜¯åŸºäºè¯„åˆ†çŸ©é˜µçš„å¸¸æ•°ï¼‰P-valueè¡¨ç¤ºåŸå§‹æ¯”å¯¹å¾—åˆ†å‡ºç°åœ¨æˆ–æ›´æç«¯çš„ä½ç½®çš„æ¦‚ç‡ã€‚P-valueè¶Šå°ï¼Œè¯´æ˜æ¯”å¯¹ä¸å¤ªå¯èƒ½æ˜¯ç”±äºéšæœºæƒ…å†µè€Œäº§ç”Ÿçš„ï¼Œä»è€Œå¢åŠ äº†å¯¹åŒæºæ€§çš„æ¨æ–­çš„å¯ä¿¡åº¦ã€‚
</code></pre>
<ul>
<li><p>3. <strong>Word method</strong></p>
<ul>
<li><p>Introduced in <strong>â€œ</strong> <strong>Database similarity searchesâ€</strong></p>
</li>
<li><p>â€œword methodâ€ å¯èƒ½æŒ‡ä»£å¯¹ç”Ÿç‰©åºåˆ—ä¸­çš„k-mer (ä¸€ä¸ª k-mer æ˜¯æŒ‡é•¿åº¦ä¸º k çš„å­ä¸²) è¿›è¡Œå¤„ç†å’Œåˆ†æçš„æ–¹æ³•ã€‚</p>
</li>
</ul>
</li>
</ul>
</blockquote>
<h4 id="Homework-2"><a href="#Homework-2" class="headerlink" title="Homework 2"></a>Homework 2</h4><p>1. There are two amino acid sequences, <strong>seq1: COELACANTH</strong> and <strong>seq2: PELICAN.</strong>**** Obtain the global alignment by using DP (the Needleman-Wunsch algorithm) . </p>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/7233de09aa50.png"></p>
<blockquote>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/4ef17d5d907b.png"></p>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/c9b2a8e7da26.png"></p>
</blockquote>
<p>2. Two different dot matrix analysis servers to analyze the sequence of the human low density lipoprotein receptor(np_000518)</p>
<blockquote>
<p>1.  <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/protein/NP_000518.1">https://www.ncbi.nlm.nih.gov/protein/NP_000518.1</a>  get the origin </p>
<p>2.  <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/Tools/seqstats/emboss_dottup/">https://www.ebi.ac.uk/Tools/seqstats/emboss_dottup/</a>  generate the result </p>
<p>3.  <a target="_blank" rel="noopener" href="https://www.bioinformatics.nl/cgi-bin/emboss/dotmatcher">https://www.bioinformatics.nl/cgi-bin/emboss/dotmatcher</a>  generate the result </p>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/a2b6cd1c9394.png"></p>
<p>The long diagonal indicates that there exists a perfect match from the beginning to the end between the two sequences. (Because the two sequences are the same.)</p>
<p>The shorter diagonals represent the local approximate matches between the two sequences. The reason that those shorter diagonals mostly occur in the lower left corner is that there are many similar patterns before the position of 400.</p>
</blockquote>
<h4 id="Database-similarity-searches"><a href="#Database-similarity-searches" class="headerlink" title="Database similarity searches"></a>Database similarity searches</h4><blockquote>
<p><em>â€œ An application of Pairwise Sequence Alignmentâ€</em></p>
</blockquote>
<blockquote>
<p><strong>The basic procedure for doing this:</strong></p>
<ul>
<li>Submit the sequence of interest (the**â€œ<strong><strong>query</strong></strong>â€ sequence**) </li>
<li>It is aligned in a pairwise manner to EVERY sequence in the<strong>database</strong></li>
<li>Based on these pairwise comparisons, all sequences that have similarity to the query are found </li>
<li>Pairwise alignments between the query and each of these similar sequences are returned as output</li>
</ul>
</blockquote>
<blockquote>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/0b2bd240e658.png"></p>
</blockquote>
<blockquote>
<p><strong>Requirements:</strong> ï¼ˆå°è¯•è®©è¿™ä¸‰è€…è¾¾åˆ°å¹³è¡¡ï¼Œå› ä¸ºæ˜¯ç›¸äº’æ£è‚˜çš„ï¼‰</p>
<p>1. <strong>sensitivity</strong> (æ¼åˆ¤è¶Šå°‘æ•æ„Ÿæ€§è¶Šå¥½)</p>
<p>2. <strong>specificity(selectivity)</strong> (è¯¯åˆ¤è¶Šå°‘ç‰¹å¼‚æ€§è¶Šå¥½)</p>
<p>3. <strong>speed</strong> (ä»æ•°æ®åº“æœç´¢è·å–è¾“å‡ºæ‰€éœ€çš„æ—¶é—´è¶Šå°‘é€Ÿåº¦è¶Šå¿«)</p>
<p>Idea: åŠ¨æ€è§„åˆ’æ˜¯ç©·ä¸¾ç®—æ³•ï¼Œæ—¶é—´æ•ˆç‡ä½ï¼Œåœ¨ä¸å½±å“å¤ªå¤šæ•æ„Ÿæ€§å’Œç‰¹å¼‚æ€§çš„æ¡ä»¶ä¸‹ï¼Œæˆ‘ä»¬å¯ä»¥é‡‡ç”¨<strong>å¯å‘å¼ç®—æ³•(Heuristic Algorithms),</strong> but ä¸èƒ½ä¿è¯æ‰¾åˆ°æœ€ä¼˜&#x2F;æœ€å‡†ç¡®çš„ç»“æœ</p>
<p><strong>Shortcut:</strong> â€œTestâ€ each sequence to see whether it is worth aligning with <strong>query_seq</strong></p>
<p>å¸¸è§çš„å¯å‘å¼ç®—æ³•ï¼š<strong>BLAST</strong> å’Œ <strong>FASTA</strong> å®ƒä»¬å°±æ˜¯æ‰€è°“çš„<strong>WORD METHOD</strong></p>
<ul>
<li><p><strong>Meritsï¼š</strong> 50-100 times faster than DP</p>
</li>
<li><p><strong>Shortsï¼š</strong> not guaranteed to find the best&#x2F;most accurate result</p>
</li>
<li><p>Based on finding short stretches of identical or nearly identical letters (which are called â€œwordsâ€, â€œ <em>k-tuples</em> â€, or â€œ <em>ktups</em> â€) in two sequences.</p>
</li>
<li><p>åŸºæœ¬è¿‡ç¨‹ï¼š</p>
<ul>
<li><p>â€‹â€‹â€‹â€‹â€‹â€‹â€‹â€‹â€‹â€‹â€‹â€‹â€‹â€‹ <strong>å¯»æ‰¾ç›¸åŒçš„word(s)ï¼š</strong> æ‰¾åˆ°ä¸¤ä¸ªåºåˆ—ä¸­å…±æœ‰çš„wordæˆ–è¯ç»„ï¼Œå¹¶å¯¹å®ƒä»¬è¿›è¡Œæ¯”å¯¹ã€‚ </p>
</li>
<li><p><strong>é€šè¿‡å¯¹é½wordæ‰©å±•æ¯”å¯¹ï¼š</strong> é€šè¿‡å¯¹é½æ¯ä¸ªwordä¸¤ä¾§çš„ç›¸ä¼¼åŒºåŸŸï¼Œæ‰©å±•æ¯”å¯¹çš„èŒƒå›´ã€‚</p>
</li>
<li><p><strong>è®¡ç®—å¯¹é½åŒºåŸŸçš„å¾—åˆ†ï¼š</strong> å¯¹æ‰©å±•åçš„å¯¹é½åŒºåŸŸè¿›è¡Œå¾—åˆ†è®¡ç®—ï¼Œé€šå¸¸ä½¿ç”¨æŸç§è¯„åˆ†çŸ©é˜µæˆ–å¾—åˆ†è§„åˆ™æ¥è¡¡é‡ç›¸ä¼¼æ€§ã€‚</p>
</li>
<li><p><strong>è¿æ¥é«˜åˆ†åŒºåŸŸï¼š</strong> å°†ç›¸é‚»çš„é«˜åˆ†å¯¹é½åŒºåŸŸè¿æ¥åœ¨ä¸€èµ·ï¼Œå½¢æˆæœ€ç»ˆçš„å±€éƒ¨æ¯”å¯¹ç»“æœã€‚</p>
</li>
</ul>
</li>
</ul>
</blockquote>
<h5 id="BLAST-B-asic-L-ocal-A-lignment-S-earch-T-ool"><a href="#BLAST-B-asic-L-ocal-A-lignment-S-earch-T-ool" class="headerlink" title="BLAST: B asic L ocal A lignment S earch T ool"></a><strong>BLAST: B</strong> asic <strong>L</strong> ocal <strong>A</strong> lignment <strong>S</strong> earch <strong>T</strong> ool</h5><blockquote>
<ol>
<li><p><strong>å±€éƒ¨æ¯”å¯¹ï¼š</strong> BLASTé‡‡ç”¨å±€éƒ¨æ¯”å¯¹çš„ç­–ç•¥ï¼Œå³å…ˆå¯»æ‰¾ä¸¤ä¸ªåºåˆ—ä¹‹é—´çš„å±€éƒ¨ç›¸ä¼¼æ€§ï¼Œç„¶åæ ¹æ®è¿™äº›å±€éƒ¨ç›¸ä¼¼æ€§è¿›è¡Œè¿›ä¸€æ­¥çš„åˆ†æã€‚</p>
</li>
<li><p><strong>æŸ¥è¯¢åºåˆ—ï¼š</strong> ç”¨æˆ·æä¾›ä¸€ä¸ªæŸ¥è¯¢åºåˆ—ï¼Œå¯ä»¥æ˜¯DNAã€RNAæˆ–è›‹ç™½è´¨åºåˆ—ã€‚</p>
</li>
<li><p><strong>æ•°æ®åº“æœç´¢ï¼š</strong> BLASTå°†æŸ¥è¯¢åºåˆ—ä¸ç›®æ ‡æ•°æ®åº“ä¸­çš„åºåˆ—è¿›è¡Œæ¯”å¯¹ã€‚ä»¥è›‹ç™½è´¨ä¸ºä¾‹ï¼Œä¸€ä¸ªé•¿åº¦ä¸ºä¸‰çš„è›‹ç™½è´¨å•è¯è¦ä¸<strong>20<em>20</em>20</strong> &#x3D;8000ä¸ªè¯æ¯”å¯¹ã€‚</p>
</li>
<li><p><strong>æ¯”å¯¹ç®—æ³•ï¼š</strong> BLASTä½¿ç”¨å¯å‘å¼ç®—æ³•ï¼Œé€šè¿‡æ„å»ºçŸ­è¯ï¼ˆwordsï¼‰çš„ç´¢å¼•æ¥åŠ é€Ÿæœç´¢ã€‚å®ƒé¦–å…ˆæ‰¾åˆ°ä¸æŸ¥è¯¢åºåˆ—ä¸­çš„çŸ­è¯ç›¸åŒ¹é…çš„ç›®æ ‡åºåˆ—ï¼Œç„¶åé€šè¿‡å±€éƒ¨æ¯”å¯¹çš„æ–¹å¼é€æ­¥æ‰©å±•åŒ¹é…åŒºåŸŸã€‚å•è¯é•¿åº¦è›‹ç™½è´¨é€šå¸¸æ˜¯3ä¸ªï¼ŒDNAé€šå¸¸æ˜¯11ä¸ª</p>
</li>
<li><p><strong>è®¡ç®—åˆ†æ•°ï¼š</strong> BLASTæ ¹æ®ç›¸ä¼¼æ€§å¾—åˆ†ï¼ˆæ ¹æ®é€‰æ‹©çš„æ›¿ä»£çŸ©é˜µï¼Œä¾‹å¦‚BLOSUMæˆ–PAMçŸ©é˜µï¼‰å¯¹åŒ¹é…åŒºåŸŸè¿›è¡Œæ‰“åˆ†ï¼Œé»˜è®¤æ˜¯BLOSUM62ã€‚ä¸€èˆ¬æ¯”å¯¹<strong>é˜ˆå€¼(threshold)</strong> æ˜¯T&#x3D;11. å¦‚æœè¶…è¿‡äº†é˜ˆå€¼ï¼Œé‚£ä¹ˆå°±è®¤ä¸ºå¯ä»¥è¢«æ›¿æ¢ï¼Œä½œä¸ºâ€seedâ€, ä¾‹å¦‚ä¸‹å›¾ï¼š</p>
<ol>
<li><p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/31b8ab8f5b38.png"></p>
</li>
<li><p>å¦‚æœåœ¨è¿™ä¸ªé…å¯¹é™„è¿‘åˆå‡ºç°äº†é…å¯¹ï¼Œå¯ä»¥å°†å®ƒä»¬<strong>è¿æ¥</strong> èµ·æ¥å½¢æˆä¸€æ®µæ›´é•¿çš„æ— é—´éš”çš„é…å¯¹ï¼ˆæ—©æœŸBLASTä¸ä¼šè¿™æ ·ï¼‰</p>
</li>
<li><p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/ef7954f03d1e.png"></p>
</li>
<li><p>åŸºäºè¿™ä¸ªé…å¯¹åºåˆ—ï¼Œç»§ç»­å‘ä¸¤è¾¹æ‰©å±•ï¼Œç›´åˆ°åˆ†æ•°å¼€å§‹ä¸‹é™ï¼Œæœ€ç»ˆå¾—åˆ°çš„å°±æ˜¯<strong>HSP(é«˜åˆ†æ®µé…å¯¹)</strong></p>
</li>
<li><p>HSPåˆ†æ•°å¤§äºæŸ<strong>æˆªæ–­å€¼(cutoff value)</strong> Sæ—¶ä¿ç•™ï¼Œå¦åˆ™ä¸¢å¼ƒï¼›</p>
</li>
<li><p>å•ä¸ªæ•°æ®åº“æ‰¾åˆ°å¤šä¸ªHSPå°±è¿æ¥èµ·æ¥å½¢æˆ<strong>æœ‰ç¼ºå£çš„</strong> é…å¯¹ï¼ˆæ—©æœŸBLASTä¸ä¼šè¿™æ ·ï¼‰</p>
</li>
<li><p>æœ€åç»™æŸ¥è¯¢åºåˆ—å’Œé…å¯¹åºåˆ—è¿›è¡ŒSWé…å¯¹ï¼Œé…å¯¹çš„æ•°æ®åº“åºåˆ—ç§°ä¸ºâ€œ<strong>å‘½ä¸­(hits)</strong> â€æŸ¥è¯¢åºåˆ—</p>
</li>
</ol>
</li>
<li><p><strong>è¾“å‡ºç»“æœï¼š</strong> BLASTè¾“å‡ºåŒ¹é…çš„åºåˆ—åŠå…¶ç›¸ä¼¼æ€§å¾—åˆ†ï¼ŒåŒæ—¶æä¾›äº†å¯è§†åŒ–çš„æ¯”å¯¹å›¾å½¢å’Œå…¶ä»–æœ‰å…³åŒ¹é…çš„ä¿¡æ¯ã€‚</p>
</li>
<li><p><strong>P-valueï¼š</strong> BLASTçš„è¾“å‡ºä¸­é€šå¸¸åŒ…æ‹¬P-valueï¼Œè¡¨ç¤ºåœ¨éšæœºæƒ…å†µä¸‹å¾—åˆ°ç›¸åŒæˆ–æ›´å¥½åŒ¹é…çš„æœŸæœ›æ¬¡æ•°ã€‚P-valueè¶Šå°ï¼Œè¡¨ç¤ºåŒ¹é…è¶Šæ˜¾è‘—ã€‚(We have mentioned this in the SW algorithm part)</p>
<ol>
<li><strong>æ³¨æ„</strong> ï¼šå®é™…ä¸Šmã€nçš„å®šä¹‰å˜åŒ–äº†ï¼Œåœ¨è¿™é‡Œå®ƒä»¬æŒ‡çš„éƒ½æ˜¯æœ‰æ•ˆå€¼ï¼Œéƒ½æ˜¯ç”¨<strong>å®é™…é•¿åº¦å‡å»ä¸¤ä¸ªå¯¹é½åºåˆ—çš„å¯¹é½é•¿åº¦çš„å¹³å‡å€¼</strong></li>
</ol>
</li>
<li><p><strong>E-valueï¼šExpected Value</strong> æŒ‡çš„æ˜¯the number of high-scoring matches we expect to observe between the query and unrelated sequences. è®¡ç®—å…¬å¼å¦‚ä¸‹</p>
<ol>
<li><p>â€‹â€‹â€‹â€‹â€‹â€‹â€‹<img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/ea06a6081db8.png"></p>
</li>
<li><p><strong>D</strong> : <strong>æ•°æ®åº“é‡Œåºåˆ—æ•°ç›®</strong> ï¼ˆæ•°æ®åº“çš„å¤§å°sizeï¼‰</p>
</li>
<li><p><strong>p</strong> : æ ¹æ®æå€¼åˆ†å¸ƒï¼Œè·å¾—å¯¹é½<strong>å¾—åˆ†â‰¥Sçš„æ¦‚ç‡</strong></p>
</li>
<li><p><strong>mâ€™: æ•°æ®åº“çš„æœ‰æ•ˆé•¿åº¦ï¼ˆæ•°æ®åº“å¤§å°çš„å¦ä¸€ç§åº¦é‡ï¼‰</strong></p>
</li>
<li><p>The  <strong>E-value</strong> is  <strong>the number of HSPâ€™s</strong> with an alignment score â‰¥ S that are expected to occur  by chance when searching a database of <em>D</em> sequences.</p>
</li>
<li><p>â€œ<em>The smaller the E-value, the less likely the alignment is due to chance</em> â€œ</p>
</li>
<li><p>æœç´¢æŸä¸€ä¸ªæ•°æ®åº“è·å¾—çš„Eå€¼ä¸èƒ½ç”¨äºå…¶ä»–æ•°æ®åº“ï¼Œå› ä¸ºæ•°æ®åº“å¤§å°è¶Šå¤§ï¼ŒEå€¼æ›´å€¾å‘äºè¶Šå¤§ã€‚</p>
</li>
</ol>
</li>
<li><p><em>â€œA p-value is a way of representing the significance of an alignment. An E-value is simply another way of representing the significance of that alignment.â€</em></p>
</li>
<li><p><strong>The normalization of scores</strong> : å¯¹é½çš„åŸå§‹åˆ†æ•°å–å†³äºè¯„åˆ†çŸ©é˜µå’Œé—´éš™æƒ©ç½šï¼Œè€Œæ ‡å‡†åŒ–åˆ†æ•°å¯ä»¥ç‹¬ç«‹äºè¯„åˆ†çŸ©é˜µæ¥è¯„ä¼°ä¸åŒBLASTæŸ¥è¯¢å¾—å‡ºçš„ç»“æœæ¥åˆ¤æ–­å“ªä¸ªå¯¹é½æ›´åˆç†ã€‚</p>
</li>
</ol>
<pre><code> 1. â€‹â€‹â€‹â€‹â€‹â€‹â€‹è®¡ç®—å…¬å¼ ![](/images/posts/Summerize-for-Bioinformatics-w/d2ea6887b295.png)

 2. _S_ = raw alignment score åŸå§‹å¯¹é½åˆ†æ•°ï¼› Î» = constant that depends on scoring matrix è¯„åˆ†çŸ©é˜µç»™å‡ºçš„å¸¸æ•°ï¼› _K_ = constant that depends on scoring matrix è¯„åˆ†çŸ©é˜µç»™å‡ºçš„å¸¸æ•°

 3. ![](/images/posts/Summerize-for-Bioinformatics-w/d3d9ece75110.png)
</code></pre>
<ol start="11">
<li><strong>LCR: Low Complexity Regions</strong> ï¼š ä½å¤æ‚åº¦åŒºåŸŸï¼Œä¿¡æ¯é‡å¤ä¸”å•ä¸€ï¼Œéƒ½æ˜¯çŸ­è€Œé‡å¤çš„ç‰‡æ®µï¼Œä¼šå¹²æ‰°æœ€ç»ˆè¾“å‡ºçš„é…å¯¹ç‰‡æ®µï¼Œåœ¨è›‹ç™½è´¨åºåˆ—ä¸­å çº¦15%ï¼Œå¯èƒ½ä¼šå¯¼è‡´ è™šå‡å‘½ä¸­å’Œè™šé«˜é…å¯¹åˆ†æ•°ï¼ˆspurious hits and artificially high alignment scoresï¼‰ã€‚</li>
</ol>
<pre><code> 1. Avoid method 1: **Filtering**. åœ¨æŸ¥è¯¢åŒºåŸŸæŠŠ LCRç‰‡æ®µç”¨å°å†™å­—æ¯æˆ–æ­§ä¹‰å­—ç¬¦æ›¿æ¢ï¼ˆ **è›‹ç™½è´¨Xï¼ŒDNAæ˜¯N** ï¼‰ï¼Œå½“ç„¶è¿™ä¹ˆåš å¯èƒ½ä¼šé—æ¼ä¸€äº›çœŸå®åŒ¹é… 

 2. Avoid Method 2: **Masking**. LCRä¸­çš„å­—ç¬¦åœ¨æ„ å»ºæŸ¥è¯¢å•è¯è¡¨ä¸­è¢«å¿½ç•¥ï¼Œä½†æ˜¯å¯ä»¥ åœ¨æ‰©å±•å’Œå¯¹é½ä¼˜åŒ–ä¸­ä½¿ç”¨
</code></pre>
</blockquote>
<h5 id="FASTA-FAST-All"><a href="#FASTA-FAST-All" class="headerlink" title="FASTA: FAST-All"></a><strong>FASTA: FAST-All</strong></h5><blockquote>
<ol>
<li><p><strong>å¼€å‘è€…ï¼š</strong> ç”± David Lipman å’Œ William Pearson äº 1988 å¹´å¼€å‘ã€‚</p>
</li>
<li><p><strong>é¦–ä¸ªå¹¿æ³›ä½¿ç”¨çš„å¿«é€Ÿæ•°æ®åº“æœç´¢ç¨‹åºï¼š</strong> FASTA æ˜¯ç¬¬ä¸€ä¸ªè¢«å¹¿æ³›ä½¿ç”¨çš„ç”¨äºè¿›è¡Œå¿«é€Ÿæ•°æ®åº“æœç´¢çš„ç¨‹åºã€‚å®ƒçš„ä¸»è¦ç›®çš„æ˜¯åœ¨å¤§å‹ç”Ÿç‰©æ•°æ®åº“ä¸­å¿«é€Ÿæ‰¾åˆ°ä¸æŸ¥è¯¢åºåˆ—ç›¸ä¼¼çš„åºåˆ—ã€‚</p>
</li>
<li><p><strong>å¯å‘å¼è¯æ³•æ–¹æ³•ï¼š</strong> FASTA ä½¿ç”¨ä¸€ç§å¯å‘å¼çš„ word æ–¹æ³•æ¥åˆ›å»ºæŸ¥è¯¢åºåˆ—ä¸æ•°æ®åº“åºåˆ—ä¹‹é—´çš„å±€éƒ¨æ¯”å¯¹ã€‚è¯¥æ–¹æ³•é€šè¿‡å¯»æ‰¾ä¸¤ä¸ªåºåˆ—ä¸­çš„è¯ï¼ˆwordsï¼‰çš„ç›¸ä¼¼æ€§æ¥è¿›è¡Œæ¯”å¯¹ã€‚</p>
</li>
<li><p><strong>ä»ç²¾ç¡®åŒ¹é…çš„è¯å¼€å§‹ï¼š</strong> FASTA åœ¨æ¯”å¯¹è¿‡ç¨‹ä¸­ä»å¯»æ‰¾ä¸¤ä¸ªåºåˆ—ä¸­çš„è¯çš„ç²¾ç¡®åŒ¹é…å¼€å§‹ã€‚åœ¨è›‹ç™½è´¨åºåˆ—ä¸­ï¼Œé€šå¸¸ä½¿ç”¨ ktup &#x3D; 1 æˆ– 2ï¼Œè€Œåœ¨DNAåºåˆ—ä¸­ï¼Œé€šå¸¸ä½¿ç”¨ 4-6ã€‚</p>
</li>
</ol>
</blockquote>
<h5 id="Comparison-between-BLAST-FASTA"><a href="#Comparison-between-BLAST-FASTA" class="headerlink" title="Comparison between BLAST &amp; FASTA"></a>Comparison between BLAST &amp; FASTA</h5><blockquote>
<p>â€¢ BLAST is faster </p>
<p>â€¢ FASTA generally produces a better final alignment </p>
<p>â€¢ FASTA is more likely to find distantly related sequences </p>
<p>â€¢ Performance of the two methods is similar for highly similar sequences </p>
<p>â€¢ Both methods are appropriate for rapid initial searches</p>
</blockquote>
<h4 id="Homework-3"><a href="#Homework-3" class="headerlink" title="Homework 3"></a>Homework 3</h4><p><strong>1.</strong> Answer the following questions about scoring an alignment.</p>
<p><strong>A.</strong> Calculate the score for the DNA sequence alignment shown below, using the scoring matrix below. Use an affine gap penalty to score the gaps, with -11 for opening the gap and -1 for each additional position in the gap. (â€œAffine gap penaltyâ€ refers to a situation when the gap opening and gap extension penalties are not the same. The gap opening penalty should be greater than the gap extension penalty.) </p>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/5ccd10240cec.png"></p>
<blockquote>
<p>**2x27â€“5x4â€“11â€“2&#x3D;21 **</p>
<p>Match scores: 27 matching bases: 27 * 2 &#x3D; 54 points</p>
<p>Mismatch score: 4 mismatched base: 4 * -5 &#x3D; -20 points</p>
<p>Gap scores: 1 gap in the first sequence: 1 * -11 &#x3D; -11 points<br> Each extension of the gap: 2 * -1 &#x3D; -2 points </p>
<p>The total gap score: -11 - 2 &#x3D; -13 points</p>
<p>Overall alignment score &#x3D; Match scores + Mismatch scores + Total Gap score &#x3D; 54 + (-20) + (-13) &#x3D; <strong><strong>21</strong></strong></p>
</blockquote>
<p><strong>B.</strong> How would the score change if you were to use a nonaffine gap penalty? To answer the question, try a nonaffine penalty of -2, and then -6.</p>
<blockquote>
<p>2x27 â€“ 5x4 â€“ 3x2 &#x3D; 28(for nonaffine penalty -2) </p>
<p>2x27 â€“ 5x4 â€“ 3x6 &#x3D; 16(for nonaffine penalty -6)</p>
<p>With non-affine gap penalty of -2 per gap:</p>
<p>There would now be 1 gap that receives a linear penalty of -2 * 3 bases &#x3D; -6 </p>
<p>Overall score is: Matches (54) + Mismatch (-20) + Gaps (-6) &#x3D; 54-20-6 &#x3D; <strong><strong>28</strong></strong></p>
<p>With non-affine penalty of -6 per gap: </p>
<p>The 1 gap gets a score of -6 * 1 gap &#x3D; -6 </p>
<p>Overall score is: Matches (54) + Mismatch (-20) + Gaps (-18) &#x3D; 54-20-18 &#x3D; <strong><strong>16</strong></strong></p>
</blockquote>
<p><strong>2.</strong> Below is part of the BLOSUM62 matrix. Answer the following questions using this matrix.</p>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/763d47eb658b.png"></p>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/d5030e19f033.png"></p>
<p><strong>A.</strong> Using this matrix, two aligned cysteines (C) would receive a score of 9 while two aligned threonines (T) would only receive a score of 5. What can you conclude about cysteine relative to threonine?</p>
<blockquote>
<p>Cysteines (C) is rarer than threonines (T) . The higher the match score the rarer the match.</p>
</blockquote>
<p><strong>B.</strong> A serine (S) aligned with a cysteine (C) would receive a negative score (-1) while a serine aligned with a threonine would receive a positive score (1). Offer a possible explanation for this in terms of physicochemical properties of the amino acid side chains.</p>
<blockquote>
<p>The Mercapto is difficult to generate hydrogen bond than Hydroxyl </p>
</blockquote>
<p><strong><strong>C.</strong></strong>**** The bestng an alignment score. Use the BLOSUM62 matrix alignment of the two amino acid sequences â€œLDSâ€ and â€œLNSâ€ is obvious (itâ€™s shown below). Given a scoring system, you could easily calculate an alignment score. Set up a matrix and use the dynamic programming algorithm to â€œproveâ€ that this is the best alignment by calculating x (see Powerpoint notes) to score aligned residues, and use a gap penalty of -1. (You may hand write the matrix in your homework rather than typing it if you like.</p>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/0dd1f343c9a7.png"></p>
<blockquote>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/cea6d3abe084.png"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/252617ffc1b5.png"></p>
<p>It receives an alignment score of 4 + 1 + 4 &#x3D; 9</p>
<p>The matrix verifies this is the maximum possible alignment score using these sequences and rules. So it proves this alignment is optimal.</p>
</blockquote>
<p><strong>3</strong><strong>.</strong> Find the optimal global alignment of the two sequences **<em>Seq</em> 1: THISLINE and <em>Seq</em> 2: ISALIGNED **based on the BLOSUM62 matrix with linear gap penalty of -4.</p>
<blockquote>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/526bbd13b42a.png"></p>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/067e65bdb364.png"></p>
</blockquote>
<h4 id="Multiple-sequence-alignment"><a href="#Multiple-sequence-alignment" class="headerlink" title="Multiple sequence alignment"></a>Multiple sequence alignment</h4><p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/e6808c3bf895.png"></p>
<p>*å…¨åŠ¨æ€ç¼–ç¨‹å¯¹äºå¤šåºåˆ—æ¯”å¯¹æ˜¯ä¸å®ç”¨çš„ã€‚åºåˆ—é•¿æ®‹åŸºå¤šæ—¶å€™æ¯”è¾ƒæŒ‡æ•°å¢é•¿</p>
<ul>
<li>Categories of Heuristic Algorithms for Multiple Sequence Alignment:</li>
</ul>
<h5 id="1-Progressive-alignment-methods"><a href="#1-Progressive-alignment-methods" class="headerlink" title="1. Progressive alignment methods"></a>1. Progressive alignment methods</h5><p>ã€first calculate a pairwise alignment for the two most closely related sequences and </p>
<p>progressively add more sequences to the alignment.ã€‘P8&#x2F;p16</p>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/8a232a542315.png"></p>
<h5 id="2-Iterative-alignment-methods"><a href="#2-Iterative-alignment-methods" class="headerlink" title="2. Iterative alignment methods"></a>2. Iterative alignment methods</h5><p>ã€Iterative alignment methods attempt to correct a problem of progressive methods: </p>
<p>Errors in the initial alignments of the most closely related sequences are propagated to the final multiple sequence alignment.ã€‘</p>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/ae7fb1ebf981.png"></p>
<h5 id="PSSMæ–¹æ³•"><a href="#PSSMæ–¹æ³•" class="headerlink" title="PSSMæ–¹æ³•"></a><strong><strong>PSSMæ–¹æ³•</strong></strong></h5><blockquote>
<p>ã€PSSMä¸­çš„æ¯ä¸ªå€¼æ˜¯åœ¨å¤šåºåˆ—æ¯”å¯¹ä¸­ç»™å®šä½ç½®å¤„è§‚å¯Ÿåˆ°çš„è¯¥æ°¨åŸºé…¸çš„è®¡æ•°é™¤ä»¥åœ¨æ¯”å¯¹ä¸­è¯¥ä½ç½®å¤„è¯¥æ°¨åŸºé…¸çš„é¢„æœŸè®¡æ•°çš„æ¯”ç‡ã€‚ç„¶åå°†è¯¥å€¼è½¬æ¢ä¸ºå¯¹æ•°åˆ»åº¦ï¼ˆé€šå¸¸ä»¥2ä¸ºåŸºæ•°ï¼‰ã€‚</p>
<p>PSSMä¸­çš„å€¼åæ˜ äº†ä»»ä½•ç»™å®šæ°¨åŸºé…¸å‡ºç°åœ¨æ¯”å¯¹ä¸­çš„æ¯ä¸ªä½ç½®çš„æ¦‚ç‡ï¼Œä»¥åŠåœ¨æ¯”å¯¹ä¸­æ¯ä¸ªä½ç½®çš„ä¿å®ˆæˆ–éä¿å®ˆå–ä»£çš„å½±å“ã€‚ã€‘</p>
</blockquote>
<h4 id="Motifs-and-domains"><a href="#Motifs-and-domains" class="headerlink" title="Motifs and domains"></a>Motifs and domains</h4><blockquote>
<p>*Motifâ€“ a short conserved sequence pattern; can be just a few amino acid residues, up to ~20.</p>
<p>*Domainâ€“ a longer conserved sequence pattern which adopts a particular three-dimensional structure and is an independent functional and structural unit; typically 40-700 residues.</p>
<p>*å‰å¸ƒæ–¯é‡‡æ ·ç®—æ³•å‰å¸ƒæ–¯é‡‡æ ·æ˜¯ä¸€ä¸ªåŸç†ä¸Šç±»ä¼¼äºEMçš„è¿­ä»£è¿‡ç¨‹ï¼Œä½†ç®—æ³•ä¸åŒã€‚åœ¨æ¯æ¬¡è¿­</p>
<p>ä»£ä¸­ï¼Œç§»é™¤ä¸€ä¸ªåºåˆ—ï¼Œå¹¶ä»å‰©ä½™åºåˆ—ä¸­æ„å»ºè¯•éªŒæ¯”å¯¹ã€‚ä¸EMä¸€æ ·ï¼ŒGibbsé‡‡æ ·åœ¨ä¸€ç»„åº</p>
<p>åˆ—ä¸­æœç´¢ç»Ÿè®¡ä¸Šæœ€å¯èƒ½çš„åŸºåºï¼Œå¹¶å¯ä»¥åœ¨æ¯ä¸ªåºåˆ—ä¸­æ‰¾åˆ°è¿™äº›åŸºåºçš„æœ€ä½³å®½åº¦å’Œæ•°é‡ã€‚</p>
<p>*é…ç½®æ–‡ä»¶HMMï¼šè½®å»“HMMå¯ä»¥ä»ä¸€ç»„æœªå¯¹é½çš„åºåˆ—ä¸­æ„å»ºã€‚å¯¹æ¨¡å‹çš„é•¿åº¦ï¼ˆåŒ¹é…çŠ¶æ€çš„æ•°é‡ï¼‰å’Œæ¦‚ç‡å‚æ•°è¿›è¡Œäº†åˆæ­¥çŒœæµ‹ã€‚ç„¶åé€šè¿‡è¿­ä»£æ›´æ–°æ¦‚ç‡å‚æ•°æ¥è®­ç»ƒæ¨¡å‹ï¼ˆè¯¥è¿‡ç¨‹å¯ä»¥ä½¿ç”¨å„ç§ç®—æ³•ï¼‰ã€‚</p>
<ul>
<li>EM-algorithmï¼šStep 1. Maximization Step 2. Expectation Step 3. Repeat Step 1 and 2.</li>
</ul>
</blockquote>
<blockquote>
<ol>
<li>æœŸæœ›æœ€å¤§åŒ–(Expectation Maximization, EM)ç®—æ³• è¯¥ç®—æ³•é€šè¿‡è¿­ä»£çš„æ–¹æ³•ä»ä¸€ç»„åºåˆ—ä¸­å‘ç°ç»Ÿè®¡ä¸Šæœ€å¯èƒ½çš„åŸºåºæ¨¡å¼(motifæˆ–domain)ã€‚å®ƒä¸ä¾èµ–äºäº‹å…ˆäº§ç”Ÿçš„å¤šé‡åºåˆ—æ¯”å¯¹ã€‚</li>
<li>å‰å¸ƒæ–¯é‡‡æ ·(Gibbs Sampler)ç®—æ³•<br> ä¸EMç®—æ³•ç±»ä¼¼çš„è¿­ä»£è¿‡ç¨‹,ä¹Ÿèƒ½åœ¨ä¸€ç»„åºåˆ—ä¸­å‘ç°motifæˆ–domain,å¹¶èƒ½æ‰¾å‡ºæ¯æ¡åºåˆ—ä¸­è¿™äº›æ¨¡å¼çš„æœ€ä½³é•¿åº¦å’Œæ•°é‡ã€‚</li>
<li>é…ç½®éšé©¬å°”å¯å¤«æ¨¡å‹(Profile Hidden Markov Model) å¯ä»¥ä»ä¸€ç»„æœªæ¯”å¯¹çš„åºåˆ—ä¸­æ„å»ºã€‚é€šè¿‡è¿­ä»£æ›´æ–°æ¦‚ç‡å‚æ•°å¯¹æ¨¡å‹è¿›è¡Œè®­ç»ƒã€‚</li>
</ol>
</blockquote>
<blockquote>
<p>è¿™äº›éƒ½æ˜¯ç”¨ç»Ÿè®¡å­¦æ–¹æ³•ä»åºåˆ—æ•°æ®é›†ä¸­å­¦ä¹ motifå’Œdomainçš„æ— ç›‘ç£å­¦ä¹ ç®—æ³•ã€‚ä¸éœ€è¦äº‹å…ˆçš„åºåˆ—æ¯”å¯¹å°±èƒ½å‘ç°åºåˆ—æ¨¡å¼ã€‚</p>
</blockquote>
<h4 id="Homework-4"><a href="#Homework-4" class="headerlink" title="Homework 4"></a>Homework 4</h4><ol>
<li><strong>Identifying protein motifs using Expectation Maximization (EM) algorithm.</strong></li>
</ol>
<p>Suppose we have 200 protein sequences of length 100, in the dataset. Each character represents an amino acid. The length of the motif is 10. The figure below shows the first 10 samples in the dataset.</p>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/bd76fccc0100.png"></p>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/d1330c486812.png">Write a program to identify the starting position of the motif in each protein sequence. Write a report which includes: a) your understanding on EM algorithm, b) the core algorithm, c) the final result (starting positions and PSSM of the motif), d) the difficulties you met.</p>
<blockquote>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/a9c5e768b0b0.png"></p>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/272b78939fdd.png"></p>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/b39e68e2bfa6.png"></p>
<ul>
<li><h5 id="My-understanding-to-Expectation-Maximization-EM-Algorithm"><a href="#My-understanding-to-Expectation-Maximization-EM-Algorithm" class="headerlink" title="My understanding to Expectation Maximization(EM) Algorithm:"></a><strong>My understanding to Expectation Maximization(EM) Algorithm:</strong></h5></li>
</ul>
</blockquote>
<blockquote>
<ol>
<li>The EM algorithm is an iterative method that estimates model parameters where there are missing, hidden, or unobserved values.</li>
<li>It proceeds in two steps that are repeated until convergence: <ol>
<li>Expectation (E) Step: Creating a function for the expectation of the log-likelihood evaluated using the current estimate for parameters. This involves estimating values for latent variables based on the current parameter values.</li>
<li>Maximization (M) Step: Finding parameters that maximize the expected log-likelihood found on the E step. These become the new parameter estimates.</li>
</ol>
</li>
<li>By iteratively doing E and M steps, EM finds the parameters that locally maximize the likelihood of the data comprising both observed and unobserved values.</li>
<li>For motif finding, the â€œunobservedâ€ data are the true starting positions of each motif instance. The expectation step estimates motif locations, while the maximization step finds an updated motif model that fits best based on those expected motif positions.</li>
</ol>
</blockquote>
<blockquote>
<ul>
<li><h5 id="Core-Algorithm"><a href="#Core-Algorithm" class="headerlink" title="Core Algorithm:"></a><strong>Core Algorithm:</strong></h5></li>
</ul>
</blockquote>
<blockquote>
<p>Fake code:</p>
<ol>
<li>Initialize motif starting positions randomly in each sequence</li>
<li>E-step: <ol>
<li>Extract motif instances from each sequence based on current start positions</li>
<li>Calculate probability of each position being the true start given current motif model</li>
</ol>
</li>
<li>M-step: <ol>
<li>Update start position for each sequence to position with highest probability from E-step</li>
<li>Update PSSM for motif based on new aligned motif instances</li>
</ol>
</li>
<li>Repeat E-step and M-step until convergence</li>
<li>Output final motif start positions and PSSM</li>
</ol>
</blockquote>
<blockquote>
<h5 id="Code-in-PyCharm-python-3-8"><a href="#Code-in-PyCharm-python-3-8" class="headerlink" title="Code in PyCharm: (python&#x3D;3.8)"></a>Code in PyCharm: (python&#x3D;3.8)</h5><p><code>python # Created at WASEDA Univ. # Author: Vector Kun # Time: 2023/12/17 12:44 import csv from random import randint as randint from math import log as log import matplotlib.pyplot as plt from numpy import mean from matplotlib.pyplot import MultipleLocator # Define basic parameters proteinsDataset = &quot;proteins.csv&quot; resultFile = &quot;result.txt&quot; trainNum = 200 trainTime = 10 motif_length = 10 def csv_reader(path): &quot;&quot;&quot;Read data from csv&quot;&quot;&quot; origin_protein_list = [] with open(path) as f: reader = csv.reader(f) for item in reader: origin_protein_list.append(item) # print(origin_protein_list) return origin_protein_list def data_preprocess(path): &quot;&quot;&quot;Remove order in the data&quot;&quot;&quot; protein_order_list = csv_reader(path)[1:] protein_list = [] for protein in protein_order_list: protein_list.append(protein[1:]) # print(protein_list) return protein_list def matrix_transposition(matrix): &quot;&quot;&quot;Transpose the matrix&quot;&quot;&quot; new_matrix = [] for i in range(len(matrix[0])): matrix1 = [] for j in range(len(matrix)): matrix1.append(matrix[j][i]) new_matrix.append(matrix1) # print(new_matrix) return new_matrix def PSSM_calculation(seqs): &quot;&quot;&quot;Calculate pssm matrix&quot;&quot;&quot; # The species of the aminos amino_acids_list = [&#39;A&#39;, &#39;C&#39;, &#39;D&#39;, &#39;E&#39;, &#39;F&#39;, &#39;G&#39;, &#39;H&#39;, &#39;I&#39;, &#39;K&#39;, &#39;L&#39;, &#39;M&#39;, &#39;N&#39;, &#39;P&#39;, &#39;Q&#39;, &#39;R&#39;, &#39;S&#39;, &#39;T&#39;, &#39;V&#39;, &#39;W&#39;, &#39;Y&#39;] # PSSM matrix weight_matrix = {} # Frequency of amino acids fre_matrix = {} # Number of occurrences of amino acids at each position count_matrix = {} # Initialization for amino in amino_acids_list: count_matrix[amino] = [0] * motif_length fre_matrix[amino] = 0 weight_matrix[amino] = [0] * motif_length # Calculate the number of amino acids in each position col_num = 0 new_seqs = matrix_transposition(seqs) for col in new_seqs: for amino in col: i = col_num count_matrix[amino][i] += 1 col_num += 1 # Calculate the frequency of amino acids for amino in count_matrix: amino_sum = 0 for count in count_matrix[amino]: amino_sum += count frequency = amino_sum / (len(seqs) * motif_length) fre_matrix[amino] = frequency # Calculate weight matrix for i in range(0, motif_length): for amino in count_matrix: # Calculate WM by dividing it by the total nums of swqs count_divided_nums = count_matrix[amino][i] / len(seqs) if fre_matrix[amino] == 0: weight = 0 # Calculate WM by dividing it by the frequency of the aminos else: weight = count_divided_nums / fre_matrix[amino] # Logarithmic operation on matrix if weight == 0: weight = -10 else: weight = log(weight, 10) weight_matrix[amino][i] += float(format(weight, &#39;.4f&#39;)) # print(count_matrix, &#39;\n&#39;, fre_matrix, &#39;\n&#39;, weight_matrix, &#39;\n&#39;) return weight_matrix def get_random_seqs(protein_list): &quot;&quot;&quot;Randomly initialize the starting position&quot;&quot;&quot; train_seq = [] for i in range(0, trainNum): head = randint(0, 90) tail = head + motif_length seq = protein_list[i][head:tail] train_seq.append(seq) # print(train_seq) return train_seq def cal_score(seqs, weight_matrix): &quot;&quot;&quot;Calculate the score based on sequences and the WM&quot;&quot;&quot; score = 0 for i in range(0, motif_length): score += weight_matrix[seqs[i]][i] return score def get_train_seqs(protein_list, weight_matrix): &quot;&quot;&quot;Calculate scores os all the motifs with different starting position, select the optimal position &quot;&quot;&quot; max_score = [] train_seqs = [] max_index = [] for protein in protein_list: score_list = [] for i in range(0, 90): test_seq = protein[i:i + motif_length] # Calculate scores of different position in one protein score_list.append(cal_score(test_seq, weight_matrix)) # Get the max one max_position = score_list.index(max(score_list)) # Get the optimal sequences of one protein new_train_seq = protein[max_position:max_position + motif_length] # Get the best score to cal ave max_score.append(max(score_list)) train_seqs.append(new_train_seq) max_index.append(max_position) return train_seqs, max_index, max_score def data_training(protein_list): &quot;&quot;&quot;Train the data&quot;&quot;&quot; # Number of iterations iteration_num = 100 Best_ave = 0 Best_iteration = 0 for i in range(iteration_num): # Average score of every iteration max_score_list = [] # Initialize the position randomly first train_seq = get_random_seqs(protein_list) # Calculate PSSM matrix weight_matrix = PSSM_calculation(train_seq) for j in range(0, iteration_num): # Estimate and choose the best position new_train_seqs, max_index, max_score = get_train_seqs(protein_list, weight_matrix) # Calculate new matrix weight_matrix = PSSM_calculation(new_train_seqs) max_score_list.append(mean(max_score)) # Plot convergence generation = [] for k in range(1, iteration_num + 1): generation.append(k) max_score_list = [] plt.plot(generation, max_score_list) y_major_locator = MultipleLocator(0.3) ax = plt.gca() ax.yaxis.set_major_locator(y_major_locator) plt.ylabel(&#39;Mean scores&#39;, fontsize=10) plt.show() # Write the result to txt f = open(&quot;result.txt&quot;, &quot;a&quot;) f.write(&quot;\n&quot; + f&quot;Iteration {i}&quot; + &#39;\n&#39; + &quot;Starting positions:&quot; + str(max_index)) f.write(&#39;\n&#39; + &quot;PSSM:&quot; + str(weight_matrix)) f.write(&#39;\n&#39; + &quot;Scores:&quot; + str(max_score_list)) f.write(&#39;\n&#39; + &quot;Average best score:&quot; + str(mean(max_score))) # Get the max score if Best_ave &lt;= mean(max_score): Best_ave = mean(max_score) Best_iteration = i print(f&quot;Iteration {i}: Best average score = {Best_ave}&quot;) print(&#39;Best iteration: &#39;, Best_iteration) if __name__ == &#39;__main__&#39;: fl = open(&quot;result.txt&quot;, &quot;wt&quot;) fl.close() # Get the protein list proteinList = data_preprocess(proteinsDataset) data_training(proteinList) </code> </p>
<ul>
<li><h5 id="The-final-result"><a href="#The-final-result" class="headerlink" title="The final result:"></a><strong>The final result:</strong></h5></li>
</ul>
</blockquote>
<p><code>python D:\Anaconda3\envs\spider\python.exe Iteration 0: Best average score = 4.3729255 Best iteration: 0 Iteration 1: Best average score = 5.199039 Best iteration: 1 Iteration 2: Best average score = 5.199039 Best iteration: 1 Iteration 3: Best average score = 5.199039 Best iteration: 1 Iteration 4: Best average score = 6.680054999999999 Best iteration: 4 Iteration 5: Best average score = 6.680054999999999 Best iteration: 4 Iteration 6: Best average score = 6.680054999999999 Best iteration: 4 Iteration 7: Best average score = 6.680054999999999 Best iteration: 4 Iteration 8: Best average score = 6.680054999999999 Best iteration: 4 Iteration 9: Best average score = 6.680054999999999 Best iteration: 4 Iteration 10: Best average score = 6.680054999999999 Best iteration: 4 Iteration 11: Best average score = 6.680054999999999 Best iteration: 4 Iteration 12: Best average score = 6.680054999999999 Best iteration: 4 Iteration 13: Best average score = 6.680054999999999 Best iteration: 4 Iteration 14: Best average score = 6.680054999999999 Best iteration: 4 Iteration 15: Best average score = 6.680054999999999 Best iteration: 4 Iteration 16: Best average score = 6.680054999999999 Best iteration: 4 Iteration 17: Best average score = 6.680054999999999 Best iteration: 4 Iteration 18: Best average score = 6.680054999999999 Best iteration: 4 Iteration 19: Best average score = 6.680054999999999 Best iteration: 4 Iteration 20: Best average score = 6.680054999999999 Best iteration: 4 Iteration 21: Best average score = 6.680054999999999 Best iteration: 4 Iteration 22: Best average score = 6.680054999999999 Best iteration: 4 Iteration 23: Best average score = 6.680054999999999 Best iteration: 4 Iteration 24: Best average score = 6.680054999999999 Best iteration: 4 Iteration 25: Best average score = 6.680054999999999 Best iteration: 4 Iteration 26: Best average score = 6.680054999999999 Best iteration: 4 Iteration 27: Best average score = 6.680054999999999 Best iteration: 4 Iteration 28: Best average score = 6.680054999999999 Best iteration: 4 Iteration 29: Best average score = 6.680054999999999 Best iteration: 4 Iteration 30: Best average score = 6.680054999999999 Best iteration: 4 Iteration 31: Best average score = 6.680054999999999 Best iteration: 4 Iteration 32: Best average score = 6.832890500000001 Best iteration: 32 Iteration 33: Best average score = 6.832890500000001 Best iteration: 32 Iteration 34: Best average score = 6.832890500000001 Best iteration: 32 Iteration 35: Best average score = 6.832890500000001 Best iteration: 32 Iteration 36: Best average score = 6.832890500000001 Best iteration: 32 Iteration 37: Best average score = 6.855327999999999 Best iteration: 37 Iteration 38: Best average score = 6.870583999999999 Best iteration: 38 Iteration 39: Best average score = 6.870583999999999 Best iteration: 38 Iteration 40: Best average score = 6.9964825 Best iteration: 40 Iteration 41: Best average score = 6.9964825 Best iteration: 40 Iteration 42: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 43: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 44: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 45: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 46: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 47: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 48: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 49: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 50: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 51: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 52: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 53: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 54: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 55: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 56: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 57: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 58: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 59: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 60: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 61: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 62: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 63: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 64: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 65: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 66: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 67: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 68: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 69: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 70: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 71: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 72: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 73: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 74: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 75: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 76: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 77: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 78: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 79: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 80: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 81: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 82: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 83: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 84: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 85: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 86: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 87: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 88: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 89: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 90: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 91: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 92: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 93: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 94: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 95: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 96: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 97: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 98: Best average score = 7.2110034999999995 Best iteration: 42 Iteration 99: Best average score = 7.2110034999999995 Best iteration: 42 Process finished with exit code 0 </code> </p>
<blockquote>
<p>From the output, we can get:</p>
<p>Iteration 99: Best average score &#x3D; 7.2110034999999995<br> Best iteration: 42</p>
<p>So the best average score of 100 iterations is roughly 7.211, which iteration number is 42.</p>
<p>The <strong>Convergence Plot</strong> is shown below:</p>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/57f6ffdb711a.png"></p>
<p>The result to iteration 42 in result.txt is:</p>
<p><code>python Iteration 42 Starting positions:[40, 23, 0, 57, 8, 73, 49, 85, 29, 46, 82, 4, 49, 14, 32, 23, 78, 41, 0, 43, 69, 82, 76, 3, 25, 5, 36, 82, 34, 74, 30, 89, 15, 54, 42, 49, 84, 66, 71, 22, 23, 11, 51, 31, 77, 45, 30, 83, 48, 7, 80, 45, 60, 33, 84, 82, 86, 13, 79, 22, 21, 12, 68, 69, 81, 38, 68, 64, 89, 65, 47, 58, 53, 12, 56, 30, 12, 72, 16, 85, 57, 42, 86, 81, 30, 8, 6, 65, 32, 82, 24, 53, 11, 26, 15, 75, 44, 41, 56, 7, 27, 76, 34, 63, 46, 9, 32, 35, 13, 27, 88, 78, 86, 76, 26, 26, 74, 44, 51, 74, 29, 87, 46, 8, 64, 34, 13, 5, 57, 19, 80, 10, 68, 1, 28, 23, 33, 34, 56, 88, 63, 41, 57, 31, 30, 29, 8, 51, 26, 29, 31, 58, 69, 84, 47, 1, 55, 88, 34, 55, 65, 66, 7, 7, 89, 76, 60, 58, 37, 73, 84, 87, 24, 6, 18, 14, 83, 1, 60, 16, 68, 42, 31, 10, 47, 8, 57, 32, 52, 51, 32, 59, 64, 2, 72, 26, 66, 58, 7, 5] PSSM:{&#39;A&#39;: [0.9723, -1.2504, -0.9494, -1.2504, -10.0, -1.2504, -0.6484, -0.9494, -10.0, -10.0], &#39;C&#39;: [-1.5514, -1.5514, 0.6891, -1.2504, -1.5514, -10.0, -1.2504, 0.6841, -1.2504, -1.5514], &#39;D&#39;: [-0.0414, -0.3424, -0.3424, -10.0, 0.1347, 0.1347, 0.1347, -0.0414, 0.3565, -0.0414], &#39;E&#39;: [-0.0414, 0.2596, -0.0414, -10.0, -0.0414, -0.0414, -10.0, -10.0, 0.2596, 0.4357], &#39;F&#39;: [0.1549, 0.331, 0.5528, -10.0, -10.0, -0.1461, -0.1461, -10.0, 0.1549, -10.0], &#39;G&#39;: [-0.9661, -0.6651, -0.7901, -0.9661, -10.0, -1.2672, -10.0, -0.9661, 0.9658, -10.0], &#39;H&#39;: [-10.0, 0.0706, 0.2467, 0.2467, -10.0, -10.0, -0.2304, 0.3716, 0.0706, 0.0706], &#39;I&#39;: [-10.0, -10.0, 0.301, -10.0, 0.426, -10.0, -0.1761, 0.301, 0.1249, 0.1249], &#39;K&#39;: [-0.1761, 0.426, -10.0, -0.1761, 0.426, -0.1761, -0.1761, -10.0, 0.1249, -0.1761], &#39;L&#39;: [-1.2648, -1.2648, -1.2648, -1.2648, -10.0, 0.9807, -0.7877, -1.2648, -10.0, -10.0], &#39;M&#39;: [-10.0, -0.0792, -10.0, 0.6198, 0.2218, -10.0, 0.2218, -10.0, -0.0792, -0.0792], &#39;N&#39;: [0.5229, -10.0, -10.0, -0.0792, -10.0, -0.0792, -10.0, -0.0792, -0.0792, 0.5229], &#39;P&#39;: [-10.0, -10.0, -1.2492, 0.6977, -10.0, -1.5502, -1.5502, -10.0, -1.5502, 0.6878], &#39;Q&#39;: [0.301, -10.0, -0.1761, -10.0, -0.1761, 0.301, -0.1761, 0.1249, 0.1249, 0.1249], &#39;R&#39;: [-0.0212, 0.1549, -10.0, 0.2798, -0.3222, -0.0212, 0.4559, -0.0212, -10.0, -0.3222], &#39;S&#39;: [-0.7924, 0.9685, -10.0, -1.2695, -10.0, -0.9685, -0.7924, -0.6675, -10.0, -10.0], &#39;T&#39;: [-1.2553, -1.2553, -1.2553, -10.0, -0.9542, -10.0, 0.9726, -10.0, -0.9542, -0.6532], &#39;V&#39;: [-0.9845, -0.9845, -10.0, -1.2856, 0.9721, -0.9845, -0.9845, -1.2856, -0.9845, -10.0], &#39;W&#39;: [0.426, -0.1761, 0.301, -10.0, -10.0, 0.1249, -10.0, -0.1761, 0.301, -0.1761], &#39;Y&#39;: [0.4559, -10.0, -10.0, -0.1461, -10.0, 0.331, -10.0, 0.331, -10.0, 0.331]} Scores:[1.6438155, 7.475081, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999, 7.525479999999999] Average best score:7.2110034999999995 </code> </p>
<ul>
<li><h5 id="The-difficulties-I-met"><a href="#The-difficulties-I-met" class="headerlink" title="The difficulties I met:"></a><strong>The difficulties I met:</strong></h5></li>
</ul>
</blockquote>
<blockquote>
<ol>
<li>Determining suitable stopping criteria and determining when the algorithm has converged to a stable solution. In practice, heuristics like a maximum number of iterations are used.</li>
<li>Avoid getting stuck in poor local optima rather than the global optimum. Using multiple random restarts can mitigate this issue.</li>
<li>Runtime complexity for large datasets. Matrix computations may be intensive for large sequence sets and long motifs. Parallelization and approximation methods can improve scalability.</li>
<li>Interpreting the resulting motif model. Statistical significance testing should be done to evaluate if the identified motif represents a true common signal vs random shared patterns.</li>
</ol>
</blockquote>
<hr>
<h3 id="Machine-Learning-Approaches"><a href="#Machine-Learning-Approaches" class="headerlink" title="Machine Learning Approaches"></a>Machine Learning Approaches</h3><h4 id="GA-for-sequencing-and-MSA"><a href="#GA-for-sequencing-and-MSA" class="headerlink" title="GA for sequencing and MSA"></a>GA for sequencing and MSA</h4><h5 id="Sequencing-by-Using-GA"><a href="#Sequencing-by-Using-GA" class="headerlink" title="Sequencing by Using GA"></a>Sequencing by Using GA</h5><blockquote>
<p>GA â€“ Genetic Algorithmï¼ˆåŸºå› äº’æ¢ã€å˜å¼‚ï¼‰</p>
<p>EDA â€“ Estimation of Distribution Algorithm</p>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/0fc8221dff2d.png"></p>
<p>*Sequencing by Using GA</p>
<p><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/843d7478e4ff.png"></p>
</blockquote>
<h5 id="Multiple-Sequence-Alignments-Using-GA"><a href="#Multiple-Sequence-Alignments-Using-GA" class="headerlink" title="Multiple Sequence Alignments Using GA"></a>Multiple Sequence Alignments Using GA</h5><p>*Using Genetic Algorithms for Pairwise and Multiple Sequence Alignments</p>
<h4 id="Protein-Function-Prediction"><a href="#Protein-Function-Prediction" class="headerlink" title="Protein Function Prediction"></a>Protein Function Prediction</h4><h5 id="Gene-Function-Prediction-Using-SVM"><a href="#Gene-Function-Prediction-Using-SVM" class="headerlink" title="Gene Function Prediction Using SVM"></a>Gene Function Prediction Using SVM</h5><blockquote>
<p>æœ¬èŠ‚è®¨è®ºäº†ä½¿ç”¨SVMè¿›è¡ŒåŸºå› åŠŸèƒ½é¢„æµ‹çš„å¤šæ ‡ç­¾åˆ†ç±»å’Œå±‚æ¬¡å¤šæ ‡ç­¾åˆ†ç±»æ–¹æ³•ã€‚</p>
<p>ä¸»è¦å†…å®¹æœ‰:</p>
<p>1. åŸºäºæ ‡ç­¾æ’åºå’Œç²¾ç»†å†³ç­–è¾¹ç•ŒSVMçš„å¤šæ ‡ç­¾åˆ†ç±»æ–¹æ³•ã€‚è¯¥æ–¹æ³•è€ƒè™‘äº†æ ·æœ¬é‡å çš„ä¿¡æ¯,é€šè¿‡Biasæ¨¡å‹æ”¹è¿›é‡å æ ·æœ¬çš„åˆ†ç±»å‡†ç¡®ç‡;å¹¶æå‡ºäº†ä¸€ä¸ªåŸºäºè¾“å…¥å®ä¾‹å’Œæ ‡ç­¾æ’åºçš„å®ä¾‹ç›¸å…³é˜ˆå€¼é€‰å–ç­–ç•¥ã€‚</p>
<p>2. åŸºäºè¿‡é‡‡æ ·å’Œå±‚æ¬¡çº¦æŸçš„å±‚æ¬¡å¤šæ ‡ç­¾åˆ†ç±»æ–¹æ³•ã€‚è¯¥æ–¹æ³•é€šè¿‡æå‡ºåˆ†å±‚SMOTEæ–¹æ³•è§£å†³æ•°æ®é›†ä¸å¹³è¡¡é—®é¢˜;å¹¶é€šè¿‡åœ¨TPRè§„åˆ™é›†æˆä¸­å¼•å…¥åˆ†ç±»å™¨æ€§èƒ½æƒé‡æ¥æ”¹è¿›é›†æˆ,é™åˆ¶å·®åˆ†ç±»å™¨çš„é”™è¯¯ä¼ æ’­æ•ˆåº”ã€‚</p>
<p>3. åœ¨Yeastå’ŒGenbaseæ•°æ®é›†ä¸Šçš„è¯•éªŒè¡¨æ˜,ä¸Šè¿°ä¸¤ç§æ–¹æ³•åœ¨åŸºå› åŠŸèƒ½é¢„æµ‹ä»»åŠ¡ä¸Šéƒ½èƒ½å–å¾—ä¸é”™çš„æ€§èƒ½ã€‚</p>
<p>4. è¿™äº›æ–¹æ³•ä¸ºè§£å†³åŸºå› æµ‹åºæ•°æ®çš„åŠŸèƒ½æ ‡æ³¨ä¸åˆ†ç±»é—®é¢˜æä¾›äº†æœ‰æ•ˆçš„æœºå™¨å­¦ä¹ æ¡†æ¶ã€‚</p>
</blockquote>
<h5 id="Protein-Function-Prediction-Using-DNN"><a href="#Protein-Function-Prediction-Using-DNN" class="headerlink" title="Protein Function Prediction Using DNN"></a>Protein Function Prediction Using DNN</h5><blockquote>
<p>æœ¬èŠ‚ä¸»è¦å†…å®¹æ˜¯å…³äºä½¿ç”¨æ·±åº¦è¿ç§»å­¦ä¹ æ–¹æ³•è¿›è¡Œè›‹ç™½è´¨åŠŸèƒ½é¢„æµ‹çš„ç ”ç©¶ã€‚</p>
<p>ä¸»è¦æœ‰ä»¥ä¸‹å‡ ä¸ªæ–¹é¢:</p>
<p>1. ä½¿ç”¨å¤šå¤´å¤šå°¾æ·±åº¦å·ç§¯ç¥ç»ç½‘ç»œæ¨¡å‹è¿›è¡ŒåŸºå› æœ¬ä½“è®º(GO)æ³¨é‡Šé¢„æµ‹ã€‚è¯¥æ¨¡å‹å¯ä»¥å®ç°çŸ¥è¯†è¿ç§»,æé«˜æ¨¡å‹æ€§èƒ½ã€‚</p>
<p>2. é€šè¿‡GOæ³¨é‡Šé¢„æµ‹çš„è¿ç§»å­¦ä¹ æ¥å¢å¼ºäºšç»†èƒå®šä½é¢„æµ‹ã€‚å…±äº«ç‰¹å¾æå–å™¨,å®ç°é¢„è®­ç»ƒå’Œå¾®è°ƒã€‚</p>
<p>3. åŸºäºè¿ç§»å­¦ä¹ çš„è›‹ç™½è´¨ç›¸äº’ä½œç”¨é¢„æµ‹,ç”¨äºå¤åˆç‰©æ£€æµ‹ã€‚è¿ç§»æ¨¡å‹ä»GOå’Œå®šä½æ³¨é‡Šä»»åŠ¡ä¸­å­¦ä¹ çŸ¥è¯†ã€‚</p>
<p>4. åœ¨GOæ³¨é‡Šã€å®šä½å’Œç›¸äº’ä½œç”¨é¢„æµ‹ä»»åŠ¡ä¹‹é—´å®ç°äº†è¿ç§»å­¦ä¹ ã€‚æå‡ºçš„æ¨¡å‹å¯ä»¥æ‰©å±•åˆ°å…¶ä»–ç”Ÿç‰©ä¿¡æ¯å­¦ä»»åŠ¡ã€‚</p>
<p>5. è¯•éªŒç»“æœè¡¨æ˜,è¿ç§»å­¦ä¹ å¯ä»¥æœ‰æ•ˆæ”¹è¿›è›‹ç™½è´¨åŠŸèƒ½é¢„æµ‹çš„æ€§èƒ½ã€‚</p>
<p>æ€»ä½“è€Œè¨€,æœ¬èŠ‚ç ”ç©¶äº†å¦‚ä½•åˆ©ç”¨æ·±åº¦å­¦ä¹ å’Œè¿ç§»å­¦ä¹ çš„æ–¹æ³•,æ”¹è¿›è›‹ç™½è´¨åŠŸèƒ½é¢„æµ‹ä¸æ ‡æ³¨çš„æ€§èƒ½ã€‚</p>
</blockquote>
<h4 id="Protein-Structure-Prediction"><a href="#Protein-Structure-Prediction" class="headerlink" title="Protein Structure Prediction"></a>Protein Structure Prediction</h4><h5 id="Protein-Folding-Based-on-EDA"><a href="#Protein-Folding-Based-on-EDA" class="headerlink" title="Protein Folding Based on EDA"></a>Protein Folding Based on EDA</h5><blockquote>
<p>æœ¬èŠ‚ä¸»è¦å†…å®¹æ˜¯å…³äºä½¿ç”¨åŸºäºEDAçš„æ··åˆæ™ºèƒ½ç®—æ³•è¿›è¡Œè›‹ç™½è´¨ç»“æ„é¢„æµ‹çš„ç ”ç©¶ã€‚</p>
<p>ä¸»è¦åŒ…å«ä»¥ä¸‹å‡ ä¸ªæ–¹é¢:</p>
<p>1. ä½¿ç”¨æ··åˆEDAç®—æ³•æ±‚è§£ç®€åŒ–çš„HPæ¨¡å‹è›‹ç™½è´¨æŠ˜å é—®é¢˜ã€‚æå‡ºå¤åˆé€‚åº”åº¦å‡½æ•°ã€å¼•å¯¼ç®—å­å±€éƒ¨æœç´¢å’Œæ”¹è¿›çš„å›æº¯ä¿®å¤æ–¹æ³•ã€‚</p>
<p>2. æå‡ºè‡ªé€‚åº”åˆ†ç¾¤EDAç®—æ³•,ä½¿ç”¨äº²å’Œä¼ æ’­èšç±»å®ç°è‡ªé€‚åº”åˆ†ç¾¤,å¹¶ä½¿ç”¨ç»å°”å…¹æ›¼æœºåˆ¶å®ç°å¹³è¡¡æœç´¢ã€‚åº”ç”¨äºè¿ç»­ä¼˜åŒ–å’ŒHPæ¨¡å‹è›‹ç™½è´¨ç»“æ„é¢„æµ‹é—®é¢˜ã€‚</p>
<p>3. è¯•éªŒç»“æœè¡¨æ˜,æ‰€æå‡ºçš„æ··åˆEDAå’Œè‡ªé€‚åº”åˆ†ç¾¤EDAç®—æ³•éƒ½å¯ä»¥æœ‰æ•ˆæé«˜è›‹ç™½è´¨ç»“æ„é¢„æµ‹çš„æ€§èƒ½ã€‚</p>
<p>4. EDAç®—æ³•ä¸ç‰¹å®šé—®é¢˜ä¿¡æ¯çš„ç»“åˆä¸ºè§£å†³è›‹ç™½è´¨ç»“æ„é¢„æµ‹ç­‰å¤æ‚ä¼˜åŒ–é—®é¢˜æä¾›äº†ä¸€ç§æœ‰æ•ˆé€”å¾„ã€‚</p>
<p>æ€»çš„æ¥è¯´, å®ƒæ¢è®¨äº†EDAç®—æ³•åœ¨è›‹ç™½è´¨ç»“æ„é¢„æµ‹é—®é¢˜ä¸Šçš„åº”ç”¨,ä»¥åŠå¦‚ä½•è®¾è®¡å‡ºæ€§èƒ½æ›´ä¼˜çš„æ··åˆEDAå’Œè‡ªé€‚åº”EDAç®—æ³•ã€‚</p>
</blockquote>
<h5 id="Protein-Folding-Based-on-Deep-Reinforcement-Learning"><a href="#Protein-Folding-Based-on-Deep-Reinforcement-Learning" class="headerlink" title="Protein Folding Based on Deep Reinforcement Learning"></a>Protein Folding Based on Deep Reinforcement Learning</h5><blockquote>
<p>æœ¬èŠ‚æ˜¯å…³äºè›‹ç™½è´¨æŠ˜å é—®é¢˜çš„ç ”ç©¶æŠ¥å‘Šã€‚ä¸»è¦å†…å®¹åŒ…æ‹¬:</p>
<p>1. ç®€ä»‹è›‹ç™½è´¨æŠ˜å é—®é¢˜å’ŒHPæ¨¡å‹<br> 2. æ·±åº¦å¼ºåŒ–å­¦ä¹ æ–¹æ³•ç”¨äºè§£å†³HPæ¨¡å‹ä¸­çš„è›‹ç™½è´¨æŠ˜å é—®é¢˜<br> 3. AlphaGoç³»åˆ—ä¸­è’™ç‰¹å¡æ´›æ ‘æœç´¢(MCTS)çš„åº”ç”¨<br> 4. å°†MCTSåº”ç”¨äºHPæ¨¡å‹ä»¥è§£å†³è›‹ç™½è´¨æŠ˜å é—®é¢˜<br> 5. è›‹ç™½è´¨ç»“æ„é¢„æµ‹çš„æœ€æ–°è¿›å±•,å¦‚AlphaFold</p>
<p>æŠ¥å‘Šä»‹ç»äº†å¦‚ä½•é€šè¿‡æ·±åº¦å¼ºåŒ–å­¦ä¹ å’Œè’™ç‰¹å¡æ´›æ ‘æœç´¢ç­‰æ–¹æ³•æ¥é¢„æµ‹è›‹ç™½è´¨åºåˆ—çš„ä¸‰ç»´ç»“æ„,è¿™æ˜¯è›‹ç™½è´¨æŠ˜å é—®é¢˜çš„æ ¸å¿ƒã€‚æ€»çš„æ¥è¯´,è¿™ç¯‡æŠ¥å‘Šèšç„¦åœ¨åˆ©ç”¨AIæŠ€æœ¯æ¥è§£å†³è›‹ç™½è´¨ç»“æ„é¢„æµ‹è¿™ä¸ªåŸºç¡€ç§‘å­¦éš¾é¢˜ä¸Šã€‚</p>
</blockquote>
<h5 id="AlphaFold2-for-Protein-Structure-Prediction"><a href="#AlphaFold2-for-Protein-Structure-Prediction" class="headerlink" title="AlphaFold2 for Protein Structure Prediction"></a>AlphaFold2 for Protein Structure Prediction</h5><blockquote>
<p>æœ¬èŠ‚æ˜¯å…³äºAlphaFold2è›‹ç™½è´¨ç»“æ„é¢„æµ‹æ–¹æ³•çš„ä»‹ç»å’Œåˆ†ææŠ¥å‘Šã€‚ä¸»è¦å†…å®¹åŒ…æ‹¬:</p>
<p>1. è›‹ç™½è´¨æŠ˜å é—®é¢˜çš„èƒŒæ™¯ç®€ä»‹<br> 2. AlphaFold2çš„æ•´ä½“æµç¨‹æ¦‚è§ˆ,åŒ…æ‹¬æ•°æ®å‡†å¤‡ã€åµŒå…¥ã€Evoformeræ¨¡å—ã€ç»“æ„æ¨¡å—ç­‰<br> 3. AlphaFold2çš„å…·ä½“æŠ€æœ¯ç»†èŠ‚,å¦‚å¤šåºåˆ—æ¯”å¯¹ã€è¡¨å¾æå–ã€æ³¨æ„åŠ›æœºåˆ¶ã€æŸå¤±å‡½æ•°ç­‰<br> 4. AlphaFold2é¢„æµ‹è›‹ç™½è´¨ç»“æ„çš„åº”ç”¨,å¦‚è›‹ç™½è´¨ç»“æ„æ•°æ®åº“</p>
<p>ç®€è€Œè¨€ä¹‹,è¿™æ˜¯ä¸€ç¯‡ä»‹ç»å’Œåˆ†æAlphaFold2åœ¨è›‹ç™½è´¨ç»“æ„é¢„æµ‹é¢†åŸŸé‡Œç¨‹ç¢‘å¼çš„è¿›å±•çš„æŠ¥å‘Šã€‚å®ƒæ¦‚è¿°äº†è¿™ä¸€æœ€å…ˆè¿›çš„AIç³»ç»Ÿæ˜¯å¦‚ä½•è§£å†³è¿™ä¸€å›°æ‰°ç”Ÿç‰©å­¦å®¶50å¹´ä¹‹ä¹…çš„éš¾é¢˜çš„ã€‚æ€»çš„æ¥è¯´,è¿™ç¯‡æŠ¥å‘Šèšç„¦åœ¨åˆ©ç”¨æ·±åº¦å­¦ä¹ æŠ€æœ¯æ¥æ¨åŠ¨åŸºç¡€ç§‘å­¦å‘å±•æ–¹é¢çš„å†…å®¹ã€‚</p>
</blockquote>
<hr>
<h3 id="References"><a href="#References" class="headerlink" title="References"></a>References</h3><ul>
<li>EMBL-EBI, <a target="_blank" rel="noopener" href="https://www2.hflab.ips.waseda.ac.jp/~jinglu/Courses/11300ABio/EBI%20Research%20-%20Functional%20Genomics%20-%20Introduction%20To%20Biology.mht" title="A Quick Introduction to Elements of Biology">A Quick Introduction to Elements of Biology</a> - Cells, Molecules, Genes, Functional Genomics, Microarrays</li>
<li>DOE Human Genome Program:  <em><a target="_blank" rel="noopener" href="https://www2.hflab.ips.waseda.ac.jp/~jinglu/Courses/11300ABio/Primer%20on%20Molecular%20Genetics%20%28DOE%20Human%20Genome%20Program,%201992%29.pdf" title="Primer on Molecular Genetics">Primer on Molecular Genetics</a></em> , 1992</li>
<li>Jin Xiong, <a target="_blank" rel="noopener" href="https://www2.hflab.ips.waseda.ac.jp/~jinglu/Courses/11300ABio/Essential%20Bioinformation%20%28Jin%20Xiong,%20Cambridge%20University%20Press,%202006%29.pdf" title="Essential Bioinformatics">Essential Bioinformatics</a>, Cambridge University Press, 2006</li>
<li>P. Larranaga et.al., â€œ<a target="_blank" rel="noopener" href="https://www2.hflab.ips.waseda.ac.jp/~jinglu/Courses/11300ABio/Machine%20Learning%20in%20Bioinformatics,%20Briefings%20in%20Bioinformatics,%207%281%29,%2086-112.pdf" title="Machine Learning in Bioinformatics">Machine Learning in Bioinformatics</a>â€œ,  <em>Briefings in Bioinformatics</em> , 7(1), 86-112, 2006</li>
<li>J.M. Claverie and C. Notredame,  <em><a target="_blank" rel="noopener" href="https://www2.hflab.ips.waseda.ac.jp/~jinglu/Courses/11300ABio/Bioinformatics.For.Dummies.2nd.Ed%20%28for%20Dummies,%202007%29.pdf" title="Bioinformatics for Dummies, 2nd Ed.">Bioinformatics for Dummies, 2nd Ed.</a></em> , Wiley Publishing Inc, 2007</li>
<li>B. Alberts et.al.,  <em><a target="_blank" rel="noopener" href="https://www2.hflab.ips.waseda.ac.jp/~jinglu/Courses/11300ABio/Molecular%20Biology%20of%20The%20Cell%205th%20Ed%20%28Garland%20Science,%202008%29.pdf" title="Molecular Biology of THE CELL, 5th Ed.">Molecular Biology of THE CELL, 5th Ed.</a></em> , Garland Science, 2008</li>
<li>N.C. Jones and P.A. Pevzner,  <em><a target="_blank" rel="noopener" href="https://www2.hflab.ips.waseda.ac.jp/~jinglu/Courses/11300ABio/An%20Introduction%20to%20Bioinformatics%20Algorithms%20%28Neil%20C.%20Jones%20and%20Pavel%20A.%20Pevzner,%20The%20MIT%20Press,%202004%29.pdf" title="An Introduction to Bioinformatics Algorithms">An Introduction to Bioinformatics Algorithms</a></em> , The MIT Press, 2004</li>
<li>DNAFTB: <a target="_blank" rel="noopener" href="http://dnaftb.org/" title="DNA from the Beginning">DNA from the Beginning</a> (An animated primer of 75 experiments that made modern genetics)</li>
</ul>
<hr>
</article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta"><i class="fas fa-circle-user fa-fw"></i>Author: </span><span class="post-copyright-info"><a target="_blank" rel="noopener" href="https://github.com/veckun">Likun Wang (ç‹ç«‹å¤)</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta"><i class="fas fa-square-arrow-up-right fa-fw"></i>Link: </span><span class="post-copyright-info"><a href="https://your-domain.com/2024/01/20/en/Summerize-for-Bioinformatics-with-ChatGPT/">https://your-domain.com/2024/01/20/en/Summerize-for-Bioinformatics-with-ChatGPT/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta"><i class="fas fa-circle-exclamation fa-fw"></i>Copyright Notice: </span><span class="post-copyright-info">All articles on this blog are licensed under <a target="_blank" rel="noopener" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a> unless otherwise stated.</span></div></div><div class="tag_share"><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/%E5%A4%8D%E4%B9%A0%E8%B5%84%E6%96%99/">å¤ä¹ èµ„æ–™</a><a class="post-meta__tags" href="/tags/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6/">ç”Ÿç‰©ä¿¡æ¯å­¦</a><a class="post-meta__tags" href="/tags/Bioinformatics/">Bioinformatics</a><a class="post-meta__tags" href="/tags/%E5%90%AF%E5%8F%91%E5%BC%8F%E7%AE%97%E6%B3%95/">å¯å‘å¼ç®—æ³•</a><a class="post-meta__tags" href="/tags/PSSM%E6%96%B9%E6%B3%95/">PSSMæ–¹æ³•</a><a class="post-meta__tags" href="/tags/GA/">GA</a><a class="post-meta__tags" href="/tags/BLAST/">BLAST</a></div><div class="post-share"><div class="social-share" data-image="/images/posts/Summerize-for-Bioinformatics-w/517441f24ba6.png" data-sites="wechat,weibo,qq,twitter,facebook"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1.1.6/sharejs/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1.1.6/sharejs/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><a class="pagination-related" href="/2024/01/20/zh-CN/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6%E4%B8%8EChatGPT%E6%80%BB%E7%BB%93/" title="ç”Ÿç‰©ä¿¡æ¯å­¦ä¸ChatGPTæ€»ç»“"><img class="cover" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/517441f24ba6.png" onerror="onerror=null;src='/img/404.jpg'" alt="cover of previous post"><div class="info"><div class="info-1"><div class="info-item-1">Previous</div><div class="info-item-2">ç”Ÿç‰©ä¿¡æ¯å­¦ä¸ChatGPTæ€»ç»“</div></div><div class="info-2"><div class="info-item-1"> æœ¬æ–‡ä¸ºè‹±æ–‡åŸæ–‡çš„ä¸­æ–‡ç¿»è¯‘ç‰ˆæœ¬è‹±æ–‡åŸæ–‡ï¼šSummerize for Bioinformatics with ChatGPT  åŸºç¡€ä»‹ç»å†å²äº‹ä»¶ç”Ÿç‰©ä¿¡æ¯å­¦ vs. è®¡ç®—ç”Ÿç‰©å­¦ ç”Ÿç‰©ä¿¡æ¯å­¦ &#x3D; åˆ›å»ºè§£å†³é—®é¢˜çš„å·¥å…·ï¼ˆç®—æ³•ã€æ•°æ®åº“ï¼‰ã€‚ç›®æ ‡æ˜¯æ„å»ºå¤„ç†ç”Ÿç‰©æ•°æ®çš„æœ‰ç”¨å·¥å…·ã€‚è¿™æ˜¯å…³äºå·¥ç¨‹çš„ã€‚ è®¡ç®—ç”Ÿç‰©å­¦ &#x3D; ä½¿ç”¨è®¡ç®—æŠ€æœ¯ç ”ç©¶ç”Ÿç‰©å­¦ã€‚ç›®æ ‡æ˜¯å­¦ä¹ å…³äºç”Ÿå‘½ç³»ç»Ÿçš„æ–°ç”Ÿç‰©å­¦çŸ¥è¯†ã€‚è¿™æ˜¯å…³äºå‘ç°çš„ã€‚  ç”Ÿç‰©ä¿¡æ¯å­¦å’Œè®¡ç®—ç”Ÿç‰©å­¦çš„å±‚çº§ ç¬¬0çº§ï¼šä¸ºå»ºæ¨¡è€Œå»ºæ¨¡ ç¬¬1çº§ï¼ˆå…¥é—¨ï¼‰ï¼šä½¿ç”¨å·²å‘å¸ƒçš„å·¥å…·åˆ†ææ•°æ®å¹¶ä¸ºå®éªŒäººå‘˜ç”Ÿæˆå‡è®¾ ç¬¬2çº§ï¼ˆç”Ÿç‰©ä¿¡æ¯å­¦ï¼‰ï¼šä¸ºæ–°æŠ€æœ¯ã€æ•°æ®é›†æˆå’Œé‡ç”¨å¼€å‘ç®—æ³•å’Œæ•°æ®åº“è¿›è¡Œæ•°æ®åˆ†æ ç¬¬3çº§ï¼ˆè®¡ç®—ç”Ÿç‰©å­¦ï¼‰ï¼šä»å…¬å…±æ•°æ®é›†æˆå’Œå»ºæ¨¡ä¸­åšå‡ºç”Ÿç‰©å­¦å‘ç° ç¬¬Xçº§ï¼šæ¥è‡ªå¤§å‹è”ç›Ÿçš„ç»¼åˆç ”ç©¶   åˆ†å­ç”Ÿç‰©å­¦å…¥é—¨ æ‰€æœ‰ç”Ÿå‘½ä¾èµ–äº3ç§å…³é”®åˆ†å­  DNA ä¿å­˜ç»†èƒå¦‚ä½•å·¥ä½œçš„ä¿¡æ¯   RNA å°†çŸ­ç‰‡æ®µä¿¡æ¯ä¼ é€’åˆ°ç»†èƒçš„ä¸åŒéƒ¨åˆ† æä¾›åˆæˆè›‹ç™½è´¨çš„æ¨¡æ¿   è›‹ç™½è´¨ å½¢æˆå‘å…¶ä»–ç»†èƒå‘é€ä¿¡å·å¹¶è°ƒèŠ‚åŸºå› æ´»åŠ¨çš„é…¶ å½¢æˆèº«ä½“çš„ä¸»è¦æˆåˆ†ï¼ˆå¦‚å¤´å‘ã€çš®è‚¤ç­‰ï¼‰ è›‹ç™½è´¨åœ¨ç”Ÿå‘½ç³»ç»Ÿä¸­çš„é‡è¦ä½œç”¨ï¼š å‚¬åŒ–ä½œç”¨ è½½ä½“å’Œè¿è¾“ è°ƒèŠ‚        ...</div></div></div></a><a class="pagination-related" href="/2024/01/18/zh-CN/%E9%9B%86%E6%88%90%E7%B3%BB%E7%BB%9F%E5%B0%81%E8%A3%85%E4%B8%8E%E7%BB%84%E8%A3%85%E6%8A%80%E6%9C%AF%E6%80%BB%E7%BB%93/" title="é›†æˆç³»ç»Ÿå°è£…ä¸ç»„è£…æŠ€æœ¯æ€»ç»“"><img class="cover" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summary-for-Packaging-and-Asse/8871f973cd33.png" onerror="onerror=null;src='/img/404.jpg'" alt="cover of next post"><div class="info text-right"><div class="info-1"><div class="info-item-1">Next</div><div class="info-item-2">é›†æˆç³»ç»Ÿå°è£…ä¸ç»„è£…æŠ€æœ¯æ€»ç»“</div></div><div class="info-2"><div class="info-item-1"> æœ¬æ–‡ä¸ºè‹±æ–‡åŸæ–‡çš„ä¸­æ–‡ç¿»è¯‘ç‰ˆæœ¬è‹±æ–‡åŸæ–‡ï¼šSummary for Packaging and Assembly Technologies for Integrated Systems  å¼•è¨€å°è£…ä¸ç»„è£…å±‚çº§ï¼š   è£¸èŠ¯ç‰‡å±‚çº§ï¼šæ™¶ä½“ç®¡ã€äºŒæç®¡ã€ç”µé˜»ã€ç”µå®¹ï¼ˆåˆ†ç«‹å…ƒä»¶ï¼‰å’Œé›†æˆç”µè·¯ï¼ˆICï¼‰çš„è£¸èŠ¯ç‰‡ å°è£…å±‚çº§ï¼šç¬¬ä¸€å±‚çº§è£¸èŠ¯ç‰‡çš„å°è£…å™¨ä»¶ å°åˆ·ç”µè·¯æ¿å±‚çº§ï¼šè¿æ¥ä¸¤ä¸ªæˆ–æ›´å¤šç¬¬äºŒå±‚çº§å°è£…å™¨ä»¶çš„å°åˆ·ç”µè·¯æ¿ æ•´ä½“ç³»ç»Ÿå±‚çº§ï¼šåŒ…å«å¤šä¸ªç¬¬ä¸‰å±‚çº§å°åˆ·ç”µè·¯æ¿åŠå…¶å¤–å£³çš„å®Œæ•´ç³»ç»Ÿ   è®¨è®ºä¸»é¢˜ï¼š  é›†æˆå™¨ä»¶çš„å°è£…æŠ€æœ¯ ç”µè·¯æ¿æŠ€æœ¯ ç»„è£…æŠ€æœ¯ å¯é æ€§è¯„ä¼°æŠ€æœ¯ æ•°å€¼åˆ†æè¯„ä¼°æŠ€æœ¯  å°è£…ç±»å‹  é›†æˆå™¨ä»¶çš„å°è£…é—®é¢˜1ï¼šå›ç­”å…³äºSIPã€ZIPå’ŒDIPå°è£…ç±»å‹çš„ä»¥ä¸‹é—®é¢˜ã€‚ 1. ä¸ZIPç›¸æ¯”ï¼ŒSIPçš„ä¼˜ç¼ºç‚¹æ˜¯ä»€ä¹ˆï¼Ÿ  SIPçš„ä¼˜ç‚¹ï¼šç»“æ„æ¯”ZIPç®€å•ï¼Œå› æ­¤åˆ¶é€ æˆæœ¬ä½ä¸”æ›´å¯é ã€‚ SIPçš„ç¼ºç‚¹ï¼šå®‰è£…å¯†åº¦ä½äºZIPã€‚  2. ä¸DIPç›¸æ¯”ï¼ŒSIPçš„ä¼˜ç¼ºç‚¹æ˜¯ä»€ä¹ˆï¼Ÿ  SIPçš„ä¼˜ç‚¹ï¼šç»“æ„æ¯”DIPç®€å•ï¼Œå› æ­¤åˆ¶é€ æˆæœ¬ä½ä¸”æ›´å¯é ã€‚ SIPçš„ç¼ºç‚¹ï¼šå®‰è£…å¯†åº¦ä½äºDIPã€‚  é—®é¢˜2ï¼šå›ç­”å…³äºSOPã€SOJå’ŒSONå°è£…ç±»å‹çš„ä»¥ä¸‹é—®é¢˜ã€‚ 1. ä¸SOJç›¸æ¯”ï¼ŒSO...</div></div></div></a></nav><div class="relatedPosts"><div class="headline"><i class="fas fa-thumbs-up fa-fw"></i><span>Related Articles</span></div><div class="relatedPosts-list"><a class="pagination-related" href="/2024/01/20/zh-CN/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6%E4%B8%8EChatGPT%E6%80%BB%E7%BB%93/" title="ç”Ÿç‰©ä¿¡æ¯å­¦ä¸ChatGPTæ€»ç»“"><img class="cover" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summerize-for-Bioinformatics-w/517441f24ba6.png" alt="cover"><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2024-01-21</div><div class="info-item-2">ç”Ÿç‰©ä¿¡æ¯å­¦ä¸ChatGPTæ€»ç»“</div></div><div class="info-2"><div class="info-item-1"> æœ¬æ–‡ä¸ºè‹±æ–‡åŸæ–‡çš„ä¸­æ–‡ç¿»è¯‘ç‰ˆæœ¬è‹±æ–‡åŸæ–‡ï¼šSummerize for Bioinformatics with ChatGPT  åŸºç¡€ä»‹ç»å†å²äº‹ä»¶ç”Ÿç‰©ä¿¡æ¯å­¦ vs. è®¡ç®—ç”Ÿç‰©å­¦ ç”Ÿç‰©ä¿¡æ¯å­¦ &#x3D; åˆ›å»ºè§£å†³é—®é¢˜çš„å·¥å…·ï¼ˆç®—æ³•ã€æ•°æ®åº“ï¼‰ã€‚ç›®æ ‡æ˜¯æ„å»ºå¤„ç†ç”Ÿç‰©æ•°æ®çš„æœ‰ç”¨å·¥å…·ã€‚è¿™æ˜¯å…³äºå·¥ç¨‹çš„ã€‚ è®¡ç®—ç”Ÿç‰©å­¦ &#x3D; ä½¿ç”¨è®¡ç®—æŠ€æœ¯ç ”ç©¶ç”Ÿç‰©å­¦ã€‚ç›®æ ‡æ˜¯å­¦ä¹ å…³äºç”Ÿå‘½ç³»ç»Ÿçš„æ–°ç”Ÿç‰©å­¦çŸ¥è¯†ã€‚è¿™æ˜¯å…³äºå‘ç°çš„ã€‚  ç”Ÿç‰©ä¿¡æ¯å­¦å’Œè®¡ç®—ç”Ÿç‰©å­¦çš„å±‚çº§ ç¬¬0çº§ï¼šä¸ºå»ºæ¨¡è€Œå»ºæ¨¡ ç¬¬1çº§ï¼ˆå…¥é—¨ï¼‰ï¼šä½¿ç”¨å·²å‘å¸ƒçš„å·¥å…·åˆ†ææ•°æ®å¹¶ä¸ºå®éªŒäººå‘˜ç”Ÿæˆå‡è®¾ ç¬¬2çº§ï¼ˆç”Ÿç‰©ä¿¡æ¯å­¦ï¼‰ï¼šä¸ºæ–°æŠ€æœ¯ã€æ•°æ®é›†æˆå’Œé‡ç”¨å¼€å‘ç®—æ³•å’Œæ•°æ®åº“è¿›è¡Œæ•°æ®åˆ†æ ç¬¬3çº§ï¼ˆè®¡ç®—ç”Ÿç‰©å­¦ï¼‰ï¼šä»å…¬å…±æ•°æ®é›†æˆå’Œå»ºæ¨¡ä¸­åšå‡ºç”Ÿç‰©å­¦å‘ç° ç¬¬Xçº§ï¼šæ¥è‡ªå¤§å‹è”ç›Ÿçš„ç»¼åˆç ”ç©¶   åˆ†å­ç”Ÿç‰©å­¦å…¥é—¨ æ‰€æœ‰ç”Ÿå‘½ä¾èµ–äº3ç§å…³é”®åˆ†å­  DNA ä¿å­˜ç»†èƒå¦‚ä½•å·¥ä½œçš„ä¿¡æ¯   RNA å°†çŸ­ç‰‡æ®µä¿¡æ¯ä¼ é€’åˆ°ç»†èƒçš„ä¸åŒéƒ¨åˆ† æä¾›åˆæˆè›‹ç™½è´¨çš„æ¨¡æ¿   è›‹ç™½è´¨ å½¢æˆå‘å…¶ä»–ç»†èƒå‘é€ä¿¡å·å¹¶è°ƒèŠ‚åŸºå› æ´»åŠ¨çš„é…¶ å½¢æˆèº«ä½“çš„ä¸»è¦æˆåˆ†ï¼ˆå¦‚å¤´å‘ã€çš®è‚¤ç­‰ï¼‰ è›‹ç™½è´¨åœ¨ç”Ÿå‘½ç³»ç»Ÿä¸­çš„é‡è¦ä½œç”¨ï¼š å‚¬åŒ–ä½œç”¨ è½½ä½“å’Œè¿è¾“ è°ƒèŠ‚        ...</div></div></div></a><a class="pagination-related" href="/2024/01/18/zh-CN/%E9%9B%86%E6%88%90%E7%B3%BB%E7%BB%9F%E5%B0%81%E8%A3%85%E4%B8%8E%E7%BB%84%E8%A3%85%E6%8A%80%E6%9C%AF%E6%80%BB%E7%BB%93/" title="é›†æˆç³»ç»Ÿå°è£…ä¸ç»„è£…æŠ€æœ¯æ€»ç»“"><img class="cover" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summary-for-Packaging-and-Asse/8871f973cd33.png" alt="cover"><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2024-01-19</div><div class="info-item-2">é›†æˆç³»ç»Ÿå°è£…ä¸ç»„è£…æŠ€æœ¯æ€»ç»“</div></div><div class="info-2"><div class="info-item-1"> æœ¬æ–‡ä¸ºè‹±æ–‡åŸæ–‡çš„ä¸­æ–‡ç¿»è¯‘ç‰ˆæœ¬è‹±æ–‡åŸæ–‡ï¼šSummary for Packaging and Assembly Technologies for Integrated Systems  å¼•è¨€å°è£…ä¸ç»„è£…å±‚çº§ï¼š   è£¸èŠ¯ç‰‡å±‚çº§ï¼šæ™¶ä½“ç®¡ã€äºŒæç®¡ã€ç”µé˜»ã€ç”µå®¹ï¼ˆåˆ†ç«‹å…ƒä»¶ï¼‰å’Œé›†æˆç”µè·¯ï¼ˆICï¼‰çš„è£¸èŠ¯ç‰‡ å°è£…å±‚çº§ï¼šç¬¬ä¸€å±‚çº§è£¸èŠ¯ç‰‡çš„å°è£…å™¨ä»¶ å°åˆ·ç”µè·¯æ¿å±‚çº§ï¼šè¿æ¥ä¸¤ä¸ªæˆ–æ›´å¤šç¬¬äºŒå±‚çº§å°è£…å™¨ä»¶çš„å°åˆ·ç”µè·¯æ¿ æ•´ä½“ç³»ç»Ÿå±‚çº§ï¼šåŒ…å«å¤šä¸ªç¬¬ä¸‰å±‚çº§å°åˆ·ç”µè·¯æ¿åŠå…¶å¤–å£³çš„å®Œæ•´ç³»ç»Ÿ   è®¨è®ºä¸»é¢˜ï¼š  é›†æˆå™¨ä»¶çš„å°è£…æŠ€æœ¯ ç”µè·¯æ¿æŠ€æœ¯ ç»„è£…æŠ€æœ¯ å¯é æ€§è¯„ä¼°æŠ€æœ¯ æ•°å€¼åˆ†æè¯„ä¼°æŠ€æœ¯  å°è£…ç±»å‹  é›†æˆå™¨ä»¶çš„å°è£…é—®é¢˜1ï¼šå›ç­”å…³äºSIPã€ZIPå’ŒDIPå°è£…ç±»å‹çš„ä»¥ä¸‹é—®é¢˜ã€‚ 1. ä¸ZIPç›¸æ¯”ï¼ŒSIPçš„ä¼˜ç¼ºç‚¹æ˜¯ä»€ä¹ˆï¼Ÿ  SIPçš„ä¼˜ç‚¹ï¼šç»“æ„æ¯”ZIPç®€å•ï¼Œå› æ­¤åˆ¶é€ æˆæœ¬ä½ä¸”æ›´å¯é ã€‚ SIPçš„ç¼ºç‚¹ï¼šå®‰è£…å¯†åº¦ä½äºZIPã€‚  2. ä¸DIPç›¸æ¯”ï¼ŒSIPçš„ä¼˜ç¼ºç‚¹æ˜¯ä»€ä¹ˆï¼Ÿ  SIPçš„ä¼˜ç‚¹ï¼šç»“æ„æ¯”DIPç®€å•ï¼Œå› æ­¤åˆ¶é€ æˆæœ¬ä½ä¸”æ›´å¯é ã€‚ SIPçš„ç¼ºç‚¹ï¼šå®‰è£…å¯†åº¦ä½äºDIPã€‚  é—®é¢˜2ï¼šå›ç­”å…³äºSOPã€SOJå’ŒSONå°è£…ç±»å‹çš„ä»¥ä¸‹é—®é¢˜ã€‚ 1. ä¸SOJç›¸æ¯”ï¼ŒSO...</div></div></div></a><a class="pagination-related" href="/2024/01/18/en/Summary-for-Packaging-and-Assembly-Technologies-for-Integrated-Systems/" title="Summary for Packaging and Assembly Technologies for Integrated Systems"><img class="cover" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/Summary-for-Packaging-and-Asse/8871f973cd33.png" alt="cover"><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2024-01-19</div><div class="info-item-2">Summary for Packaging and Assembly Technologies for Integrated Systems</div></div><div class="info-2"><div class="info-item-1"> æœ¬æ–‡è¿ç§»è‡ªCSDNåšå®¢åŸæ–‡é“¾æ¥ï¼šSummary for Packaging and Assembly Technologies for Integrated SystemsğŸ“Š 1035 é˜…è¯» | ğŸ‘ 23 ç‚¹èµ | ğŸ’¬ 2 è¯„è®º | â­ 25 æ”¶è—  ç›®å½•   Introduction Type of Packages: Packaging of integrated devices Question 1: Question 2: Question 3: Question 4: Question 5: Report 1: Front-end and back-end process Question 6: Question 7: Inspection Process Report 2: Prototyping and mass production Question 8: Question 9: Question 10:  Question 11: Report 3: Printed circuit boards (PCBs) What is PCB? Question ...</div></div></div></a></div></div><hr class="custom-hr"/><div id="post-comment"><div class="comment-head"><div class="comment-headline"><i class="fas fa-comments fa-fw"></i><span> Comments</span></div></div><div class="comment-wrap"><div><div id="twikoo-wrap"></div></div></div></div></div><div class="aside-content" id="aside-content"><div class="card-widget card-info text-center"><div class="avatar-img"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/img/avatar.png" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/></div><div class="author-info-name">Likun Wang (ç‹ç«‹å¤)</div><div class="author-info-description">Algorithm Engineer | AI Engineer
Waseda University M.E.
Deep Learning & TPU Development
</div><div class="site-data"><a href="/archives/"><div class="headline">Articles</div><div class="length-num">302</div></a><a href="/tags/"><div class="headline">Tags</div><div class="length-num">311</div></a><a href="/categories/"><div class="headline">Categories</div><div class="length-num">21</div></a></div><a id="card-info-btn" target="_blank" rel="noopener" href="https://github.com/veckun"><i class="fab fa-github"></i><span>Follow Me</span></a><div class="card-info-social-icons"><a class="social-icon" href="https://github.com/veckun" target="_blank" title="Github"><i class="fab fa-github" style="color: #24292e;"></i></a><a class="social-icon" href="mailto:vector_kun@ruri.waseda.jp" target="_blank" title="Email"><i class="fas fa-envelope" style="color: #4a7dbe;"></i></a><a class="social-icon" href="https://linkedin.com/in/veckun" target="_blank" title="LinkedIn"><i class="fab fa-linkedin" style="color: #0077b5;"></i></a><a class="social-icon" href="javascript:void(0)" target="_blank" title="WeChat vectorkun"><i class="fab fa-weixin" style="color: #07c160;"></i></a><a class="social-icon" href="https://blog.csdn.net/m0_59180666" target="_blank" title="CSDN"><i class="fab fa-blogger" style="color: #fc5531;"></i></a></div></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn fa-shake"></i><span>Announcement</span></div><div class="announcement_content">Welcome to my blog! Sharing AI, algorithms & tech insights.</div></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>Contents</span><span class="toc-percentage"></span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-3"><a class="toc-link" href="#Basic-Introduction"><span class="toc-number">1.</span> <span class="toc-text">Basic Introduction</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#Historical-Events"><span class="toc-number">1.1.</span> <span class="toc-text">Historical Events</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#Bioinformatics-vs-Computational-Biology"><span class="toc-number">1.1.1.</span> <span class="toc-text">Bioinformatics vs. Computational Biology</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Levels-of-Bioinfo-CompBio"><span class="toc-number">1.1.2.</span> <span class="toc-text">Levels of Bioinfo &amp; CompBio</span></a></li></ol></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Molecular-Biology-Primer"><span class="toc-number">1.2.</span> <span class="toc-text">Molecular Biology Primer</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#Homework-1"><span class="toc-number">1.2.1.</span> <span class="toc-text">Homework 1</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#A-brief-Introduction-to-DNA-RNA"><span class="toc-number">1.2.2.</span> <span class="toc-text">A brief Introduction to DNA &amp; RNA</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#A-brief-Introduction-to-Protein"><span class="toc-number">1.2.3.</span> <span class="toc-text">A brief Introduction to Protein</span></a></li></ol></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Sequencing-Technologies"><span class="toc-number">1.3.</span> <span class="toc-text">Sequencing Technologies</span></a></li></ol></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Sequence-Alignment"><span class="toc-number">2.</span> <span class="toc-text">Sequence Alignment</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#Pairwise-sequence-alignment"><span class="toc-number">2.1.</span> <span class="toc-text">Pairwise sequence alignment</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#Basic-notions"><span class="toc-number">2.1.1.</span> <span class="toc-text">Basic notions</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Goal-of-Pairwise-Alignment"><span class="toc-number">2.1.2.</span> <span class="toc-text">Goal of Pairwise Alignment:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Algorithms-for-G-L-Alignments"><span class="toc-number">2.1.3.</span> <span class="toc-text">Algorithms for G&amp;L Alignments</span></a></li></ol></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Homework-2"><span class="toc-number">2.2.</span> <span class="toc-text">Homework 2</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Database-similarity-searches"><span class="toc-number">2.3.</span> <span class="toc-text">Database similarity searches</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#BLAST-B-asic-L-ocal-A-lignment-S-earch-T-ool"><span class="toc-number">2.3.1.</span> <span class="toc-text">BLAST: B asic L ocal A lignment S earch T ool</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#FASTA-FAST-All"><span class="toc-number">2.3.2.</span> <span class="toc-text">FASTA: FAST-All</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Comparison-between-BLAST-FASTA"><span class="toc-number">2.3.3.</span> <span class="toc-text">Comparison between BLAST &amp; FASTA</span></a></li></ol></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Homework-3"><span class="toc-number">2.4.</span> <span class="toc-text">Homework 3</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Multiple-sequence-alignment"><span class="toc-number">2.5.</span> <span class="toc-text">Multiple sequence alignment</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#1-Progressive-alignment-methods"><span class="toc-number">2.5.1.</span> <span class="toc-text">1. Progressive alignment methods</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#2-Iterative-alignment-methods"><span class="toc-number">2.5.2.</span> <span class="toc-text">2. Iterative alignment methods</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#PSSM%E6%96%B9%E6%B3%95"><span class="toc-number">2.5.3.</span> <span class="toc-text">PSSMæ–¹æ³•</span></a></li></ol></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Motifs-and-domains"><span class="toc-number">2.6.</span> <span class="toc-text">Motifs and domains</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Homework-4"><span class="toc-number">2.7.</span> <span class="toc-text">Homework 4</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#My-understanding-to-Expectation-Maximization-EM-Algorithm"><span class="toc-number">2.7.1.</span> <span class="toc-text">My understanding to Expectation Maximization(EM) Algorithm:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Core-Algorithm"><span class="toc-number">2.7.2.</span> <span class="toc-text">Core Algorithm:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Code-in-PyCharm-python-3-8"><span class="toc-number">2.7.3.</span> <span class="toc-text">Code in PyCharm: (python&#x3D;3.8)</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#The-final-result"><span class="toc-number">2.7.4.</span> <span class="toc-text">The final result:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#The-difficulties-I-met"><span class="toc-number">2.7.5.</span> <span class="toc-text">The difficulties I met:</span></a></li></ol></li></ol></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Machine-Learning-Approaches"><span class="toc-number">3.</span> <span class="toc-text">Machine Learning Approaches</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#GA-for-sequencing-and-MSA"><span class="toc-number">3.1.</span> <span class="toc-text">GA for sequencing and MSA</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#Sequencing-by-Using-GA"><span class="toc-number">3.1.1.</span> <span class="toc-text">Sequencing by Using GA</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Multiple-Sequence-Alignments-Using-GA"><span class="toc-number">3.1.2.</span> <span class="toc-text">Multiple Sequence Alignments Using GA</span></a></li></ol></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Protein-Function-Prediction"><span class="toc-number">3.2.</span> <span class="toc-text">Protein Function Prediction</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#Gene-Function-Prediction-Using-SVM"><span class="toc-number">3.2.1.</span> <span class="toc-text">Gene Function Prediction Using SVM</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Protein-Function-Prediction-Using-DNN"><span class="toc-number">3.2.2.</span> <span class="toc-text">Protein Function Prediction Using DNN</span></a></li></ol></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Protein-Structure-Prediction"><span class="toc-number">3.3.</span> <span class="toc-text">Protein Structure Prediction</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#Protein-Folding-Based-on-EDA"><span class="toc-number">3.3.1.</span> <span class="toc-text">Protein Folding Based on EDA</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Protein-Folding-Based-on-Deep-Reinforcement-Learning"><span class="toc-number">3.3.2.</span> <span class="toc-text">Protein Folding Based on Deep Reinforcement Learning</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#AlphaFold2-for-Protein-Structure-Prediction"><span class="toc-number">3.3.3.</span> <span class="toc-text">AlphaFold2 for Protein Structure Prediction</span></a></li></ol></li></ol></li><li class="toc-item toc-level-3"><a class="toc-link" href="#References"><span class="toc-number">4.</span> <span class="toc-text">References</span></a></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>Recent Posts</span></div><div class="aside-list"><div class="aside-list-item"><a class="thumbnail" href="/2026/01/04/en/Kiro-Batch-Registration-Vulnerability-Whistleblower/" title="On 'I Report Myself': The 'Modern-Day Hero' Behind Kiro's Batch Registration Vulnerability"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/kiro-whistleblower.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="On 'I Report Myself': The 'Modern-Day Hero' Behind Kiro's Batch Registration Vulnerability"/></a><div class="content"><a class="title" href="/2026/01/04/en/Kiro-Batch-Registration-Vulnerability-Whistleblower/" title="On 'I Report Myself': The 'Modern-Day Hero' Behind Kiro's Batch Registration Vulnerability">On 'I Report Myself': The 'Modern-Day Hero' Behind Kiro's Batch Registration Vulnerability</a><time datetime="2026-01-04T15:00:00.000Z" title="Created 2026-01-05 00:00:00">2026-01-05</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2026/01/04/zh-CN/%E8%AE%BA%E3%80%8A%E6%88%91-%E4%B8%BE-%E6%8A%A5-%E6%88%91-%E8%87%AA-%E5%B7%B1%E3%80%8B%EF%BC%9AKiro-%E6%89%B9%E9%87%8F%E6%B3%A8%E5%86%8C%E6%BC%8F%E6%B4%9E%E8%83%8C%E5%90%8E%E7%9A%84%E2%80%9C%E5%BD%93%E4%BB%A3%E6%B4%BB%E9%9B%B7%E9%94%8B%E2%80%9D/" title="è®ºã€Šæˆ‘ ä¸¾ æŠ¥ æˆ‘ è‡ª å·±ã€‹ï¼šKiro æ‰¹é‡æ³¨å†Œæ¼æ´èƒŒåçš„&quot;å½“ä»£æ´»é›·é”‹&quot;"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/kiro-whistleblower.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="è®ºã€Šæˆ‘ ä¸¾ æŠ¥ æˆ‘ è‡ª å·±ã€‹ï¼šKiro æ‰¹é‡æ³¨å†Œæ¼æ´èƒŒåçš„&quot;å½“ä»£æ´»é›·é”‹&quot;"/></a><div class="content"><a class="title" href="/2026/01/04/zh-CN/%E8%AE%BA%E3%80%8A%E6%88%91-%E4%B8%BE-%E6%8A%A5-%E6%88%91-%E8%87%AA-%E5%B7%B1%E3%80%8B%EF%BC%9AKiro-%E6%89%B9%E9%87%8F%E6%B3%A8%E5%86%8C%E6%BC%8F%E6%B4%9E%E8%83%8C%E5%90%8E%E7%9A%84%E2%80%9C%E5%BD%93%E4%BB%A3%E6%B4%BB%E9%9B%B7%E9%94%8B%E2%80%9D/" title="è®ºã€Šæˆ‘ ä¸¾ æŠ¥ æˆ‘ è‡ª å·±ã€‹ï¼šKiro æ‰¹é‡æ³¨å†Œæ¼æ´èƒŒåçš„&quot;å½“ä»£æ´»é›·é”‹&quot;">è®ºã€Šæˆ‘ ä¸¾ æŠ¥ æˆ‘ è‡ª å·±ã€‹ï¼šKiro æ‰¹é‡æ³¨å†Œæ¼æ´èƒŒåçš„&quot;å½“ä»£æ´»é›·é”‹&quot;</a><time datetime="2026-01-04T15:00:00.000Z" title="Created 2026-01-05 00:00:00">2026-01-05</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2025/10/12/en/Common-Archive-Formats-Explained/" title="Common Archive Formats Explained: Differences and Extraction Methods"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="https://cdn.jsdelivr.net/gh/jerryc127/CDN@latest/cover/default_bg.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="Common Archive Formats Explained: Differences and Extraction Methods"/></a><div class="content"><a class="title" href="/2025/10/12/en/Common-Archive-Formats-Explained/" title="Common Archive Formats Explained: Differences and Extraction Methods">Common Archive Formats Explained: Differences and Extraction Methods</a><time datetime="2025-10-12T15:00:00.000Z" title="Created 2025-10-13 00:00:00">2025-10-13</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2025/10/12/zh-CN/%E5%B8%B8%E8%A7%81%E5%8E%8B%E7%BC%A9%E5%8C%85%E6%A0%BC%E5%BC%8F%E8%AF%A6%E8%A7%A3%EF%BC%9A%E5%8C%BA%E5%88%AB%E5%8F%8A%E5%9C%A8%E4%B8%8D%E5%90%8C%E7%B3%BB%E7%BB%9F%E4%B8%AD%E7%9A%84%E8%A7%A3%E5%8E%8B%E6%96%B9%E5%BC%8F/" title="å¸¸è§å‹ç¼©åŒ…æ ¼å¼è¯¦è§£ï¼šåŒºåˆ«åŠåœ¨ä¸åŒç³»ç»Ÿä¸­çš„è§£å‹æ–¹å¼"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="https://cdn.jsdelivr.net/gh/jerryc127/CDN@latest/cover/default_bg.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="å¸¸è§å‹ç¼©åŒ…æ ¼å¼è¯¦è§£ï¼šåŒºåˆ«åŠåœ¨ä¸åŒç³»ç»Ÿä¸­çš„è§£å‹æ–¹å¼"/></a><div class="content"><a class="title" href="/2025/10/12/zh-CN/%E5%B8%B8%E8%A7%81%E5%8E%8B%E7%BC%A9%E5%8C%85%E6%A0%BC%E5%BC%8F%E8%AF%A6%E8%A7%A3%EF%BC%9A%E5%8C%BA%E5%88%AB%E5%8F%8A%E5%9C%A8%E4%B8%8D%E5%90%8C%E7%B3%BB%E7%BB%9F%E4%B8%AD%E7%9A%84%E8%A7%A3%E5%8E%8B%E6%96%B9%E5%BC%8F/" title="å¸¸è§å‹ç¼©åŒ…æ ¼å¼è¯¦è§£ï¼šåŒºåˆ«åŠåœ¨ä¸åŒç³»ç»Ÿä¸­çš„è§£å‹æ–¹å¼">å¸¸è§å‹ç¼©åŒ…æ ¼å¼è¯¦è§£ï¼šåŒºåˆ«åŠåœ¨ä¸åŒç³»ç»Ÿä¸­çš„è§£å‹æ–¹å¼</a><time datetime="2025-10-12T15:00:00.000Z" title="Created 2025-10-13 00:00:00">2025-10-13</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2025/09/18/en/Master-Python-Debugger-pdb-in-10-Minutes/" title="Master Python Debugger pdb in 10 Minutes"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lazy-src="/images/posts/10%E5%88%86%E9%92%9F%E6%8E%8C%E6%8F%A1Python%E8%B0%83%E8%AF%95%E5%99%A8pdb/0d565b653736.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="Master Python Debugger pdb in 10 Minutes"/></a><div class="content"><a class="title" href="/2025/09/18/en/Master-Python-Debugger-pdb-in-10-Minutes/" title="Master Python Debugger pdb in 10 Minutes">Master Python Debugger pdb in 10 Minutes</a><time datetime="2025-09-18T15:00:00.000Z" title="Created 2025-09-19 00:00:00">2025-09-19</time></div></div></div></div></div></div></main><footer id="footer"><div class="footer-other"><div class="footer-copyright"><span class="copyright">&copy;&nbsp;2025 - 2026 By Likun Wang (ç‹ç«‹å¤)</span><span class="framework-info"><span>Framework </span><a target="_blank" rel="noopener" href="https://hexo.io">Hexo 8.1.1</a><span class="footer-separator">|</span><span>Theme </span><a target="_blank" rel="noopener" href="https://github.com/jerryc127/hexo-theme-butterfly">Butterfly 5.5.3</a></span></div><div class="footer_custom_text"><p>Likun Wang | Algorithm Engineer</p></div></div></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="Reading Mode"><i class="fas fa-book-open"></i></button><button id="darkmode" type="button" title="Toggle Between Light and Dark Mode"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="Toggle Between Single-column and Double-column"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside-config" type="button" title="Settings"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="Table of Contents"><i class="fas fa-list-ul"></i></button><a id="to_comment" href="#post-comment" title="Scroll to Comments"><i class="fas fa-comments"></i></a><button id="go-up" type="button" title="Back to Top"><span class="scroll-percent"></span><i class="fas fa-arrow-up"></i></button></div></div><div><script src="/js/utils.js?v=5.5.3"></script><script src="/js/main.js?v=5.5.3"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/ui@6.1.7/dist/fancybox/fancybox.umd.min.js"></script><script src="https://cdn.jsdelivr.net/npm/instant.page@5.2.0/instantpage.min.js" type="module"></script><script src="https://cdn.jsdelivr.net/npm/vanilla-lazyload@19.1.3/dist/lazyload.iife.min.js"></script><script src="https://cdn.jsdelivr.net/npm/node-snackbar@0.1.16/dist/snackbar.min.js"></script><div class="js-pjax"><script>(() => {
  const loadMathjax = () => {
    if (!window.MathJax) {
      window.MathJax = {
        loader: {
          load: [
            // Four font extension packages (optional)
            //- '[tex]/bbm',
            //- '[tex]/bboldx',
            //- '[tex]/dsfont',
            '[tex]/mhchem'
          ],
          paths: {
            'mathjax-newcm': '[mathjax]/../@mathjax/mathjax-newcm-font',

            //- // Four font extension packages (optional)
            //- 'mathjax-bbm-extension': '[mathjax]/../@mathjax/mathjax-bbm-font-extension',
            //- 'mathjax-bboldx-extension': '[mathjax]/../@mathjax/mathjax-bboldx-font-extension',
            //- 'mathjax-dsfont-extension': '[mathjax]/../@mathjax/mathjax-dsfont-font-extension',
            'mathjax-mhchem-extension': '[mathjax]/../@mathjax/mathjax-mhchem-font-extension'
          }
        },
        output: {
          font: 'mathjax-newcm',
        },
        tex: {
          inlineMath: [['$', '$'], ['\\(', '\\)']],
          tags: 'ams',
          packages: {
            '[+]': [
              'mhchem'
            ]
          }
        },
        chtml: {
          scale: 1.1
        },
        options: {
          enableMenu: true,
          menuOptions: {
            settings: {
              enrich: false  // Turn off Braille and voice narration text automatic generation
            }
          },
          renderActions: {
            findScript: [10, doc => {
              for (const node of document.querySelectorAll('script[type^="math/tex"]')) {
                const display = !!node.type.match(/; *mode=display/)
                const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display)
                const text = document.createTextNode('')
                node.parentNode.replaceChild(text, node)
                math.start = {node: text, delim: '', n: 0}
                math.end = {node: text, delim: '', n: 0}
                doc.math.push(math)
              }
            }, '']
          }
        }
      }

      const script = document.createElement('script')
      script.src = 'https://cdn.jsdelivr.net/npm/mathjax@4.0.0/tex-mml-chtml.min.js'
      script.id = 'MathJax-script'
      script.async = true
      document.head.appendChild(script)
    } else {
      MathJax.startup.document.state(0)
      MathJax.texReset()
      MathJax.typesetPromise()
    }
  }

  btf.addGlobalFn('encrypt', loadMathjax, 'mathjax')
  window.pjax ? loadMathjax() : window.addEventListener('load', loadMathjax)
})()</script><script>(() => {
  const runMermaid = ele => {
    window.loadMermaid = true
    const theme = document.documentElement.getAttribute('data-theme') === 'dark' ? 'dark' : 'default'

    ele.forEach((item, index) => {
      const mermaidSrc = item.firstElementChild
      const config = mermaidSrc.dataset.config ? JSON.parse(mermaidSrc.dataset.config) : {}
      if (!config.theme) {
        config.theme = theme
      }
      const mermaidThemeConfig = `%%{init: ${JSON.stringify(config)}}%%\n`
      const mermaidID = `mermaid-${index}`
      const mermaidDefinition = mermaidThemeConfig + mermaidSrc.textContent

      const renderFn = mermaid.render(mermaidID, mermaidDefinition)
      const renderMermaid = svg => {
        mermaidSrc.insertAdjacentHTML('afterend', svg)
      }

      // mermaid v9 and v10 compatibility
      typeof renderFn === 'string' ? renderMermaid(renderFn) : renderFn.then(({ svg }) => renderMermaid(svg))
    })
  }

  const codeToMermaid = () => {
    const codeMermaidEle = document.querySelectorAll('pre > code.mermaid')
    if (codeMermaidEle.length === 0) return

    codeMermaidEle.forEach(ele => {
      const preEle = document.createElement('pre')
      preEle.className = 'mermaid-src'
      preEle.hidden = true
      preEle.textContent = ele.textContent
      const newEle = document.createElement('div')
      newEle.className = 'mermaid-wrap'
      newEle.appendChild(preEle)
      ele.parentNode.replaceWith(newEle)
    })
  }

  const loadMermaid = () => {
    if (true) codeToMermaid()
    const $mermaid = document.querySelectorAll('#article-container .mermaid-wrap')
    if ($mermaid.length === 0) return

    const runMermaidFn = () => runMermaid($mermaid)
    btf.addGlobalFn('themeChange', runMermaidFn, 'mermaid')
    window.loadMermaid ? runMermaidFn() : btf.getScript('https://cdn.jsdelivr.net/npm/mermaid@11.12.2/dist/mermaid.min.js').then(runMermaidFn)
  }

  btf.addGlobalFn('encrypt', loadMermaid, 'mermaid')
  window.pjax ? loadMermaid() : document.addEventListener('DOMContentLoaded', loadMermaid)
})()</script><script>(() => {
  const isShuoshuo = GLOBAL_CONFIG_SITE.pageType === 'shuoshuo'
  const option = null

  const getCount = () => {
    const countELement = document.getElementById('twikoo-count')
    if(!countELement) return
    twikoo.getCommentsCount({
      envId: 'https://your-twikoo.vercel.app',
      region: '',
      urls: [window.location.pathname],
      includeReply: false
    }).then(res => {
      countELement.textContent = res[0].count
    }).catch(err => {
      console.error(err)
    })
  }

  const init = (el = document, path = location.pathname) => {
    twikoo.init({
      el: el.querySelector('#twikoo-wrap'),
      envId: 'https://your-twikoo.vercel.app',
      region: '',
      onCommentLoaded: () => {
        btf.loadLightbox(document.querySelectorAll('#twikoo .tk-content img:not(.tk-owo-emotion)'))
      },
      ...option,
      path: isShuoshuo ? path : (option && option.path) || path
    })

    GLOBAL_CONFIG_SITE.pageType === 'post' && getCount()

    isShuoshuo && (window.shuoshuoComment.destroyTwikoo = () => {
      if (el.children.length) {
        el.innerHTML = ''
        el.classList.add('no-comment')
      }
    })
  }

  const loadTwikoo = (el, path) => {
    if (typeof twikoo === 'object') setTimeout(() => init(el, path), 0)
    else btf.getScript('https://cdn.jsdelivr.net/npm/twikoo@1.6.44/dist/twikoo.all.min.js').then(() => init(el, path))
  }

  if (isShuoshuo) {
    'Twikoo' === 'Twikoo'
      ? window.shuoshuoComment = { loadComment: loadTwikoo }
      : window.loadOtherComment = loadTwikoo
    return
  }

  if ('Twikoo' === 'Twikoo' || !true) {
    if (true) btf.loadComment(document.getElementById('twikoo-wrap'), loadTwikoo)
    else loadTwikoo()
  } else {
    window.loadOtherComment = loadTwikoo
  }
})()</script></div><script src="/js/lang-router.js"></script><script src="/js/csdn-stats.js"></script><script src="https://cdn.jsdelivr.net/npm/pjax@0.2.8/pjax.min.js" defer="defer"></script><script>document.addEventListener('DOMContentLoaded', () => {
  const pjaxSelectors = ["head > title","#config-diff","#body-wrap","#rightside-config-hide","#rightside-config-show",".js-pjax"]

  window.pjax = new Pjax({
    elements: 'a:not([target="_blank"])',
    selectors: pjaxSelectors,
    cacheBust: false,
    analytics: false,
    scrollRestoration: false
  })

  const triggerPjaxFn = (val) => {
    if (!val) return
    Object.values(val).forEach(fn => {
      try {
        fn()
      } catch (err) {
        console.debug('Pjax callback failed:', err)
      }
    })
  }

  document.addEventListener('pjax:send', () => {
    // removeEventListener
    btf.removeGlobalFnEvent('pjaxSendOnce')
    btf.removeGlobalFnEvent('themeChange')

    // reset readmode
    const $bodyClassList = document.body.classList
    if ($bodyClassList.contains('read-mode')) $bodyClassList.remove('read-mode')

    triggerPjaxFn(window.globalFn.pjaxSend)
  })

  document.addEventListener('pjax:complete', () => {
    btf.removeGlobalFnEvent('pjaxCompleteOnce')
    document.querySelectorAll('script[data-pjax]').forEach(item => {
      const newScript = document.createElement('script')
      const content = item.text || item.textContent || item.innerHTML || ""
      Array.from(item.attributes).forEach(attr => newScript.setAttribute(attr.name, attr.value))
      newScript.appendChild(document.createTextNode(content))
      item.parentNode.replaceChild(newScript, item)
    })

    triggerPjaxFn(window.globalFn.pjaxComplete)
  })

  document.addEventListener('pjax:error', e => {
    if (e.request.status === 404) {
      true
        ? pjax.loadUrl('/404.html')
        : window.location.href = e.request.responseURL
    }
  })
})</script><script async data-pjax src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script><div id="local-search"><div class="search-dialog"><nav class="search-nav"><span class="search-dialog-title">Search</span><i class="fas fa-spinner fa-pulse" id="loading-status" hidden="hidden"></i><button class="search-close-button"><i class="fas fa-times"></i></button></nav><div class="text-center" id="loading-database"><i class="fas fa-spinner fa-pulse"></i><span>  Loading Database</span></div><div class="local-search-input"><input placeholder="Search..." type="text"/></div><hr/><div id="local-search-results"></div><div class="ais-Pagination" id="local-search-pagination" style="display:none;"><ul class="ais-Pagination-list"></ul></div><div id="local-search-stats"></div></div><div id="search-mask"></div><script src="/js/search/local-search.js?v=5.5.3"></script></div></div></body></html>